// @twilio-paste/display-pill-group/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  DisplayPill: () => DisplayPill,
  DisplayPillGroup: () => DisplayPillGroup
});
module.exports = __toCommonJS(src_exports);

// src/DisplayPill.tsx
var import_anchor = require("@twilio-paste/anchor");
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));
var DisplayPill = React.forwardRef(
  ({ element = "DISPLAY_PILL", ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(import_box.Box, { as: "li", listStyleType: "none" }, /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        ...props.href ? (0, import_anchor.secureExternalLink)(props.href) : {},
        ref,
        element,
        as: props.href ? "a" : "div",
        alignItems: "center",
        backgroundColor: "colorBackgroundWeak",
        borderRadius: "borderRadiusPill",
        boxShadow: "shadowBorderWeaker",
        color: "colorTextWeak",
        columnGap: "space20",
        cursor: props.href ? "pointer" : "default",
        display: "flex",
        fontSize: "fontSize20",
        fontWeight: "fontWeightMedium",
        lineHeight: "lineHeight10",
        paddingX: "space30",
        paddingY: "space20",
        outline: "none",
        _hover: props.href ? {
          backgroundColor: "colorBackground",
          boxShadow: "shadowBorder",
          textDecoration: "none"
        } : void 0,
        _focus: {
          boxShadow: "shadowFocus",
          textDecoration: "none"
        }
      },
      props.children
    ));
  }
);
DisplayPill.displayName = "DisplayPill";

// src/DisplayPillGroup.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var DisplayPillGroup = React2.forwardRef(
  ({ element = "DISPLAY_PILL_GROUP", ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        element,
        ref,
        as: "ul",
        margin: "space0",
        padding: "space0",
        display: "flex",
        flexWrap: "wrap",
        rowGap: "space20",
        columnGap: "space20"
      },
      props.children
    );
  }
);
DisplayPillGroup.displayName = "DisplayPillGroup";
