// @twilio-paste/display-pill-group/dist/index.debug.es.js
// src/DisplayPill.tsx
import { secureExternalLink } from "@twilio-paste/anchor";
import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
import * as React from "react";
var DisplayPill = React.forwardRef(
  ({ element = "DISPLAY_PILL", ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(Box, { as: "li", listStyleType: "none" }, /* @__PURE__ */ React.createElement(
      Box,
      {
        ...safelySpreadBoxProps(props),
        ...props.href ? secureExternalLink(props.href) : {},
        ref,
        element,
        as: props.href ? "a" : "div",
        alignItems: "center",
        backgroundColor: "colorBackgroundWeak",
        borderRadius: "borderRadiusPill",
        boxShadow: "shadowBorderWeaker",
        color: "colorTextWeak",
        columnGap: "space20",
        cursor: props.href ? "pointer" : "default",
        display: "flex",
        fontSize: "fontSize20",
        fontWeight: "fontWeightMedium",
        lineHeight: "lineHeight10",
        paddingX: "space30",
        paddingY: "space20",
        outline: "none",
        _hover: props.href ? {
          backgroundColor: "colorBackground",
          boxShadow: "shadowBorder",
          textDecoration: "none"
        } : void 0,
        _focus: {
          boxShadow: "shadowFocus",
          textDecoration: "none"
        }
      },
      props.children
    ));
  }
);
DisplayPill.displayName = "DisplayPill";

// src/DisplayPillGroup.tsx
import { Box as Box2, safelySpreadBoxProps as safelySpreadBoxProps2 } from "@twilio-paste/box";
import * as React2 from "react";
var DisplayPillGroup = React2.forwardRef(
  ({ element = "DISPLAY_PILL_GROUP", ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      Box2,
      {
        ...safelySpreadBoxProps2(props),
        element,
        ref,
        as: "ul",
        margin: "space0",
        padding: "space0",
        display: "flex",
        flexWrap: "wrap",
        rowGap: "space20",
        columnGap: "space20"
      },
      props.children
    );
  }
);
DisplayPillGroup.displayName = "DisplayPillGroup";
export {
  DisplayPill,
  DisplayPillGroup
};
