// @twilio-paste/side-panel/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  SidePanel: () => SidePanel,
  SidePanelBadgeButton: () => SidePanelBadgeButton,
  SidePanelBody: () => SidePanelBody,
  SidePanelButton: () => SidePanelButton,
  SidePanelContainer: () => SidePanelContainer,
  SidePanelContext: () => SidePanelContext,
  SidePanelFooter: () => SidePanelFooter,
  SidePanelHeader: () => SidePanelHeader,
  SidePanelHeaderActions: () => SidePanelHeaderActions,
  SidePanelPushContentWrapper: () => SidePanelPushContentWrapper
});
module.exports = __toCommonJS(src_exports);

// src/SidePanel.tsx
var import_animation_library = require("@twilio-paste/animation-library");
var import_box = require("@twilio-paste/box");
var import_utils = require("@twilio-paste/utils");
var React2 = __toESM(require("react"));

// src/SidePanelContext.tsx
var React = __toESM(require("react"));
var SidePanelContext = React.createContext({});

// src/SidePanel.tsx
var StyledSidePanelWrapper = React2.forwardRef((props, ref) => /* @__PURE__ */ React2.createElement(
  import_box.Box,
  {
    ...props,
    display: "flex",
    ref,
    position: "sticky",
    zIndex: "zIndex30",
    top: props.top,
    right: 0,
    paddingRight: ["space0", "space40", "space40"],
    width: ["100%", "size40", "size40"],
    height: props.height
  }
));
StyledSidePanelWrapper.displayName = "StyledSidePanelWrapper";
var AnimatedStyledSidePanelWrapper = (0, import_animation_library.animated)(StyledSidePanelWrapper);
var config = {
  mass: 0.3,
  tension: 288,
  friction: 20
};
var transitionStyles = {
  from: { opacity: 0, transform: "translateX(100%)" },
  enter: { opacity: 1, transform: "translateX(0%)" },
  leave: { opacity: 0, transform: "translateX(100%)" },
  config
};
var mobileTransitionStyles = {
  from: { opacity: 0, transform: "translateY(100%)" },
  enter: { opacity: 1, transform: "translateY(0%)" },
  leave: { opacity: 0, transform: "translateY(100%)" },
  config
};
var SidePanel = React2.forwardRef(
  ({ element = "SIDE_PANEL", label, children, ...props }, ref) => {
    const { sidePanelId, isOpen } = React2.useContext(SidePanelContext);
    const { breakpointIndex } = (0, import_utils.useWindowSize)();
    const transitions = breakpointIndex === 0 ? (0, import_animation_library.useTransition)(isOpen, mobileTransitionStyles) : (0, import_animation_library.useTransition)(isOpen, transitionStyles);
    const screenSize = window.innerHeight;
    const sidePanelRef = React2.useRef(null);
    const mergedSidePanelRef = (0, import_utils.useMergeRefs)(sidePanelRef, ref);
    const [offsetY, setOffsetY] = React2.useState(0);
    React2.useEffect(() => {
      const boundingClientRect = sidePanelRef?.current?.getBoundingClientRect();
      setOffsetY(boundingClientRect?.y || 0);
    }, []);
    return /* @__PURE__ */ React2.createElement(React2.Fragment, null, transitions(
      (styles, item) => item && /* @__PURE__ */ React2.createElement(
        import_box.Box,
        {
          ...(0, import_box.safelySpreadBoxProps)(props),
          position: "absolute",
          role: "dialog",
          "aria-label": label,
          top: 0,
          right: 0,
          width: ["100%", "auto", "auto"],
          height: "100%",
          element,
          id: sidePanelId
        },
        /* @__PURE__ */ React2.createElement(
          AnimatedStyledSidePanelWrapper,
          {
            ref: mergedSidePanelRef,
            element: `ANIMATED_${element}_WRAPPER`,
            style: styles,
            height: screenSize - offsetY,
            top: offsetY
          },
          /* @__PURE__ */ React2.createElement(
            import_box.Box,
            {
              display: "flex",
              maxHeight: "100%",
              flexDirection: "column",
              width: ["100%", "size40", "size40"],
              borderStyle: "solid",
              borderRadius: ["borderRadius0", "borderRadius70", "borderRadius70"],
              borderWidth: "borderWidth10",
              borderColor: "colorBorderWeaker",
              backgroundColor: "colorBackgroundBody",
              marginTop: "space40",
              marginBottom: ["space0", "space40", "space40"],
              paddingBottom: "space70",
              overflowY: "hidden",
              element: `INNER_${element}`
            },
            children
          )
        )
      )
    ));
  }
);
SidePanel.displayName = "SidePanel";

// src/SidePanelPushContentWrapper.tsx
var import_animation_library2 = require("@twilio-paste/animation-library");
var import_box2 = require("@twilio-paste/box");
var import_theme = require("@twilio-paste/theme");
var import_utils2 = require("@twilio-paste/utils");
var React3 = __toESM(require("react"));
var StyledContentWrapper = React3.forwardRef((props, ref) => /* @__PURE__ */ React3.createElement(import_box2.Box, { ...props, as: "div", position: "relative", ref }));
StyledContentWrapper.displayName = "StyledContentWrapper";
var AnimatedStyledContentWrapper = (0, import_animation_library2.animated)(StyledContentWrapper);
var SidePanelPushContentWrapper = React3.forwardRef(
  ({ element = "SIDE_PANEL_PUSH_CONTENT_WRAPPER", children, ...props }, ref) => {
    const { isOpen } = React3.useContext(SidePanelContext);
    const theme = (0, import_theme.useTheme)();
    const { breakpointIndex } = (0, import_utils2.useWindowSize)();
    const styles = (0, import_animation_library2.useSpring)({
      marginRight: isOpen ? theme.sizes.size40 : "0rem",
      config: {
        mass: 0.3,
        tension: 288,
        friction: 20
      }
    });
    return /* @__PURE__ */ React3.createElement(
      AnimatedStyledContentWrapper,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        ref,
        style: breakpointIndex === 0 ? void 0 : styles,
        marginRight: ["space0", theme.sizes.size40],
        minWidth: "size0",
        element
      },
      children
    );
  }
);
SidePanelPushContentWrapper.displayName = "SidePanelPushContentWrapper";

// src/SidePanelHeader.tsx
var import_box3 = require("@twilio-paste/box");
var import_button = require("@twilio-paste/button");
var import_CloseIcon = require("@twilio-paste/icons/cjs/CloseIcon");
var React4 = __toESM(require("react"));
var SidePanelCloseButton = ({
  setIsOpen,
  i18nCloseSidePanelTitle,
  sidePanelId,
  isOpen,
  element
}) => {
  const internalButtonRef = React4.useRef(null);
  React4.useEffect(() => {
    internalButtonRef.current?.focus({ preventScroll: true });
  }, []);
  return /* @__PURE__ */ React4.createElement(
    import_button.Button,
    {
      variant: "secondary_icon",
      size: "reset",
      onClick: () => {
        setIsOpen(false);
      },
      ref: internalButtonRef,
      "aria-controls": sidePanelId,
      "aria-expanded": isOpen,
      "aria-label": i18nCloseSidePanelTitle,
      element
    },
    /* @__PURE__ */ React4.createElement(import_CloseIcon.CloseIcon, { decorative: false, title: i18nCloseSidePanelTitle, size: "sizeIcon50" })
  );
};
SidePanelCloseButton.displayName = "SidePanelCloseButton";
var SidePanelHeader = React4.forwardRef(
  ({ element = "SIDE_PANEL_HEADER", children, ...props }, ref) => {
    const { sidePanelId, isOpen, setIsOpen, i18nCloseSidePanelTitle } = React4.useContext(SidePanelContext);
    return /* @__PURE__ */ React4.createElement(
      import_box3.Box,
      {
        width: "100%",
        padding: "space70",
        display: "flex",
        columnGap: "space30",
        alignItems: "flex-start",
        justifyContent: "space-between",
        ...(0, import_box3.safelySpreadBoxProps)(props),
        ref,
        element
      },
      children,
      /* @__PURE__ */ React4.createElement(
        SidePanelCloseButton,
        {
          setIsOpen,
          i18nCloseSidePanelTitle,
          sidePanelId,
          isOpen,
          element: `${element}_CLOSE_BUTTON`
        }
      )
    );
  }
);
SidePanelHeader.displayName = "SidePanelHeader";

// src/SidePanelHeaderActions.tsx
var import_box4 = require("@twilio-paste/box");
var React5 = __toESM(require("react"));
var SidePanelHeaderActions = React5.forwardRef(
  ({ element = "SIDE_PANEL_HEADER_ACTIONS", children, ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement(
      import_box4.Box,
      {
        flexGrow: 1,
        display: "flex",
        justifyContent: "flex-end",
        columnGap: "space20",
        ...(0, import_box4.safelySpreadBoxProps)(props),
        ref,
        element
      },
      children
    );
  }
);
SidePanelHeaderActions.displayName = "SidePanelHeaderActions";

// src/SidePanelBody.tsx
var import_box5 = require("@twilio-paste/box");
var import_styling_library = require("@twilio-paste/styling-library");
var React6 = __toESM(require("react"));
var SidePanelBodyWrapper = import_styling_library.styled.div(({ theme }) => {
  const { colorBackgroundBody, colorBackgroundStrong, colorBackgroundStronger } = theme.backgroundColors;
  return (0, import_styling_library.css)({
    "::-webkit-scrollbar": {
      width: "6px"
    },
    "::-webkit-scrollbar-track": {
      marginTop: "8px",
      marginBottom: "8px",
      background: colorBackgroundBody,
      borderRadius: "10px"
    },
    "::-webkit-scrollbar-thumb": {
      background: colorBackgroundStrong,
      borderRadius: "10px"
    },
    "::-webkit-scrollbar-thumb:hover": {
      background: colorBackgroundStronger
    }
  });
});
var SidePanelBody = React6.forwardRef(
  ({ element = "SIDE_PANEL_BODY", children, ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(
      import_box5.Box,
      {
        ...(0, import_box5.safelySpreadBoxProps)(props),
        as: SidePanelBodyWrapper,
        width: "100%",
        height: "100%",
        paddingLeft: "space70",
        paddingRight: "18px",
        display: "flex",
        ref,
        element,
        overflowY: "scroll"
      },
      children
    );
  }
);
SidePanelBody.displayName = "SidePanelBody";

// src/SidePanelContainer.tsx
var import_box6 = require("@twilio-paste/box");
var import_uid_library = require("@twilio-paste/uid-library");
var React7 = __toESM(require("react"));
var SidePanelContainer = React7.forwardRef(
  ({
    element = "SIDE_PANEL_CONTAINER",
    children,
    sidePanelId,
    isOpen,
    setIsOpen,
    i18nCloseSidePanelTitle = "close the side panel",
    i18nOpenSidePanelTitle = "open the side panel",
    ...props
  }, ref) => {
    if (sidePanelId === void 0)
      sidePanelId = (0, import_uid_library.useUID)();
    return /* @__PURE__ */ React7.createElement(
      SidePanelContext.Provider,
      {
        value: { sidePanelId, isOpen, setIsOpen, i18nCloseSidePanelTitle, i18nOpenSidePanelTitle }
      },
      /* @__PURE__ */ React7.createElement(import_box6.Box, { ...(0, import_box6.safelySpreadBoxProps)(props), position: "relative", ref, element }, children)
    );
  }
);
SidePanelContainer.displayName = "SidePanelContainer";

// src/SidePanelButton.tsx
var import_button2 = require("@twilio-paste/button");
var React8 = __toESM(require("react"));
var SidePanelButton = React8.forwardRef(
  ({ children, element = "SIDE_PANEL_BUTTON", ...sidePanelButtonProps }, ref) => {
    const { i18nCloseSidePanelTitle, i18nOpenSidePanelTitle, isOpen, setIsOpen, sidePanelId } = React8.useContext(SidePanelContext);
    return /* @__PURE__ */ React8.createElement(
      import_button2.Button,
      {
        element,
        "aria-label": isOpen ? i18nCloseSidePanelTitle : i18nOpenSidePanelTitle,
        onClick: () => setIsOpen(!isOpen),
        "aria-expanded": isOpen,
        "aria-controls": sidePanelId,
        ...sidePanelButtonProps,
        ref
      },
      children
    );
  }
);
SidePanelButton.displayName = "SidePanelButton";

// src/SidePanelBadgeButton.tsx
var import_badge = require("@twilio-paste/badge");
var React9 = __toESM(require("react"));
var ButtonBadge = React9.forwardRef(function ButtonBadge2({ children, ...props }, ref) {
  return /* @__PURE__ */ React9.createElement(import_badge.Badge, { ...props, as: "button", ref }, children);
});
ButtonBadge.displayName = "ButtonBadge";
var SidePanelBadgeButton = React9.forwardRef(
  ({ children, element = "SIDE_PANEL_BADGE", ...sidePanelButtonProps }, ref) => {
    const { i18nCloseSidePanelTitle, i18nOpenSidePanelTitle, isOpen, setIsOpen, sidePanelId } = React9.useContext(SidePanelContext);
    return /* @__PURE__ */ React9.createElement(
      ButtonBadge,
      {
        ...sidePanelButtonProps,
        as: "button",
        element,
        "aria-label": isOpen ? i18nCloseSidePanelTitle : i18nOpenSidePanelTitle,
        onClick: () => setIsOpen(!isOpen),
        "aria-expanded": isOpen,
        "aria-controls": sidePanelId,
        ref
      },
      children
    );
  }
);
SidePanelBadgeButton.displayName = "SidePanelBadgeButton";

// src/SidePanelFooter.tsx
var import_box7 = require("@twilio-paste/box");
var React10 = __toESM(require("react"));
var SidePanelFooter = React10.forwardRef(
  ({ element = "SIDE_PANEL_FOOTER", variant = "default", children, justifyContent = "flex-start", ...props }, ref) => {
    return /* @__PURE__ */ React10.createElement(
      import_box7.Box,
      {
        width: "100%",
        paddingX: variant === "chat" ? "space50" : "space70",
        paddingBottom: "space50",
        paddingTop: variant === "chat" ? "space0" : "space50",
        boxShadow: variant === "chat" ? "none" : "shadow",
        marginBottom: "spaceNegative70",
        zIndex: "zIndex20",
        display: "flex",
        columnGap: "space40",
        rowGap: "space40",
        justifyContent,
        flexWrap: "wrap",
        ...(0, import_box7.safelySpreadBoxProps)(props),
        ref,
        element
      },
      children
    );
  }
);
SidePanelFooter.displayName = "SidePanelFooter";
