// @twilio-paste/radio-group/dist/index.debug.es.js
// src/Radio.tsx
import {
  BaseRadioCheckboxControl,
  BaseRadioCheckboxHelpText,
  BaseRadioCheckboxLabel,
  BaseRadioCheckboxLabelText
} from "@twilio-paste/base-radio-checkbox";
import { Box } from "@twilio-paste/box";
import { useUID } from "@twilio-paste/uid-library";
import * as React2 from "react";

// src/RadioContext.tsx
import * as React from "react";
var RadioContext = React.createContext({
  name: "",
  value: "",
  disabled: false,
  hasError: false,
  onChange: () => {
  }
});

// src/Radio.tsx
var HiddenRadio = React2.forwardRef((props, ref) => /* @__PURE__ */ React2.createElement(
  Box,
  {
    as: "input",
    type: "radio",
    size: "size0",
    border: "none",
    overflow: "hidden",
    padding: "space0",
    margin: "space0",
    whiteSpace: "nowrap",
    textTransform: "none",
    position: "absolute",
    clip: "rect(0 0 0 0)",
    ref,
    ...props
  }
));
HiddenRadio.displayName = "HiddenRadio";
var Radio = React2.forwardRef(
  ({
    id,
    name,
    element = "RADIO",
    value,
    checked,
    defaultChecked,
    disabled,
    hasError,
    onChange,
    children,
    helpText,
    ...props
  }, ref) => {
    if (checked != null && defaultChecked != null) {
      throw new Error(
        `[Paste Radio] Do not provide both 'defaultChecked' and 'checked' to Radio at the same time. Please consider if you want this component to be controlled or uncontrolled.`
      );
    }
    const [checkedState, setCheckedState] = React2.useState(defaultChecked);
    const radioGroupContext = React2.useContext(RadioContext);
    const helpTextId = useUID();
    const radioId = id ? id : useUID();
    const isControlled = React2.useMemo(() => checked !== void 0 || radioGroupContext.value !== "", []);
    const handleChange = React2.useCallback(
      (event) => {
        if (onChange) {
          onChange(event);
        } else {
          radioGroupContext.onChange(event);
        }
        if (!isControlled) {
          setCheckedState(event.target.checked);
        }
      },
      [onChange, radioGroupContext, isControlled]
    );
    const state = {
      name: name != null ? name : radioGroupContext.name,
      disabled: disabled != null ? disabled : radioGroupContext.disabled,
      hasError: hasError != null ? hasError : radioGroupContext.hasError
    };
    if (isControlled) {
      if (radioGroupContext.value !== "") {
        state.checked = radioGroupContext.value === value;
      } else {
        state.checked = checked;
      }
    } else {
      state.defaultChecked = defaultChecked;
    }
    const mergedChecked = isControlled ? state.checked || state.defaultChecked : checkedState;
    return /* @__PURE__ */ React2.createElement(
      Box,
      {
        element,
        position: "relative",
        display: "inline-flex",
        alignItems: "flex-start",
        flexDirection: "column",
        verticalAlign: "top"
      },
      /* @__PURE__ */ React2.createElement(
        HiddenRadio,
        {
          ...props,
          ...state,
          value,
          "aria-describedby": helpTextId,
          "aria-invalid": state.hasError,
          onChange: handleChange,
          id: radioId,
          ref
        }
      ),
      /* @__PURE__ */ React2.createElement(BaseRadioCheckboxLabel, { disabled: state.disabled, htmlFor: radioId }, /* @__PURE__ */ React2.createElement(
        BaseRadioCheckboxControl,
        {
          element: `${element}_CONTROL`,
          borderRadius: "borderRadiusCircle",
          disabled: state.disabled,
          type: "radio",
          _checkedAndDisabledSibling: {
            color: "colorTextWeaker"
          }
        },
        /* @__PURE__ */ React2.createElement(
          Box,
          {
            as: "span",
            element: `${element}_CONTROL_CIRCLE`,
            lineHeight: "lineHeight0",
            display: mergedChecked ? "block" : "none",
            color: "inherit",
            size: "sizeIcon10"
          },
          /* @__PURE__ */ React2.createElement("svg", { role: "img", "aria-hidden": true, width: "100%", height: "100%", viewBox: "0 0 16 16", fill: "none" }, /* @__PURE__ */ React2.createElement("circle", { fill: "currentColor", cx: 8, cy: 8, r: 3 }))
        )
      ), /* @__PURE__ */ React2.createElement(BaseRadioCheckboxLabelText, { element: `${element}_LABEL_TEXT`, fontWeight: "fontWeightMedium" }, children)),
      helpText && /* @__PURE__ */ React2.createElement(BaseRadioCheckboxHelpText, { element: `${element}_HELP_TEXT_WRAPPER`, helpTextId }, helpText)
    );
  }
);
Radio.displayName = "Radio";

// src/RadioGroup.tsx
import { InlineControlGroup } from "@twilio-paste/inline-control-group";
import * as React3 from "react";
var RadioGroup = React3.forwardRef(
  ({
    name,
    element = "RADIO_GROUP",
    value,
    onChange,
    disabled = false,
    errorText,
    children,
    i18nRequiredLabel = "(required)",
    ...props
  }, ref) => {
    const onChangeHandler = React3.useCallback(
      (event) => {
        if (onChange != null) {
          onChange(event.target.value);
        }
      },
      [onChange]
    );
    const contextValue = React3.useMemo(() => {
      return {
        name,
        value: value || "",
        disabled,
        hasError: errorText != null,
        onChange: onChangeHandler
      };
    }, [name, value, disabled, errorText, onChangeHandler]);
    return /* @__PURE__ */ React3.createElement(RadioContext.Provider, { value: contextValue }, /* @__PURE__ */ React3.createElement(
      InlineControlGroup,
      {
        element,
        ...props,
        disabled,
        errorText,
        name,
        ref,
        i18nRequiredLabel
      },
      children
    ));
  }
);
RadioGroup.displayName = "RadioGroup";
export {
  HiddenRadio,
  Radio,
  RadioGroup
};
