// @twilio-paste/meter/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Meter: () => Meter,
  MeterLabel: () => MeterLabel
});
module.exports = __toCommonJS(src_exports);

// src/Meter.tsx
var import_box = require("@twilio-paste/box");
var import_react_spectrum_library = require("@twilio-paste/react-spectrum-library");
var import_text = require("@twilio-paste/text");
var React = __toESM(require("react"));

// src/constants.ts
var LABEL_SUFFIX = "METER_LABEL";

// src/Meter.tsx
var Meter = React.forwardRef(
  ({ element = "METER", id, minLabel, maxLabel, ...props }, ref) => {
    const { value = 0, minValue = 0, maxValue = 100 } = props;
    let labelledBy = props["aria-labelledby"];
    if (labelledBy == null && props["aria-label"] == null && id != null) {
      labelledBy = `${id}${LABEL_SUFFIX}`;
    }
    const { meterProps } = (0, import_react_spectrum_library.useMeter)({
      ...props,
      "aria-labelledby": labelledBy
    });
    const percentage = (value - minValue) / (maxValue - minValue);
    const fillWidth = `${Math.round(percentage * 100)}%`;
    return /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        as: "div",
        ...meterProps,
        role: "meter",
        id,
        ref,
        width: "100%",
        position: "relative",
        element,
        "aria-labelledby": labelledBy
      },
      /* @__PURE__ */ React.createElement(
        import_box.Box,
        {
          height: "10px",
          backgroundColor: "colorBackgroundStrong",
          borderRadius: "borderRadiusPill",
          element: `${element}_BAR`
        },
        /* @__PURE__ */ React.createElement(
          import_box.Box,
          {
            width: fillWidth,
            height: "10px",
            backgroundColor: "colorBackgroundPrimaryStronger",
            borderTopLeftRadius: "borderRadiusPill",
            borderBottomLeftRadius: "borderRadiusPill",
            borderTopRightRadius: fillWidth === "100%" ? "borderRadiusPill" : "borderRadius10",
            borderBottomRightRadius: fillWidth === "100%" ? "borderRadiusPill" : "borderRadius10",
            element: `${element}_FILL`
          }
        )
      ),
      (minLabel || maxLabel) && /* @__PURE__ */ React.createElement(
        import_box.Box,
        {
          display: "flex",
          flexDirection: "row",
          justifyContent: "space-between",
          columnGap: "space20",
          marginTop: "space20",
          "aria-hidden": "true",
          overflowWrap: "anywhere",
          element: `${element}_MIN_MAX_WRAPPER`
        },
        minLabel ? /* @__PURE__ */ React.createElement(import_text.Text, { as: "span", color: "colorTextWeak", fontWeight: "fontWeightNormal", element: `${element}_MIN` }, minLabel) : /* @__PURE__ */ React.createElement("span", null),
        maxLabel ? /* @__PURE__ */ React.createElement(
          import_text.Text,
          {
            as: "span",
            color: "colorTextWeak",
            fontWeight: "fontWeightNormal",
            textAlign: "end",
            element: `${element}_MAX`
          },
          maxLabel
        ) : /* @__PURE__ */ React.createElement("span", null)
      )
    );
  }
);
Meter.displayName = "Meter";

// src/MeterLabel.tsx
var import_box2 = require("@twilio-paste/box");
var import_label = require("@twilio-paste/label");
var import_text2 = require("@twilio-paste/text");
var React2 = __toESM(require("react"));
var MeterLabel = React2.forwardRef(
  ({ element = "METER_LABEL", children, htmlFor, valueLabel, ...labelProps }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      import_box2.Box,
      {
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "flex-end",
        element: `${element}_WRAPPER`
      },
      /* @__PURE__ */ React2.createElement(import_label.Label, { ...labelProps, as: "div", element, id: `${htmlFor}${LABEL_SUFFIX}`, ref }, children),
      valueLabel && /* @__PURE__ */ React2.createElement(
        import_text2.Text,
        {
          as: "span",
          fontWeight: "fontWeightSemibold",
          textAlign: "end",
          marginBottom: "space20",
          marginLeft: "space20",
          "aria-hidden": "true",
          element: `${element}_VALUE_LABEL`
        },
        valueLabel
      )
    );
  }
);
MeterLabel.displayName = "MeterLabel";
