// @twilio-paste/spinner/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Spinner: () => Spinner
});
module.exports = __toCommonJS(src_exports);

// src/Spinner.tsx
var import_IconWrapper = require("@twilio-paste/icons/cjs/helpers/IconWrapper");
var import_theme = require("@twilio-paste/theme");
var import_uid_library = require("@twilio-paste/uid-library");
var React = __toESM(require("react"));

// src/constants.ts
var circleRadius = 45;
var circleGeometry = {
  cx: 50,
  cy: 50,
  r: circleRadius
};
var circleCircumference = Math.PI * 2 * circleRadius;

// src/styled.tsx
var import_styling_library2 = require("@twilio-paste/styling-library");

// src/keyframes.tsx
var import_styling_library = require("@twilio-paste/styling-library");
var SvgKeyframes = import_styling_library.keyframes`
  0%,
  15% {
    stroke-dashoffset: ${circleCircumference * 0.9999};
    transform: rotate(0);
  }

  50%,
  75% {
    stroke-dashoffset: ${circleCircumference * 0.2};
    transform: rotate(45deg);
  }

  100% {
    stroke-dashoffset: ${circleCircumference * 0.9999};
    transform: rotate(360deg);
  }
`;
var CircleKeyframes = import_styling_library.keyframes`
  0% {
    transform: rotateZ(0deg);
  }
  100% {
    transform: rotateZ(360deg);
  }
`;
var SafariSvgKeyframes = import_styling_library.keyframes`
  0%,
  75% {
    stroke-dashoffset: 90%;
  }

  100% {
    stroke-dashoffset: 90%;
    transform: rotate(360deg);
  }
`;

// src/styled.tsx
var StyledCircleTrack = import_styling_library2.styled.circle({
  transformOrigin: "center",
  opacity: 0.25
});
var AnimatedStyledCircle = import_styling_library2.styled.circle(({ show, isSafari }) => {
  return {
    transformOrigin: "center",
    animation: isSafari ? "none" : `1.5s ease-in-out infinite both ${CircleKeyframes}`,
    strokeDasharray: circleCircumference,
    opacity: show ? 1 : 0
  };
});
var StyledSvg = import_styling_library2.styled.svg(({ isSafari }) => {
  return {
    height: "100%",
    width: "100%",
    display: "block",
    animation: isSafari ? `1s infinite linear ${SafariSvgKeyframes}` : `4.25s linear infinite both ${SvgKeyframes}`
  };
});

// src/Spinner.tsx
var Spinner = React.forwardRef(
  ({ size, color = "currentColor", title, as, display, decorative, delay = 250, element = "SPINNER" }, ref) => {
    const titleId = `spinner-${(0, import_uid_library.useUID)()}`;
    const {
      borderWidths: { borderWidth40 }
    } = (0, import_theme.useTheme)();
    const [show, setShow] = React.useState(delay === 0);
    if (!decorative && title == null) {
      throw new Error("[Spinner]: Missing a title for non-decorative icon.");
    }
    React.useEffect(() => {
      if (delay === 0)
        return void 0;
      const showTimer = setTimeout(() => setShow(true), delay);
      return () => clearTimeout(showTimer);
    }, [delay]);
    const [isSafari, setIsSafari] = React.useState(null);
    React.useEffect(() => {
      setIsSafari(navigator.userAgent.includes("Safari") && !navigator.userAgent.includes("Chrome"));
    }, []);
    if (isSafari == null) {
      return null;
    }
    return /* @__PURE__ */ React.createElement(
      import_IconWrapper.IconWrapper,
      {
        as,
        element,
        display,
        size,
        color,
        "aria-hidden": decorative,
        ref
      },
      /* @__PURE__ */ React.createElement(
        StyledSvg,
        {
          viewBox: "0 0 100 100",
          xmlns: "http://www.w3.org/2000/svg",
          "aria-labelledby": titleId,
          isSafari
        },
        title ? /* @__PURE__ */ React.createElement("title", { id: titleId }, title) : null,
        /* @__PURE__ */ React.createElement("g", { strokeWidth: borderWidth40, stroke: "currentColor", strokeLinecap: "round", fill: "transparent" }, /* @__PURE__ */ React.createElement(StyledCircleTrack, { ...circleGeometry }), /* @__PURE__ */ React.createElement(AnimatedStyledCircle, { show, ...circleGeometry, isSafari }))
      )
    );
  }
);
Spinner.displayName = "Spinner";
