// @twilio-paste/sidebar/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Sidebar: () => Sidebar,
  SidebarBetaBadge: () => SidebarBetaBadge,
  SidebarBody: () => SidebarBody,
  SidebarCollapseButton: () => SidebarCollapseButton,
  SidebarContext: () => SidebarContext,
  SidebarFooter: () => SidebarFooter,
  SidebarHeader: () => SidebarHeader,
  SidebarHeaderIconButton: () => SidebarHeaderIconButton,
  SidebarHeaderLabel: () => SidebarHeaderLabel,
  SidebarNavigation: () => SidebarNavigation,
  SidebarNavigationDisclosure: () => SidebarNavigationDisclosure,
  SidebarNavigationDisclosureContent: () => SidebarNavigationDisclosureContent,
  SidebarNavigationDisclosureHeading: () => SidebarNavigationDisclosureHeading,
  SidebarNavigationDisclosureHeadingWrapper: () => SidebarNavigationDisclosureHeadingWrapper,
  SidebarNavigationItem: () => SidebarNavigationItem,
  SidebarNavigationSeparator: () => SidebarNavigationSeparator,
  SidebarOverlayContentWrapper: () => SidebarOverlayContentWrapper,
  SidebarPushContentWrapper: () => SidebarPushContentWrapper,
  useSidebarNavigationDisclosureState: () => import_disclosure_primitive.useDisclosurePrimitiveState
});
module.exports = __toCommonJS(src_exports);

// src/Sidebar.tsx
var import_animation_library = require("@twilio-paste/animation-library");
var import_box2 = require("@twilio-paste/box");
var import_theme = require("@twilio-paste/theme");
var import_uid_library = require("@twilio-paste/uid-library");
var import_utils = require("@twilio-paste/utils");
var React3 = __toESM(require("react"));

// src/SidebarContext.tsx
var React = __toESM(require("react"));
var SidebarContext = React.createContext({});

// src/SidebarSkipLinks.tsx
var import_anchor = require("@twilio-paste/anchor");
var import_box = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var SidebarSkipLinkContainer = ({
  element,
  children
}) => {
  return /* @__PURE__ */ React2.createElement(
    import_box.Box,
    {
      element,
      backgroundColor: "colorBackgroundBody",
      borderRadius: "borderRadius20",
      boxShadow: "shadowLow",
      left: 10,
      paddingX: "space60",
      paddingY: "space30",
      position: "fixed",
      top: 10,
      transform: "translateY(-150%)",
      transition: "transform 0.1s",
      zIndex: "zIndex90",
      display: "flex",
      columnGap: "space60",
      _focusWithin: { transform: "translateY(0%)" }
    },
    children
  );
};
SidebarSkipLinkContainer.displayName = "SkipLinkContainer";
var SidebarSkipLinks = ({
  element = "SIDEBAR_SKIPLINKS",
  mainContentSkipLinkID,
  sidebarNavigationSkipLinkID,
  topbarSkipLinkID,
  i18nMainContentSkipLinkText = "Skip to content",
  i18nNavigationSkipLinkText = "Skip to navigation",
  i18nTopbarSkipLinkText = "Skip to topbar"
}) => {
  return /* @__PURE__ */ React2.createElement(SidebarSkipLinkContainer, { element }, /* @__PURE__ */ React2.createElement(import_anchor.Anchor, { element: `${element}_LINK`, href: `#${mainContentSkipLinkID}` }, i18nMainContentSkipLinkText), /* @__PURE__ */ React2.createElement(import_anchor.Anchor, { element: `${element}_LINK`, href: `#${sidebarNavigationSkipLinkID}` }, i18nNavigationSkipLinkText), /* @__PURE__ */ React2.createElement(import_anchor.Anchor, { element: `${element}_LINK`, href: `#${topbarSkipLinkID}` }, i18nTopbarSkipLinkText));
};
SidebarSkipLinks.displayName = "SkipLinks";

// src/Sidebar.tsx
var StyledSidebar = React3.forwardRef((props, ref) => /* @__PURE__ */ React3.createElement(
  import_box2.Box,
  {
    ...props,
    as: "aside",
    overflow: "visible",
    display: "flex",
    flexDirection: "column",
    ref,
    borderRightStyle: "solid",
    borderRightWidth: "borderWidth10",
    borderRightColor: "colorBorderInverseWeaker",
    backgroundColor: "colorBackgroundInverse",
    zIndex: "zIndex50",
    position: "fixed",
    top: 0,
    left: 0,
    bottom: 0
  }
));
StyledSidebar.displayName = "StyledSidebar";
var AnimatedStyledSidebar = (0, import_animation_library.animated)(StyledSidebar);
var config = {
  mass: 0.3,
  tension: 288,
  friction: 20
};
var getCompactSpringConfig = (collapsed, sizeSidebar, sizeSidebarCompact) => ({
  opacity: 1,
  width: collapsed ? sizeSidebarCompact : sizeSidebar,
  transform: `translateX(0%)`,
  config
});
var getHiddenSpringConfig = (collapsed, sizeSidebar) => ({
  opacity: collapsed ? 0 : 1,
  width: sizeSidebar,
  transform: collapsed ? `translateX(-100%)` : `translateX(0%)`,
  config
});
var getMobileSpringConfig = (collapsed) => ({
  opacity: collapsed ? 0 : 1,
  width: "100%",
  transform: collapsed ? `translateX(-100%)` : `translateX(0%)`,
  config
});
var Sidebar = React3.forwardRef(
  ({
    collapsed = false,
    variant = "default",
    element = "SIDEBAR",
    mainContentSkipLinkID,
    sidebarNavigationSkipLinkID,
    topbarSkipLinkID,
    i18nMainContentSkipLinkText,
    i18nNavigationSkipLinkText,
    i18nTopbarSkipLinkText,
    children,
    ...props
  }, ref) => {
    const sidebarId = (0, import_uid_library.useUID)();
    const { breakpointIndex } = (0, import_utils.useWindowSize)();
    const theme = (0, import_theme.useTheme)();
    const isMobile = breakpointIndex === 0;
    const isCompact = variant === "compact";
    let springConfig;
    if (isMobile) {
      springConfig = getMobileSpringConfig(collapsed);
    } else if (isCompact) {
      springConfig = getCompactSpringConfig(collapsed, theme.sizes.sizeSidebar, theme.sizes.sizeSidebarCompact);
    } else {
      springConfig = getHiddenSpringConfig(collapsed, theme.sizes.sizeSidebar);
    }
    const styles = (0, import_animation_library.useSpring)(springConfig);
    return /* @__PURE__ */ React3.createElement(SidebarContext.Provider, { value: { collapsed, variant, sidebarId, sidebarNavigationSkipLinkID } }, /* @__PURE__ */ React3.createElement(
      SidebarSkipLinks,
      {
        element: `${element}_SKIPLINKS`,
        mainContentSkipLinkID,
        sidebarNavigationSkipLinkID,
        topbarSkipLinkID,
        i18nMainContentSkipLinkText,
        i18nNavigationSkipLinkText,
        i18nTopbarSkipLinkText
      }
    ), /* @__PURE__ */ React3.createElement(
      AnimatedStyledSidebar,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        ref,
        element,
        width: ["100%", isCompact && collapsed ? "sizeSidebarCompact" : "sizeSidebar"],
        style: styles,
        id: sidebarId
      },
      children
    ));
  }
);
Sidebar.displayName = "Sidebar";

// src/SidebarBody.tsx
var import_box3 = require("@twilio-paste/box");
var React4 = __toESM(require("react"));
var SidebarBody = React4.forwardRef(
  ({ element = "SIDEBAR_BODY", children, ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(
      import_box3.Box,
      {
        ...(0, import_box3.safelySpreadBoxProps)(props),
        as: "div",
        element,
        ref,
        maxHeight: "100%",
        overflow: "hidden",
        flexGrow: 1
      },
      children
    );
  }
);
SidebarBody.displayName = "SidebarBody";

// src/content-wrappers/SidebarPushContentWrapper.tsx
var import_animation_library2 = require("@twilio-paste/animation-library");
var import_box4 = require("@twilio-paste/box");
var import_theme2 = require("@twilio-paste/theme");
var import_utils2 = require("@twilio-paste/utils");
var React5 = __toESM(require("react"));
var StyledContentWrapper = React5.forwardRef((props, ref) => /* @__PURE__ */ React5.createElement(import_box4.Box, { ...props, as: "div", position: "relative", ref }));
StyledContentWrapper.displayName = "StyledContentWrapper";
var AnimatedStyledContentWrapper = (0, import_animation_library2.animated)(StyledContentWrapper);
var SidebarPushContentWrapper = React5.forwardRef(
  ({ collapsed = false, variant = "default", element = "SIDEBAR_PUSH_CONTENT_WRAPPER", children, ...props }, ref) => {
    const theme = (0, import_theme2.useTheme)();
    const isCompact = variant === "compact";
    const { breakpointIndex } = (0, import_utils2.useWindowSize)();
    const styles = (0, import_animation_library2.useSpring)({
      marginLeft: !collapsed ? theme.sizes.sizeSidebar : isCompact ? theme.sizes.sizeSidebarCompact : `0rem`,
      config: {
        mass: 0.3,
        tension: 288,
        friction: 20
      }
    });
    return /* @__PURE__ */ React5.createElement(
      AnimatedStyledContentWrapper,
      {
        ...(0, import_box4.safelySpreadBoxProps)(props),
        ref,
        element,
        style: breakpointIndex === void 0 || breakpointIndex === 0 ? void 0 : styles,
        marginLeft: ["space0", theme.sizes.sizeSidebar]
      },
      children
    );
  }
);
SidebarPushContentWrapper.displayName = "SidebarPushContentWrapper";

// src/content-wrappers/SidebarOverlayContentWrapper.tsx
var import_box5 = require("@twilio-paste/box");
var import_utils3 = require("@twilio-paste/utils");
var React6 = __toESM(require("react"));
var SidebarOverlayContentWrapper = React6.forwardRef(
  ({ variant = "default", element = "SIDEBAR_OVERLAY_CONTENT_WRAPPER", children, ...props }, ref) => {
    const { breakpointIndex } = (0, import_utils3.useWindowSize)();
    const isMobile = breakpointIndex === 0;
    if (!isMobile && variant === "compact") {
      return /* @__PURE__ */ React6.createElement(import_box5.Box, { display: "flex", justifyContent: "flex-start" }, /* @__PURE__ */ React6.createElement(import_box5.Box, { role: "presentation", width: "sizeSidebarCompact", height: "100%", minHeight: "1px", flexShrink: 0 }), /* @__PURE__ */ React6.createElement(import_box5.Box, { ...(0, import_box5.safelySpreadBoxProps)(props), element, as: "div", ref, width: "100%" }, children));
    }
    return /* @__PURE__ */ React6.createElement(import_box5.Box, { ...(0, import_box5.safelySpreadBoxProps)(props), element, as: "div", ref }, children);
  }
);
SidebarOverlayContentWrapper.displayName = "SidebarOverlayContentWrapper";

// src/header/SidebarHeader.tsx
var import_box6 = require("@twilio-paste/box");
var React7 = __toESM(require("react"));
var SidebarHeader = React7.forwardRef(
  ({ element = "SIDEBAR_HEADER", children, ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(
      import_box6.Box,
      {
        ...(0, import_box6.safelySpreadBoxProps)(props),
        element,
        ref,
        paddingX: "space60",
        minHeight: "sizeTopbar",
        borderBottomWidth: "borderWidth10",
        borderBottomStyle: "solid",
        borderBottomColor: "colorBorderInverseWeaker",
        display: "flex",
        alignItems: "center",
        justifyContent: "flex-start",
        flexShrink: 0,
        columnGap: "space50",
        overflow: "hidden"
      },
      children
    );
  }
);
SidebarHeader.displayName = "SidebarHeader";

// src/header/SidebarHeaderLabel.tsx
var import_box7 = require("@twilio-paste/box");
var React8 = __toESM(require("react"));
var SidebarHeaderLabel = React8.forwardRef(
  ({ element = "SIDEBAR_HEADER_LABEL", children, ...props }, ref) => {
    const { collapsed } = React8.useContext(SidebarContext);
    return /* @__PURE__ */ React8.createElement(
      import_box7.Box,
      {
        ref,
        ...(0, import_box7.safelySpreadBoxProps)(props),
        opacity: collapsed ? "0" : "1",
        element,
        fontSize: "fontSize30",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight30",
        letterSpacing: "-.02em",
        color: "colorTextInverse",
        whiteSpace: "nowrap",
        textOverflow: "ellipsis",
        overflow: "hidden",
        margin: "space0"
      },
      children
    );
  }
);
SidebarHeaderLabel.displayName = "SidebarHeaderLabel";

// src/header/SidebarHeaderIconButton.tsx
var import_button = require("@twilio-paste/button");
var React9 = __toESM(require("react"));
var SidebarHeaderIconButton = React9.forwardRef(
  ({ element = "SIDEBAR_HEADER_ICON_BUTTON", ...props }, ref) => {
    return /* @__PURE__ */ React9.createElement(import_button.Button, { ...props, size: "icon", variant: "inverse", element, ref });
  }
);
SidebarHeaderIconButton.displayName = "SidebarHeaderIconButton";

// src/footer/SidebarFooter.tsx
var import_box8 = require("@twilio-paste/box");
var React10 = __toESM(require("react"));
var SidebarFooter = React10.forwardRef(
  ({ element = "SIDEBAR_FOOTER", ...props }, ref) => {
    return /* @__PURE__ */ React10.createElement(
      import_box8.Box,
      {
        ...(0, import_box8.safelySpreadBoxProps)(props),
        flexShrink: 0,
        position: "relative",
        padding: "space60",
        transition: "padding-right 100ms ease",
        ref,
        element
      }
    );
  }
);
SidebarFooter.displayName = "SidebarFooter";

// src/footer/SidebarCollapseButton.tsx
var import_box9 = require("@twilio-paste/box");
var import_button2 = require("@twilio-paste/button");
var import_ChevronDoubleLeftIcon = require("@twilio-paste/icons/cjs/ChevronDoubleLeftIcon");
var import_ChevronDoubleRightIcon = require("@twilio-paste/icons/cjs/ChevronDoubleRightIcon");
var import_screen_reader_only = require("@twilio-paste/screen-reader-only");
var React11 = __toESM(require("react"));
var SidebarCollapseButton = React11.forwardRef(
  ({ i18nExpandLabel, i18nCollapseLabel, element = "SIDEBAR_COLLAPSE_BUTTON", ...props }, ref) => {
    const { collapsed, sidebarId } = React11.useContext(SidebarContext);
    return /* @__PURE__ */ React11.createElement(import_box9.Box, { display: "flex", justifyContent: "flex-end" }, /* @__PURE__ */ React11.createElement(
      import_button2.Button,
      {
        ref,
        size: "icon",
        variant: "inverse",
        element,
        ...props,
        tabIndex: 0,
        "aria-expanded": !collapsed,
        "aria-controls": sidebarId
      },
      collapsed ? /* @__PURE__ */ React11.createElement(React11.Fragment, null, /* @__PURE__ */ React11.createElement(import_ChevronDoubleRightIcon.ChevronDoubleRightIcon, { size: "sizeIcon20", decorative: true, element: `${element}_EXPAND` }), /* @__PURE__ */ React11.createElement(import_screen_reader_only.ScreenReaderOnly, null, i18nExpandLabel)) : /* @__PURE__ */ React11.createElement(React11.Fragment, null, /* @__PURE__ */ React11.createElement(import_ChevronDoubleLeftIcon.ChevronDoubleLeftIcon, { size: "sizeIcon20", decorative: true, element: `${element}_COLLAPSE` }), /* @__PURE__ */ React11.createElement(import_screen_reader_only.ScreenReaderOnly, null, i18nCollapseLabel))
    ));
  }
);
SidebarCollapseButton.displayName = "SidebarCollapseButton";

// src/navigation/SidebarNavigation.tsx
var import_box10 = require("@twilio-paste/box");
var import_styling_library = require("@twilio-paste/styling-library");
var React13 = __toESM(require("react"));

// src/navigation/SidebarNavigationContext.tsx
var React12 = __toESM(require("react"));
var SidebarNavigationContext = React12.createContext({});

// src/navigation/SidebarNavigation.tsx
var SidebarNavigationWrapper = import_styling_library.styled.nav(({ theme }) => {
  const { colorBackgroundInverse, colorBackgroundInverseStronger } = theme.backgroundColors;
  return (0, import_styling_library.css)({
    overflowScrolling: "touch",
    background: `linear-gradient(${colorBackgroundInverse} 33%, rgba(20,28,44, 0)),
    linear-gradient(rgba(20,28,44, 0), ${colorBackgroundInverse} 66%) 0 100%,
    radial-gradient(farthest-side at 50% 100%, rgba(57, 71, 98, 0.5), rgba(0,0,0,0)) 0 100%`,
    backgroundRepeat: "no-repeat",
    backgroundSize: "100% 15px, 100% 15px, 100% 5px, 100% 5px",
    backgroundAttachment: `local, local, scroll, scroll`,
    "::-webkit-scrollbar": {
      width: "6px"
    },
    "::-webkit-scrollbar-track": {
      marginTop: "8px",
      background: colorBackgroundInverse
    },
    "::-webkit-scrollbar-thumb": {
      background: colorBackgroundInverseStronger,
      borderRadius: "10px"
    },
    "::-webkit-scrollbar-thumb:hover": {
      background: colorBackgroundInverseStronger
    }
  });
});
var SidebarNavigation = React13.forwardRef(
  ({ element = "SIDEBAR_NAVIGATION", hideItemsOnCollapse = false, hierarchical = false, children, ...props }, ref) => {
    const { collapsed, sidebarNavigationSkipLinkID } = React13.useContext(SidebarContext);
    return /* @__PURE__ */ React13.createElement(
      SidebarNavigationContext.Provider,
      {
        value: {
          hideItemsOnCollapse,
          hierarchical
        }
      },
      /* @__PURE__ */ React13.createElement(
        import_box10.Box,
        {
          ...(0, import_box10.safelySpreadBoxProps)(props),
          as: SidebarNavigationWrapper,
          id: sidebarNavigationSkipLinkID,
          "aria-label": props["aria-label"],
          element,
          ref,
          transition: "opacity 150ms ease",
          maxHeight: "100%",
          overflowY: "auto",
          overflowX: "hidden",
          paddingY: "space50",
          paddingLeft: "space60",
          paddingRight: collapsed ? "space40" : "space60",
          flexGrow: 1
        },
        children
      )
    );
  }
);
SidebarNavigation.displayName = "SidebarNavigation";

// src/navigation/SidebarNavigationDisclosure.tsx
var import_box11 = require("@twilio-paste/box");
var import_disclosure_primitive = require("@twilio-paste/disclosure-primitive");
var React15 = __toESM(require("react"));

// src/navigation/SidebarNavigationDisclosureContext.tsx
var React14 = __toESM(require("react"));
var SidebarNavigationDisclosureContext = React14.createContext({});

// src/navigation/SidebarNavigationDisclosure.tsx
var SidebarNavigationDisclosure = React15.forwardRef(
  ({ children, element = "SIDEBAR_NAVIGATION_DISCLOSURE", state, ...props }, ref) => {
    const { collapsed } = React15.useContext(SidebarContext);
    const { disclosure: parentDisclosure } = React15.useContext(SidebarNavigationDisclosureContext);
    const disclosure = state || (0, import_disclosure_primitive.useDisclosurePrimitiveState)({ animated: false, ...props });
    const disclosureContext = {
      disclosure,
      nested: parentDisclosure != null
    };
    return /* @__PURE__ */ React15.createElement(SidebarNavigationDisclosureContext.Provider, { value: disclosureContext }, /* @__PURE__ */ React15.createElement(
      import_box11.Box,
      {
        ...(0, import_box11.safelySpreadBoxProps)(props),
        element,
        ref,
        display: collapsed ? "none" : "block",
        marginBottom: "space30"
      },
      children
    ));
  }
);
SidebarNavigationDisclosure.displayName = "SidebarNavigationDisclosure";

// src/navigation/SidebarNavigationDisclosureHeadingWrapper.tsx
var import_box12 = require("@twilio-paste/box");
var React16 = __toESM(require("react"));
var SidebarNavigationDisclosureHeadingWrapper = React16.forwardRef(({ element = "SIDEBAR_NAVIGATION_DISCLOSURE_HEADING_WRAPPER", ...props }, ref) => {
  return /* @__PURE__ */ React16.createElement(
    import_box12.Box,
    {
      ...(0, import_box12.safelySpreadBoxProps)(props),
      element,
      ref,
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      columnGap: "space30"
    }
  );
});
SidebarNavigationDisclosureHeadingWrapper.displayName = "SidebarNavigationDisclosureHeadingWrapper";

// src/navigation/SidebarNavigationDisclosureHeading.tsx
var import_box13 = require("@twilio-paste/box");
var import_disclosure_primitive2 = require("@twilio-paste/disclosure-primitive");
var import_ChevronDisclosureIcon = require("@twilio-paste/icons/cjs/ChevronDisclosureIcon");
var import_theme3 = require("@twilio-paste/theme");
var React17 = __toESM(require("react"));

// src/navigation/styles.tsx
var sidebarNavigationLabelStyles = {
  fontStyle: "normal",
  fontWeight: "fontWeightNormal",
  fontSize: "fontSize30",
  lineHeight: "lineHeight20",
  color: "colorTextInverseWeak",
  transition: "all 120ms ease",
  borderRadius: "borderRadius20",
  cursor: "pointer",
  display: "flex",
  flexGrow: 1,
  columnGap: "space20",
  outline: "none",
  paddingY: "space30",
  position: "relative",
  textDecoration: "none",
  _focus: {
    boxShadow: "shadowFocusInverseInset"
  }
};
var sidebarNavigationLabelNestedStyles = {
  ...sidebarNavigationLabelStyles,
  paddingY: "space20",
  fontSize: "fontSize20"
};
var sidebarNavigationLabelSelectedStyles = {
  fontWeight: "fontWeightSemibold",
  color: "colorTextInverse"
};
var sidebarNavigationItemStyles = {
  ...sidebarNavigationLabelStyles,
  background: "none",
  border: "none",
  outline: "none",
  overflow: "hidden",
  paddingLeft: "space30",
  paddingRight: "space30",
  marginBottom: "space30",
  columnGap: "space30",
  _hover: {
    textDecoration: "underline"
  }
};
var sidebarNavigationItemHierarchicalStyles = {
  paddingLeft: "space70"
};
var sidebarNavigationItemNestedStyles = {
  paddingY: "space20",
  flexGrow: 1,
  width: "100%",
  fontSize: "fontSize20",
  marginBottom: "space20",
  paddingLeft: "space80"
};
var sidebarNavigationItemCollapsedStyles = {
  color: "colorTextIconInverse",
  _hover: {
    background: "none",
    color: "colorTextInverse"
  }
};
var sidebarNavigationItemSelectedStyles = {
  ...sidebarNavigationLabelSelectedStyles,
  backgroundColor: "colorBackgroundInverseStrong",
  color: "colorTextInverse"
};

// src/navigation/SidebarNavigationDisclosureHeading.tsx
var useComputeDisclosureHeadingStyles = ({ nested, selected }) => {
  let styles = {};
  if (nested) {
    styles = sidebarNavigationLabelNestedStyles;
  } else {
    styles = sidebarNavigationLabelStyles;
  }
  if (selected) {
    styles = { ...styles, ...sidebarNavigationLabelSelectedStyles };
  }
  return styles;
};
var useAdjustIconColor = ({ icon, selected }) => {
  return React17.useMemo(() => {
    if (!icon) {
      return null;
    }
    if (icon && React17.isValidElement(icon) && !selected) {
      const iconElement = icon;
      return React17.cloneElement(iconElement, {
        color: "colorTextIconInverse"
      });
    }
    return icon;
  }, [icon, selected]);
};
var StyledDisclosureHeading = React17.forwardRef(
  ({ children, element = "SIDEBAR_NAVIGATION_DISCLOSURE_HEADING", selected, icon, ...props }, ref) => {
    const { collapsed, variant } = React17.useContext(SidebarContext);
    const [shouldIconMove, setShouldIconMove] = React17.useState(false);
    const { nested } = React17.useContext(SidebarNavigationDisclosureContext);
    const isExpanded = props["aria-expanded"];
    const theme = (0, import_theme3.useTheme)();
    const isCompact = variant === "compact";
    const [visible, setVisible] = React17.useState(!isCompact ? true : !isExpanded);
    const timeout = React17.useRef(0);
    React17.useEffect(() => {
      clearTimeout(timeout.current);
      if (!isCompact) {
        return;
      }
      timeout.current = setTimeout(() => {
        setVisible(!collapsed);
      }, 120);
    }, [collapsed, isCompact]);
    const disclosureHeadingStyles = useComputeDisclosureHeadingStyles({ nested, selected });
    const adjustedIcon = useAdjustIconColor({ icon, selected });
    return /* @__PURE__ */ React17.createElement(
      import_box13.Box,
      {
        ...(0, import_box13.safelySpreadBoxProps)(props),
        ref,
        element,
        onMouseEnter: () => setShouldIconMove(true),
        onMouseLeave: () => setShouldIconMove(false),
        ...disclosureHeadingStyles
      },
      /* @__PURE__ */ React17.createElement(
        import_box13.Box,
        {
          as: "span",
          element: `${element}_EXPAND_ICON`,
          display: "flex",
          color: "colorTextIconInverse",
          height: "sizeIcon20",
          width: "sizeIcon20",
          transition: "transform 150ms ease",
          transform: `
          ${shouldIconMove ? `translateX(${theme.space.space10})` : "translateX(0)"}
          ${isExpanded ? `rotate(90deg)` : "rotate(0deg)"}`
        },
        /* @__PURE__ */ React17.createElement(import_ChevronDisclosureIcon.ChevronDisclosureIcon, { color: "inherit", decorative: true, size: "sizeIcon20" })
      ),
      adjustedIcon,
      /* @__PURE__ */ React17.createElement(
        import_box13.Box,
        {
          as: "span",
          display: "block",
          marginLeft: "space20",
          transition: "all 120ms ease",
          float: visible ? "none" : "left",
          opacity: visible ? 1 : 0,
          whiteSpace: visible ? "normal" : "nowrap"
        },
        collapsed ? null : children
      )
    );
  }
);
StyledDisclosureHeading.displayName = "StyledDisclosureHeading";
var SidebarNavigationDisclosureHeading = React17.forwardRef(
  (props, ref) => {
    const { disclosure } = React17.useContext(SidebarNavigationDisclosureContext);
    return /* @__PURE__ */ React17.createElement(import_disclosure_primitive2.DisclosurePrimitive, { ...disclosure, ref, ...props, as: StyledDisclosureHeading });
  }
);
SidebarNavigationDisclosureHeading.displayName = "SidebarNavigationDisclosureHeading";

// src/navigation/SidebarNavigationDisclosureContent.tsx
var import_box14 = require("@twilio-paste/box");
var import_disclosure_primitive3 = require("@twilio-paste/disclosure-primitive");
var React18 = __toESM(require("react"));
var SidebarNavigationDisclosureContent = React18.forwardRef(
  ({ children, element = "SIDEBAR_NAVIGATION_DISCLOSURE_CONTENT", ...props }, ref) => {
    const { disclosure } = React18.useContext(SidebarNavigationDisclosureContext);
    return /* @__PURE__ */ React18.createElement(
      import_disclosure_primitive3.DisclosurePrimitiveContent,
      {
        ...disclosure,
        ...(0, import_box14.safelySpreadBoxProps)(props),
        as: import_box14.Box,
        element,
        paddingLeft: "space70",
        color: "colorTextInverseWeak",
        marginTop: "space20",
        ref
      },
      children
    );
  }
);
SidebarNavigationDisclosureContent.displayName = "SidebarNavigationDisclosureContent";

// src/navigation/SidebarNavigationItem.tsx
var import_anchor2 = require("@twilio-paste/anchor");
var import_box15 = require("@twilio-paste/box");
var React19 = __toESM(require("react"));
var SidebarNavigationItem = React19.forwardRef(
  ({ element = "SIDEBAR_NAVIGATION_ITEM", selected, children, icon, ...props }, ref) => {
    const { collapsed, variant } = React19.useContext(SidebarContext);
    const { disclosure } = React19.useContext(SidebarNavigationDisclosureContext);
    const { hideItemsOnCollapse, hierarchical } = React19.useContext(SidebarNavigationContext);
    const isCompact = variant === "compact";
    const [visible, setVisible] = React19.useState(!isCompact ? true : !collapsed);
    const timeout = React19.useRef(0);
    React19.useEffect(() => {
      clearTimeout(timeout.current);
      if (!isCompact) {
        return;
      }
      timeout.current = setTimeout(() => {
        setVisible(!collapsed);
      }, 120);
    }, [collapsed, isCompact]);
    const isNested = disclosure != null;
    const styles = React19.useMemo(
      () => ({
        ...sidebarNavigationItemStyles,
        ...hierarchical && sidebarNavigationItemHierarchicalStyles,
        ...isNested && sidebarNavigationItemNestedStyles,
        ...collapsed && sidebarNavigationItemCollapsedStyles,
        ...selected && sidebarNavigationItemSelectedStyles,
        display: collapsed && hideItemsOnCollapse ? "none" : "flex",
        width: collapsed ? "36px" : "100%"
      }),
      [isNested, selected, collapsed, hideItemsOnCollapse, hierarchical]
    );
    return /* @__PURE__ */ React19.createElement(
      import_box15.Box,
      {
        ...(0, import_box15.safelySpreadBoxProps)(props),
        ...props.href ? (0, import_anchor2.secureExternalLink)(props.href) : {},
        ref,
        element,
        as: "a",
        "aria-current": selected ? "page" : void 0,
        ...styles
      },
      icon && /* @__PURE__ */ React19.createElement(import_box15.Box, { as: "span", color: selected ? "colorTextInverse" : "colorTextIconInverse" }, icon),
      /* @__PURE__ */ React19.createElement(
        import_box15.Box,
        {
          as: "span",
          display: "flex",
          alignItems: "center",
          justifyContent: "space-between",
          columnGap: "space20",
          transition: "all 120ms ease",
          flexGrow: 1,
          float: visible ? "none" : "left",
          opacity: visible ? 1 : 0,
          whiteSpace: visible ? "normal" : "nowrap"
        },
        collapsed ? null : children
      )
    );
  }
);
SidebarNavigationItem.displayName = "SidebarNavigationItem";

// src/navigation/SidebarNavigationSeparator.tsx
var import_box16 = require("@twilio-paste/box");
var React20 = __toESM(require("react"));
var SidebarNavigationSeparator = React20.forwardRef(
  ({ element = "SIDEBAR_NAVIGATION_SEPARATOR", ...props }, ref) => {
    const { collapsed } = React20.useContext(SidebarContext);
    return /* @__PURE__ */ React20.createElement(
      import_box16.Box,
      {
        ...(0, import_box16.safelySpreadBoxProps)(props),
        display: collapsed ? "none" : "block",
        "aria-orientation": "horizontal",
        element,
        marginY: "space50",
        as: "hr",
        borderWidth: "borderWidth0",
        borderColor: "colorBorderInverseWeaker",
        borderStyle: "solid",
        borderBottomWidth: "borderWidth10",
        ref
      }
    );
  }
);
SidebarNavigationSeparator.displayName = "SidebarNavigationSeparator";

// src/SidebarBetaBadge.tsx
var import_badge = require("@twilio-paste/badge");
var import_box17 = require("@twilio-paste/box");
var React21 = __toESM(require("react"));
var SidebarBetaBadgeSpanStyles = {
  ...import_badge.badgeBaseStyles,
  paddingX: "space20",
  paddingY: "space10",
  boxShadow: "shadowBorderInverseNewWeaker",
  backgroundColor: "colorBackgroundInverseStrong",
  color: "colorTextInverseNew"
};
var SidebarBetaBadgeButtonStyles = {
  ...SidebarBetaBadgeSpanStyles,
  boxShadow: "shadowBorderBottomInverseNewWeaker",
  cursor: "pointer",
  _hover: { top: "1px", boxShadow: "none" },
  _focus: { top: "1px", boxShadow: "shadowFocusInverse" }
};
var SidebarBetaBadge = React21.forwardRef(
  ({ children, as, element = "SIDEBAR_BETA_BADGE", onClick, ...props }, ref) => {
    const resizedChildren = (0, import_badge.useResizeChildIcons)(children);
    const styles = as === "button" ? SidebarBetaBadgeButtonStyles : SidebarBetaBadgeSpanStyles;
    const handleOnClick = as === "button" ? onClick : void 0;
    return /* @__PURE__ */ React21.createElement(import_box17.Box, { ...(0, import_box17.safelySpreadBoxProps)(props), ...styles, as, ref, element, onClick: handleOnClick }, resizedChildren);
  }
);
SidebarBetaBadge.displayName = "SidebarBetaBadge";
