// @twilio-paste/data-grid/dist/index.debug.es.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));

// ../../../../node_modules/lodash/isObjectLike.js
var require_isObjectLike = __commonJS({
  "../../../../node_modules/lodash/isObjectLike.js"(exports, module) {
    function isObjectLike(value) {
      return value != null && typeof value == "object";
    }
    module.exports = isObjectLike;
  }
});

// ../../../../node_modules/lodash/_freeGlobal.js
var require_freeGlobal = __commonJS({
  "../../../../node_modules/lodash/_freeGlobal.js"(exports, module) {
    var freeGlobal = typeof global == "object" && global && global.Object === Object && global;
    module.exports = freeGlobal;
  }
});

// ../../../../node_modules/lodash/_root.js
var require_root = __commonJS({
  "../../../../node_modules/lodash/_root.js"(exports, module) {
    var freeGlobal = require_freeGlobal();
    var freeSelf = typeof self == "object" && self && self.Object === Object && self;
    var root = freeGlobal || freeSelf || Function("return this")();
    module.exports = root;
  }
});

// ../../../../node_modules/lodash/_Symbol.js
var require_Symbol = __commonJS({
  "../../../../node_modules/lodash/_Symbol.js"(exports, module) {
    var root = require_root();
    var Symbol2 = root.Symbol;
    module.exports = Symbol2;
  }
});

// ../../../../node_modules/lodash/_getRawTag.js
var require_getRawTag = __commonJS({
  "../../../../node_modules/lodash/_getRawTag.js"(exports, module) {
    var Symbol2 = require_Symbol();
    var objectProto = Object.prototype;
    var hasOwnProperty = objectProto.hasOwnProperty;
    var nativeObjectToString = objectProto.toString;
    var symToStringTag = Symbol2 ? Symbol2.toStringTag : void 0;
    function getRawTag(value) {
      var isOwn = hasOwnProperty.call(value, symToStringTag), tag = value[symToStringTag];
      try {
        value[symToStringTag] = void 0;
        var unmasked = true;
      } catch (e) {
      }
      var result = nativeObjectToString.call(value);
      if (unmasked) {
        if (isOwn) {
          value[symToStringTag] = tag;
        } else {
          delete value[symToStringTag];
        }
      }
      return result;
    }
    module.exports = getRawTag;
  }
});

// ../../../../node_modules/lodash/_objectToString.js
var require_objectToString = __commonJS({
  "../../../../node_modules/lodash/_objectToString.js"(exports, module) {
    var objectProto = Object.prototype;
    var nativeObjectToString = objectProto.toString;
    function objectToString(value) {
      return nativeObjectToString.call(value);
    }
    module.exports = objectToString;
  }
});

// ../../../../node_modules/lodash/_baseGetTag.js
var require_baseGetTag = __commonJS({
  "../../../../node_modules/lodash/_baseGetTag.js"(exports, module) {
    var Symbol2 = require_Symbol();
    var getRawTag = require_getRawTag();
    var objectToString = require_objectToString();
    var nullTag = "[object Null]";
    var undefinedTag = "[object Undefined]";
    var symToStringTag = Symbol2 ? Symbol2.toStringTag : void 0;
    function baseGetTag(value) {
      if (value == null) {
        return value === void 0 ? undefinedTag : nullTag;
      }
      return symToStringTag && symToStringTag in Object(value) ? getRawTag(value) : objectToString(value);
    }
    module.exports = baseGetTag;
  }
});

// ../../../../node_modules/lodash/_overArg.js
var require_overArg = __commonJS({
  "../../../../node_modules/lodash/_overArg.js"(exports, module) {
    function overArg(func, transform) {
      return function(arg) {
        return func(transform(arg));
      };
    }
    module.exports = overArg;
  }
});

// ../../../../node_modules/lodash/_getPrototype.js
var require_getPrototype = __commonJS({
  "../../../../node_modules/lodash/_getPrototype.js"(exports, module) {
    var overArg = require_overArg();
    var getPrototype = overArg(Object.getPrototypeOf, Object);
    module.exports = getPrototype;
  }
});

// ../../../../node_modules/lodash/isPlainObject.js
var require_isPlainObject = __commonJS({
  "../../../../node_modules/lodash/isPlainObject.js"(exports, module) {
    var baseGetTag = require_baseGetTag();
    var getPrototype = require_getPrototype();
    var isObjectLike = require_isObjectLike();
    var objectTag = "[object Object]";
    var funcProto = Function.prototype;
    var objectProto = Object.prototype;
    var funcToString = funcProto.toString;
    var hasOwnProperty = objectProto.hasOwnProperty;
    var objectCtorString = funcToString.call(Object);
    function isPlainObject(value) {
      if (!isObjectLike(value) || baseGetTag(value) != objectTag) {
        return false;
      }
      var proto = getPrototype(value);
      if (proto === null) {
        return true;
      }
      var Ctor = hasOwnProperty.call(proto, "constructor") && proto.constructor;
      return typeof Ctor == "function" && Ctor instanceof Ctor && funcToString.call(Ctor) == objectCtorString;
    }
    module.exports = isPlainObject;
  }
});

// ../../../../node_modules/lodash/isElement.js
var require_isElement = __commonJS({
  "../../../../node_modules/lodash/isElement.js"(exports, module) {
    var isObjectLike = require_isObjectLike();
    var isPlainObject = require_isPlainObject();
    function isElement2(value) {
      return isObjectLike(value) && value.nodeType === 1 && !isPlainObject(value);
    }
    module.exports = isElement2;
  }
});

// src/DataGrid.tsx
import { Composite, useCompositeState } from "@twilio-paste/reakit-library";
import { Table } from "@twilio-paste/table";
import { useUID } from "@twilio-paste/uid-library";
import * as React2 from "react";

// src/DataGridContext.tsx
import * as React from "react";
var DataGridContext = React.createContext({
  actionable: false,
  striped: false
});

// src/utils/reakit/reakit-matches.ts
function matches(element, selectors) {
  if ("matches" in element) {
    return element.matches(selectors);
  }
  if ("msMatchesSelector" in element) {
    return element.msMatchesSelector(selectors);
  }
  return element.webkitMatchesSelector(selectors);
}

// src/utils/reakit/reakit-tabbable-modded.ts
var selector = "input:not([type='hidden']):not([disabled]), select:not([disabled]), textarea:not([disabled]), a[href], button:not([disabled]), [tabindex], iframe, object, embed, area[href], audio[controls], video[controls], [contenteditable]:not([contenteditable='false'])";
function isFocusable(element) {
  return matches(element, selector);
}
function getAllFocusableIn(container) {
  const allFocusable = Array.from(container.querySelectorAll(selector));
  return allFocusable.filter(isFocusable);
}
function getFirstFocusableIn(container) {
  const [first] = getAllFocusableIn(container);
  return first || null;
}

// src/utils/reakit/reakit-getDocument.ts
function getDocument(element) {
  return element ? element.ownerDocument || element : document;
}

// src/utils/reakit/reakit-getActiveElement-modded.ts
function getActiveElement(element) {
  const { activeElement } = getDocument(element);
  if (!activeElement?.nodeName) {
    return null;
  }
  return activeElement;
}

// src/utils/reakit/reakit-hasFocus.ts
function hasFocus(element) {
  const activeElement = getActiveElement(element);
  if (!activeElement)
    return false;
  if (activeElement === element)
    return true;
  const activeDescendant = activeElement.getAttribute("aria-activedescendant");
  if (!activeDescendant)
    return false;
  return activeDescendant === element.id;
}

// src/utils/reakit/reakit-ensureFocus.ts
function ensureFocus(element, { preventScroll, isActive = hasFocus } = {}) {
  if (isActive(element))
    return -1;
  element.focus({ preventScroll });
  if (isActive(element))
    return -1;
  return requestAnimationFrame(() => {
    element.focus({ preventScroll });
  });
}

// src/utils/cell-management.ts
var updateTabIndexForActionable = (cellElement, actionable) => {
  getAllFocusableIn(cellElement).forEach((element) => {
    element.tabIndex = actionable ? 0 : -1;
  });
};
var delayedSetFocusable = (element) => {
  setTimeout(() => {
    if (element != null) {
      element.tabIndex = 0;
    }
  }, 150);
};
var isCell = (element) => {
  return element.tagName === "TD" || element.tagName === "TH";
};
var getClosestCellFrom = (element, dataGridId) => {
  if (element == null)
    return null;
  if (!dataGridId)
    return null;
  return element.closest(`[id="${dataGridId}"] th, [id="${dataGridId}"] td`);
};
var getClosestGridCellFromCurrentFocus = (dataGridId) => {
  const focusedElement = getActiveElement();
  if (!focusedElement) {
    return null;
  }
  if (isCell(focusedElement)) {
    return focusedElement;
  }
  return getClosestCellFrom(focusedElement, dataGridId);
};

// src/DataGrid.tsx
var DataGrid = React2.forwardRef(
  ({ element = "DATA_GRID", role = "grid", striped = false, ...props }, ref) => {
    const dataGridId = `data-grid-${useUID()}`;
    const lastFocusedElement = React2.useRef(null);
    const compositeState = useCompositeState({ unstable_virtual: false });
    const [actionable, setActionable] = React2.useState(false);
    const handleMouseDown = React2.useCallback(() => {
      setActionable(true);
    }, []);
    const handleFocus = React2.useCallback((e) => {
      if (e.target != null) {
        lastFocusedElement.current = e.target;
      }
    }, []);
    const handleBlur = React2.useCallback(
      (event) => {
        const isDataGridBlurred = !event.currentTarget.contains(event.relatedTarget);
        if (isDataGridBlurred) {
          setActionable(false);
          if (lastFocusedElement.current != null) {
            const closestCell = getClosestCellFrom(lastFocusedElement.current, dataGridId);
            if (closestCell) {
              delayedSetFocusable(closestCell);
            }
          }
        }
      },
      [dataGridId]
    );
    const handleKeypress = React2.useCallback(
      (event) => {
        switch (event.key) {
          case "Enter": {
            setActionable(true);
            const activeElement = getActiveElement();
            if (activeElement && isCell(activeElement)) {
              const firstFocusableElement = getFirstFocusableIn(activeElement);
              if (firstFocusableElement) {
                ensureFocus(firstFocusableElement);
                activeElement.tabIndex = actionable ? 0 : -1;
              }
            }
            break;
          }
          case "Escape": {
            setActionable(false);
            const closestCell = getClosestGridCellFromCurrentFocus(dataGridId);
            if (closestCell) {
              ensureFocus(closestCell);
              delayedSetFocusable(closestCell);
            }
            break;
          }
          default:
            break;
        }
      },
      [actionable, dataGridId]
    );
    const dataGridState = {
      ...compositeState,
      actionable,
      striped
    };
    return /* @__PURE__ */ React2.createElement(DataGridContext.Provider, { value: dataGridState }, /* @__PURE__ */ React2.createElement(
      Composite,
      {
        ...props,
        ...compositeState,
        id: dataGridId,
        ref,
        as: Table,
        element,
        role,
        onKeyDown: handleKeypress,
        onMouseDown: handleMouseDown,
        onFocus: handleFocus,
        onBlur: handleBlur,
        isActionable: actionable,
        "data-actionable": actionable
      }
    ));
  }
);
DataGrid.displayName = "DataGrid";

// src/DataGridHead.tsx
import { THead } from "@twilio-paste/table";
import * as React3 from "react";
var DataGridHead = React3.forwardRef(
  ({ element = "DATA_GRID_HEAD", ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(THead, { ...props, element, ref });
  }
);
DataGridHead.displayName = "DataGridHead";

// src/DataGridRow.tsx
import { CompositeGroup } from "@twilio-paste/reakit-library";
import * as React5 from "react";

// src/table/Tr.tsx
import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
import { css, styled } from "@twilio-paste/styling-library";
import * as React4 from "react";
var StyledTr = styled.tr(
  css({
    "&[aria-selected=true] > td": {
      borderColor: "colorBorderPrimaryWeaker"
    },
    "&[aria-selected=true] > th": {
      borderColor: "colorBorderPrimaryWeaker"
    }
  })
);
var Tr = React4.forwardRef(
  ({ element = "DATA_GRID_TR", striped, ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(
      Box,
      {
        ...safelySpreadBoxProps(props),
        ref,
        element,
        as: StyledTr,
        "aria-selected": props.selected,
        _even: { backgroundColor: striped ? "colorBackgroundRowStriped" : "transparent" },
        _selected: {
          backgroundColor: "colorBackgroundPrimaryWeakest"
        }
      }
    );
  }
);
Tr.displayName = "Tr";

// src/DataGridRow.tsx
var DataGridRow = React5.forwardRef(
  ({ element = "DATA_GRID_ROW", ...props }, ref) => {
    const dataGridState = React5.useContext(DataGridContext);
    return /* @__PURE__ */ React5.createElement(CompositeGroup, { as: Tr, ...props, element, ...dataGridState, ref, role: "row" });
  }
);
DataGridRow.displayName = "DataGridRow";

// src/DataGridHeader.tsx
import * as React9 from "react";

// src/DataGridCell.tsx
var import_isElement = __toESM(require_isElement());
import { CompositeItem } from "@twilio-paste/reakit-library";
import { useMutationObservable } from "@twilio-paste/utils";
import * as React8 from "react";

// src/table/Td.tsx
import { Box as Box2, safelySpreadBoxProps as safelySpreadBoxProps2 } from "@twilio-paste/box";
import * as React6 from "react";
var Td = React6.forwardRef(
  ({ textAlign = "left", whiteSpace, element = "DATA_GRID_TD", ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(
      Box2,
      {
        ...safelySpreadBoxProps2(props),
        ref,
        as: "td",
        element,
        borderStyle: "solid",
        borderColor: "colorBorderWeaker",
        borderWidth: "borderWidth0",
        borderBottomWidth: "borderWidth10",
        fontSize: "fontSize30",
        lineHeight: "lineHeight30",
        padding: "space50",
        position: "relative",
        textAlign,
        whiteSpace,
        verticalAlign: "inherit",
        wordWrap: "break-word",
        color: "inherit",
        _first: {
          paddingLeft: "space50"
        },
        _last: {
          paddingRight: "space50"
        },
        _focus: {
          outline: "none",
          boxShadow: "shadowFocusInset"
        }
      }
    );
  }
);
Td.displayName = "Td";

// src/table/Th.tsx
import { Box as Box3, safelySpreadBoxProps as safelySpreadBoxProps3 } from "@twilio-paste/box";
import * as React7 from "react";
var Th = React7.forwardRef(
  ({ width, textAlign = "left", whiteSpace, element = "DATA_GRID_TH", ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(
      Box3,
      {
        ...safelySpreadBoxProps3(props),
        ref,
        element,
        as: "th",
        borderBottomStyle: "solid",
        borderBottomColor: "colorBorderWeaker",
        borderBottomWidth: "borderWidth10",
        fontSize: "fontSize30",
        lineHeight: "lineHeight30",
        fontWeight: "fontWeightSemibold",
        paddingY: "space40",
        paddingX: "space50",
        width,
        position: "relative",
        textAlign,
        whiteSpace,
        verticalAlign: "inherit",
        color: "inherit",
        _first: {
          paddingLeft: "space50"
        },
        _last: {
          paddingRight: "space50"
        },
        _focus: {
          outline: "none",
          boxShadow: "shadowFocusInset"
        }
      }
    );
  }
);
Th.displayName = "Th";

// src/DataGridCell.tsx
var DataGridCell = ({
  element = "DATA_GRID_CELL",
  as = "td",
  ...props
}) => {
  const dataGridState = React8.useContext(DataGridContext);
  const cellRef = React8.useRef();
  const handleMouseDown = React8.useCallback(() => {
    ensureFocus(cellRef.current);
  }, []);
  const onListMutation = React8.useCallback(
    (mutationList) => {
      if (mutationList.length === 0) {
        return;
      }
      mutationList.forEach((mutation) => {
        if (!(0, import_isElement.default)(mutation.target)) {
          return;
        }
        const target = mutation.target;
        if (mutation.attributeName === "tabindex" && dataGridState.actionable && target.tabIndex === 0 && isCell(target)) {
          target.tabIndex = -1;
        }
      });
    },
    [dataGridState.actionable]
  );
  useMutationObservable(cellRef.current, onListMutation);
  React8.useEffect(() => {
    setTimeout(() => {
      if (cellRef.current) {
        updateTabIndexForActionable(cellRef.current, dataGridState.actionable);
      }
    }, 10);
  }, [dataGridState.actionable]);
  return /* @__PURE__ */ React8.createElement(
    CompositeItem,
    {
      ...props,
      ...dataGridState,
      element,
      ref: cellRef,
      as: as === "td" ? Td : Th,
      onClick: handleMouseDown
    }
  );
};
DataGridCell.displayName = "DataGridCell";

// src/DataGridHeader.tsx
var DataGridHeader = ({
  element = "DATA_GRID_HEADER",
  ...props
}) => {
  return /* @__PURE__ */ React9.createElement(DataGridCell, { element, ...props, as: "th" });
};
DataGridHeader.displayName = "DataGridHeader";

// src/DataGridHeaderSort.tsx
import { Button } from "@twilio-paste/button";
import { ArrowDownIcon } from "@twilio-paste/icons/esm/ArrowDownIcon";
import { ArrowUpIcon } from "@twilio-paste/icons/esm/ArrowUpIcon";
import { UnsortedIcon } from "@twilio-paste/icons/esm/UnsortedIcon";
import { ScreenReaderOnly } from "@twilio-paste/screen-reader-only";
import * as React10 from "react";
var DataGridHeaderSortIcon = ({
  direction,
  element
}) => {
  switch (direction) {
    case "ascending":
      return /* @__PURE__ */ React10.createElement(ArrowUpIcon, { decorative: true, element });
    case "descending":
      return /* @__PURE__ */ React10.createElement(ArrowDownIcon, { decorative: true, element });
    case "none":
      return /* @__PURE__ */ React10.createElement(UnsortedIcon, { decorative: true, element });
    default:
      return null;
  }
};
DataGridHeaderSortIcon.displayName = "DataGridHeaderSortIcon";
var DataGridHeaderSort = ({
  direction,
  onClick,
  element = "DATA_GRID_HEADER_SORT",
  i18nAscendingLabel = "Sort ascending",
  i18nDescendingLabel = "Sort descending",
  i18nUnsortedLabel = "Unsorted",
  ...props
}) => {
  const i18nLabelDirectionMap = {
    ascending: i18nAscendingLabel,
    descending: i18nDescendingLabel,
    none: i18nUnsortedLabel
  };
  return /* @__PURE__ */ React10.createElement(Button, { element, variant: "reset", size: "reset", onClick, ...props }, /* @__PURE__ */ React10.createElement(DataGridHeaderSortIcon, { element: `${element}_ICON`, direction }), /* @__PURE__ */ React10.createElement(ScreenReaderOnly, null, i18nLabelDirectionMap[direction]));
};
DataGridHeaderSort.displayName = "DataGridHeaderSort";

// src/DataGridBody.tsx
import { TBody } from "@twilio-paste/table";
import * as React11 from "react";
var DataGridBody = React11.forwardRef(
  ({ element = "DATA_GRID_BODY", ...props }, ref) => {
    return /* @__PURE__ */ React11.createElement(TBody, { ...props, element, ref });
  }
);
DataGridBody.displayName = "DataGridBody";

// src/DataGridFoot.tsx
import { TFoot } from "@twilio-paste/table";
import * as React12 from "react";
var DataGridFoot = React12.forwardRef(
  ({ element = "DATA_GRID_FOOT", ...props }, ref) => {
    return /* @__PURE__ */ React12.createElement(TFoot, { ...props, element, ref });
  }
);
DataGridFoot.displayName = "DataGridFoot";
export {
  DataGrid,
  DataGridBody,
  DataGridCell,
  DataGridFoot,
  DataGridHead,
  DataGridHeader,
  DataGridHeaderSort,
  DataGridRow
};
