// @twilio-paste/side-modal/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  SideModal: () => SideModal,
  SideModalBody: () => SideModalBody,
  SideModalButton: () => SideModalButton,
  SideModalContainer: () => SideModalContainer,
  SideModalFooter: () => SideModalFooter,
  SideModalFooterActions: () => SideModalFooterActions,
  SideModalHeader: () => SideModalHeader,
  SideModalHeading: () => SideModalHeading,
  useSideModalState: () => useSideModalState
});
module.exports = __toCommonJS(src_exports);
var import_non_modal_dialog_primitive4 = require("@twilio-paste/non-modal-dialog-primitive");

// src/SideModal.tsx
var import_animation_library = require("@twilio-paste/animation-library");
var import_box = require("@twilio-paste/box");
var import_non_modal_dialog_primitive = require("@twilio-paste/non-modal-dialog-primitive");
var import_theme = require("@twilio-paste/theme");
var React2 = __toESM(require("react"));

// src/SideModalContext.tsx
var React = __toESM(require("react"));
var SideModalContext = React.createContext({});

// src/SideModal.tsx
var AnimatedBox = (0, import_animation_library.animated)(import_box.Box);
var getAnimationStates = () => ({
  from: { opacity: 0, transform: `translateX(100%)` },
  enter: { opacity: 1, transform: `translateX(0%)` },
  leave: { opacity: 0, transform: `translateX(100%)` },
  config: {
    mass: 0.5,
    tension: 220,
    friction: 20
  }
});
var SideModal = React2.forwardRef(
  ({ children, element = "SIDE_MODAL", hideOnEsc = true, ...props }, ref) => {
    const dialog = React2.useContext(SideModalContext);
    const transitions = (0, import_animation_library.useTransition)(dialog.visible, getAnimationStates());
    return /* @__PURE__ */ React2.createElement(
      import_non_modal_dialog_primitive.NonModalDialogPrimitive,
      {
        ...dialog,
        ...(0, import_box.safelySpreadBoxProps)(props),
        preventBodyScroll: false,
        hideOnClickOutside: false,
        hideOnEsc,
        element: `${element}_CONTAINER`,
        ref,
        as: import_box.Box,
        zIndex: "zIndex80",
        position: "fixed",
        top: "0 !important",
        right: "0 !important",
        bottom: "0 !important",
        left: "auto !important",
        transform: "none !important",
        _focus: {
          outline: "none"
        }
      },
      /* @__PURE__ */ React2.createElement(import_theme.StyledBase, null, transitions((styles, item) => {
        return item && /* @__PURE__ */ React2.createElement(
          AnimatedBox,
          {
            style: styles,
            element,
            display: "grid",
            gridTemplateRows: "auto 1fr auto",
            height: "100vh",
            backgroundColor: "colorBackgroundBody",
            boxShadow: "shadow",
            width: "size80"
          },
          children
        );
      }))
    );
  }
);
SideModal.displayName = "SideModal";

// src/SideModalButton.tsx
var import_button = require("@twilio-paste/button");
var import_non_modal_dialog_primitive2 = require("@twilio-paste/non-modal-dialog-primitive");
var React3 = __toESM(require("react"));
var SideModalButton = React3.forwardRef(
  ({ children, element = "SIDE_MODAL_BUTTON", ...props }, ref) => {
    const dialog = React3.useContext(SideModalContext);
    return /* @__PURE__ */ React3.createElement(import_non_modal_dialog_primitive2.NonModalDialogDisclosurePrimitive, { element, ...dialog, ...props, as: import_button.Button, ref }, children);
  }
);
SideModalButton.displayName = "SideModalButton";

// src/SideModalHeader.tsx
var import_box2 = require("@twilio-paste/box");
var import_button2 = require("@twilio-paste/button");
var import_CloseIcon = require("@twilio-paste/icons/cjs/CloseIcon");
var import_screen_reader_only = require("@twilio-paste/screen-reader-only");
var React4 = __toESM(require("react"));
var SideModalHeader = React4.forwardRef(
  ({ children, element = "SIDE_MODAL_HEADER", i18nDismissLabel = "close", ...props }, ref) => {
    const { hide } = React4.useContext(SideModalContext);
    return /* @__PURE__ */ React4.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        ref,
        paddingRight: "space90",
        paddingLeft: "space90",
        paddingTop: "space90",
        element,
        display: "flex",
        flexDirection: "row",
        alignItems: "flex-start"
      },
      /* @__PURE__ */ React4.createElement(import_box2.Box, { display: "flex", flexGrow: 1, marginRight: "space70" }, children),
      /* @__PURE__ */ React4.createElement(import_button2.Button, { element: `${element}_CLOSE_BUTTON`, variant: "secondary_icon", size: "reset", onClick: hide }, /* @__PURE__ */ React4.createElement(import_CloseIcon.CloseIcon, { element: `${element}_CLOSE_ICON`, decorative: true, size: "sizeIcon50" }), /* @__PURE__ */ React4.createElement(import_screen_reader_only.ScreenReaderOnly, null, i18nDismissLabel))
    );
  }
);
SideModalHeader.displayName = "SideModalHeader";

// src/SideModalHeading.tsx
var import_heading = require("@twilio-paste/heading");
var React5 = __toESM(require("react"));
var SideModalHeading = React5.forwardRef(
  ({ children, as = "h2", element = "SIDE_MODAL_HEADING", ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement(import_heading.Heading, { ...props, as, element, marginBottom: "space0", variant: "heading30", ref }, children);
  }
);
SideModalHeading.displayName = "SideModalHeading";

// src/SideModalContainer.tsx
var import_non_modal_dialog_primitive3 = require("@twilio-paste/non-modal-dialog-primitive");
var React6 = __toESM(require("react"));
var BaseSideModalContainer = ({
  gutter,
  placement,
  modal,
  children,
  state,
  ...initialState
}) => {
  const dialog = state || (0, import_non_modal_dialog_primitive3.useNonModalDialogPrimitiveState)({
    modal: true,
    unstable_fixed: true,
    ...initialState
  });
  return /* @__PURE__ */ React6.createElement(SideModalContext.Provider, { value: { ...dialog } }, children);
};
BaseSideModalContainer.displayName = "BaseSideModalContainer";
var SideModalContainer = React6.memo(BaseSideModalContainer);
SideModalContainer.displayName = "SideModalContainer";

// src/SideModalBody.tsx
var import_box3 = require("@twilio-paste/box");
var React7 = __toESM(require("react"));
var SideModalBody = React7.forwardRef(
  ({ children, element = "SIDE_MODAL_BODY", ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(import_box3.Box, { ...(0, import_box3.safelySpreadBoxProps)(props), element, ref, overflowY: "auto", padding: "space90" }, children);
  }
);
SideModalBody.displayName = "SideModalBody";

// src/SideModalFooter.tsx
var import_box4 = require("@twilio-paste/box");
var React8 = __toESM(require("react"));
var SideModalFooter = React8.forwardRef(
  ({ children, element = "SIDE_MODAL_FOOTER", ...props }, ref) => {
    return /* @__PURE__ */ React8.createElement(
      import_box4.Box,
      {
        ...(0, import_box4.safelySpreadBoxProps)(props),
        as: "div",
        element,
        ref,
        paddingRight: "space90",
        paddingLeft: "space90",
        paddingBottom: "space90",
        display: "flex"
      },
      children
    );
  }
);
SideModalFooter.displayName = "SideModalFooter";

// src/SideModalFooterActions.tsx
var import_box5 = require("@twilio-paste/box");
var React9 = __toESM(require("react"));
var SideModalFooterActions = React9.forwardRef(
  ({ children, element = "SIDE_MODAL_FOOTER_ACTIONS", justifyContent = "flex-start", ...props }, ref) => {
    return /* @__PURE__ */ React9.createElement(
      import_box5.Box,
      {
        ...(0, import_box5.safelySpreadBoxProps)(props),
        display: "flex",
        justifyContent,
        flexWrap: "wrap",
        flexGrow: 1,
        element,
        columnGap: "space50",
        alignItems: "center",
        ref
      },
      children
    );
  }
);
SideModalFooterActions.displayName = "SideModalFooterActions";

// src/index.tsx
var useSideModalState = (options) => {
  return (0, import_non_modal_dialog_primitive4.useNonModalDialogPrimitiveState)({ modal: true, unstable_fixed: true, ...options });
};
