// @twilio-paste/table/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  TBody: () => TBody,
  TFoot: () => TFoot,
  THead: () => THead,
  Table: () => Table,
  Td: () => Td,
  Th: () => Th,
  Tr: () => Tr
});
module.exports = __toCommonJS(src_exports);

// src/Table.tsx
var import_box = require("@twilio-paste/box");
var React2 = __toESM(require("react"));

// src/TableContext.tsx
var React = __toESM(require("react"));
var TableContext = React.createContext({ striped: false });

// src/Table.tsx
var Table = React2.forwardRef(
  ({
    element = "TABLE",
    id,
    isActionable,
    scrollHorizontally,
    noWrap,
    striped = false,
    tableLayout = "auto",
    variant = "default",
    ...props
  }, ref) => {
    const tableContext = {
      striped
    };
    return /* @__PURE__ */ React2.createElement(TableContext.Provider, { value: tableContext }, /* @__PURE__ */ React2.createElement(
      import_box.Box,
      {
        id,
        element: `${element}_WRAPPER`,
        overflowX: scrollHorizontally ? "auto" : null,
        whiteSpace: noWrap ? "nowrap" : null,
        boxShadow: isActionable ? "shadowFocus" : null
      },
      /* @__PURE__ */ React2.createElement(
        import_box.Box,
        {
          ...(0, import_box.safelySpreadBoxProps)(props),
          ref,
          as: "table",
          borderCollapse: "separate",
          borderColor: "colorBorderWeaker",
          borderSpacing: "0",
          borderStyle: "solid",
          borderWidth: variant === "borderless" ? "borderWidth0" : "borderWidth10",
          element,
          tableLayout: tableLayout === "fixed" ? "fixed" : "auto",
          width: "100%"
        }
      )
    ));
  }
);
Table.displayName = "Table";

// src/THead.tsx
var import_box2 = require("@twilio-paste/box");
var import_styling_library = require("@twilio-paste/styling-library");
var React3 = __toESM(require("react"));
var getStickyStyles = ({ stickyHeader }) => {
  if (!stickyHeader)
    return {};
  return {
    position: "sticky",
    zIndex: "zIndex10"
  };
};
var StyledTHead = import_styling_library.styled.thead(
  (0, import_styling_library.css)({
    backgroundColor: "colorBackgroundWeak",
    color: "colorTextWeak",
    top: (props) => props.top
  })
);
var THead = React3.forwardRef(
  ({ element = "THEAD", stickyHeader = false, top = stickyHeader ? "-1px" : null, ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        ref,
        as: StyledTHead,
        element,
        top,
        ...getStickyStyles({ stickyHeader })
      }
    );
  }
);
THead.displayName = "THead";

// src/TBody.tsx
var import_box3 = require("@twilio-paste/box");
var import_styling_library2 = require("@twilio-paste/styling-library");
var React4 = __toESM(require("react"));
var bodyCellStyles = {
  borderBottomWidth: "borderWidth0"
};
var StyledTBody = import_styling_library2.styled.tbody(
  (0, import_styling_library2.css)({
    backgroundColor: "colorBackgroundBody",
    "& > tr:last-of-type > td": bodyCellStyles,
    "& > tr:last-of-type > th": bodyCellStyles
  })
);
var TBody = React4.forwardRef(
  ({ element = "TBODY", children, ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(import_box3.Box, { as: StyledTBody, ...(0, import_box3.safelySpreadBoxProps)(props), ref, element, color: "colorText" }, children);
  }
);
TBody.displayName = "TBody";

// src/TFoot.tsx
var import_box4 = require("@twilio-paste/box");
var import_styling_library3 = require("@twilio-paste/styling-library");
var React5 = __toESM(require("react"));
var footerCellStyles = {
  borderBottomWidth: "borderWidth0",
  borderTopWidth: "borderWidth10",
  fontWeight: "fontWeightSemibold",
  paddingY: "space40"
};
var StyledTFoot = import_styling_library3.styled.tfoot(
  (0, import_styling_library3.css)({
    "& > tr > td": footerCellStyles,
    "& > tr > th": footerCellStyles
  })
);
var TFoot = React5.forwardRef(({ element = "TFOOT", ...props }, ref) => {
  return /* @__PURE__ */ React5.createElement(
    import_box4.Box,
    {
      ...(0, import_box4.safelySpreadBoxProps)(props),
      ref,
      as: StyledTFoot,
      backgroundColor: "colorBackgroundWeak",
      color: "colorTextWeak",
      element
    }
  );
});
TFoot.displayName = "TFoot";

// src/Tr.tsx
var import_box5 = require("@twilio-paste/box");
var React6 = __toESM(require("react"));
var Tr = React6.forwardRef(
  ({ element = "TR", verticalAlign = "middle", ...props }, ref) => {
    const { striped } = React6.useContext(TableContext);
    return /* @__PURE__ */ React6.createElement(
      import_box5.Box,
      {
        ...(0, import_box5.safelySpreadBoxProps)(props),
        ref,
        as: "tr",
        element,
        verticalAlign,
        _even: { backgroundColor: striped ? "colorBackgroundRowStriped" : "transparent" },
        _last: { borderWidth: "borderWidth0" }
      }
    );
  }
);
Tr.displayName = "Tr";

// src/Td.tsx
var import_box6 = require("@twilio-paste/box");
var React7 = __toESM(require("react"));
var Td = React7.forwardRef(
  ({ element = "TD", textAlign = "left", whiteSpace, ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(
      import_box6.Box,
      {
        ...(0, import_box6.safelySpreadBoxProps)(props),
        ref,
        as: "td",
        borderStyle: "solid",
        borderColor: "colorBorderWeaker",
        borderWidth: "borderWidth0",
        borderBottomWidth: "borderWidth10",
        element,
        fontSize: "fontSize30",
        lineHeight: "lineHeight30",
        padding: "space50",
        position: "relative",
        textAlign,
        whiteSpace,
        verticalAlign: "inherit",
        wordWrap: "break-word",
        color: "inherit"
      }
    );
  }
);
Td.displayName = "Td";

// src/Th.tsx
var import_box7 = require("@twilio-paste/box");
var React8 = __toESM(require("react"));
var Th = React8.forwardRef(
  ({ element = "TH", textAlign = "left", whiteSpace, width, ...props }, ref) => {
    return /* @__PURE__ */ React8.createElement(
      import_box7.Box,
      {
        ...(0, import_box7.safelySpreadBoxProps)(props),
        ref,
        as: "th",
        borderBottomStyle: "solid",
        borderBottomColor: "colorBorderWeaker",
        borderBottomWidth: "borderWidth10",
        element,
        fontSize: "fontSize30",
        lineHeight: "lineHeight30",
        fontWeight: "fontWeightSemibold",
        paddingBottom: "space40",
        paddingLeft: "space50",
        paddingRight: "space50",
        paddingTop: "space40",
        position: "relative",
        textAlign,
        whiteSpace,
        verticalAlign: "inherit",
        width,
        wordWrap: "break-word",
        color: "inherit"
      }
    );
  }
);
Th.displayName = "Th";
