// @twilio-paste/media-object/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  MediaBody: () => MediaBody,
  MediaFigure: () => MediaFigure,
  MediaObject: () => MediaObject
});
module.exports = __toCommonJS(src_exports);

// src/MediaObject.tsx
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));
var MediaObject = React.forwardRef(
  ({ as = "span", children, element = "MEDIA_OBJECT", marginBottom, marginTop, verticalAlign = "top", ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        alignItems: verticalAlign === "top" ? "flex-start" : "center",
        as,
        display: "flex",
        element,
        marginBottom,
        marginTop,
        ref
      },
      children
    );
  }
);
MediaObject.displayName = "MediaObject";

// src/MediaFigure.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var MediaFigure = React2.forwardRef(
  ({ align = "start", as = "span", children, element = "MEDIA_FIGURE", spacing = "space0", ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        as,
        display: "flex",
        element,
        flexShrink: 0,
        marginLeft: align === "end" ? spacing : void 0,
        marginRight: align === "start" ? spacing : void 0,
        ref
      },
      children
    );
  }
);
MediaFigure.displayName = "MediaFigure";

// src/MediaBody.tsx
var import_box3 = require("@twilio-paste/box");
var React3 = __toESM(require("react"));
var MediaBody = React3.forwardRef(
  ({ as = "span", children, element = "MEDIA_BODY", ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(import_box3.Box, { ...(0, import_box3.safelySpreadBoxProps)(props), as, element, flex: 1, minWidth: "size0", ref }, children);
  }
);
MediaBody.displayName = "MediaBody";
