// @twilio-paste/visual-picker/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  VisualPickerCheckbox: () => VisualPickerCheckbox,
  VisualPickerCheckboxGroup: () => VisualPickerCheckboxGroup,
  VisualPickerRadio: () => VisualPickerRadio,
  VisualPickerRadioGroup: () => VisualPickerRadioGroup
});
module.exports = __toCommonJS(src_exports);

// src/VisualPickerRadioGroup.tsx
var import_inline_control_group = require("@twilio-paste/inline-control-group");
var React2 = __toESM(require("react"));

// src/VisualPickerContext.tsx
var React = __toESM(require("react"));
var VisualPickerRadioContext = React.createContext({
  groupIsDisabled: false,
  name: "",
  groupHasError: false,
  onChange: () => {
  },
  value: "",
  orientation: "vertical"
});
var VisualPickerCheckboxContext = React.createContext({
  groupIsDisabled: false,
  name: "",
  groupHasError: false,
  orientation: "vertical"
});

// src/VisualPickerRadioGroup.tsx
var VisualPickerRadioGroup = React2.forwardRef(
  ({
    orientation = "vertical",
    children,
    disabled = false,
    legend,
    onChange,
    name,
    errorText,
    value,
    element = "VISUAL_PICKER_RADIO_GROUP",
    i18nRequiredLabel = "(required)",
    ...props
  }, ref) => {
    const onChangeHandler = React2.useCallback(
      (event) => {
        onChange(event.target.value);
      },
      [onChange]
    );
    const contextValue = React2.useMemo(() => {
      return {
        name,
        value: value || "",
        groupIsDisabled: disabled,
        groupHasError: errorText != null,
        onChange: onChangeHandler,
        orientation
      };
    }, [name, value, disabled, errorText, onChangeHandler, orientation]);
    return /* @__PURE__ */ React2.createElement(VisualPickerRadioContext.Provider, { value: contextValue }, /* @__PURE__ */ React2.createElement(
      import_inline_control_group.InlineControlGroup,
      {
        ...props,
        legend,
        ref,
        name,
        element,
        errorText,
        orientation,
        disabled,
        i18nRequiredLabel,
        fieldStyleProps: {
          marginTop: orientation === "horizontal" ? "space40" : "space30",
          marginRight: orientation === "horizontal" ? "space30" : void 0
        }
      },
      children
    ));
  }
);
VisualPickerRadioGroup.displayName = "VisualPickerRadioGroup";

// src/VisualPickerRadio.tsx
var import_base_radio_checkbox = require("@twilio-paste/base-radio-checkbox");
var import_box = require("@twilio-paste/box");
var import_radio_group = require("@twilio-paste/radio-group");
var import_screen_reader_only = require("@twilio-paste/screen-reader-only");
var import_uid_library = require("@twilio-paste/uid-library");
var import_utils = require("@twilio-paste/utils");
var React3 = __toESM(require("react"));
var VisualPickerRadio = React3.forwardRef(
  ({ id, name, element = "VISUAL_PICKER_RADIO", value, disabled, hasError, onChange, children, labelText, ...props }, ref) => {
    const [isHovering, setIsHovering] = React3.useState(false);
    const visualPickerRadioGroupContext = React3.useContext(VisualPickerRadioContext);
    const helpTextId = (0, import_uid_library.useUID)();
    const radioId = id ? id : (0, import_uid_library.useUID)();
    const labelId = (0, import_uid_library.useUID)();
    const handleChange = React3.useCallback(
      (event) => {
        if (onChange != null) {
          onChange(event);
        } else {
          visualPickerRadioGroupContext.onChange(event);
        }
      },
      [onChange, visualPickerRadioGroupContext.onChange]
    );
    const state = {
      name: visualPickerRadioGroupContext.name || name || visualPickerRadioGroupContext.name,
      disabled: disabled != null ? disabled : visualPickerRadioGroupContext.groupIsDisabled,
      hasError: hasError != null ? hasError : visualPickerRadioGroupContext.groupHasError,
      checked: visualPickerRadioGroupContext.value === value
    };
    const { groupHasError, groupIsDisabled } = visualPickerRadioGroupContext;
    const internalRef = React3.useRef();
    const mergedRef = (0, import_utils.useMergeRefs)(internalRef, ref);
    let pickerBorderColor = "colorBorderWeak";
    if (disabled || groupIsDisabled)
      pickerBorderColor = "colorBorderWeak";
    else if (!state.checked) {
      if ((hasError || groupHasError) && !isHovering)
        pickerBorderColor = "colorBorderError";
      else if ((hasError || groupHasError) && isHovering)
        pickerBorderColor = "colorBorderErrorStrong";
      else if (isHovering)
        pickerBorderColor = "colorBorderPrimary";
    } else if (state.checked) {
      if (hasError || groupHasError)
        pickerBorderColor = "colorBorderError";
      else
        pickerBorderColor = "colorBorderPrimary";
    }
    return /* @__PURE__ */ React3.createElement(
      import_box.Box,
      {
        element,
        position: "relative",
        display: "inline-flex",
        alignItems: visualPickerRadioGroupContext.orientation === "vertical" ? "center" : "flex-start",
        flexDirection: "row",
        height: "100%",
        width: "100%",
        backgroundColor: disabled || groupIsDisabled ? "colorBackgroundWeak" : void 0,
        borderStyle: "solid",
        borderColor: pickerBorderColor,
        borderWidth: "borderWidth10",
        borderRadius: "borderRadius30",
        padding: "space50",
        onMouseEnter: () => setIsHovering(true),
        onMouseLeave: () => setIsHovering(false),
        onClick: (e) => {
          internalRef.current?.click();
          internalRef.current?.focus();
          e.stopPropagation();
        },
        _hover: { cursor: disabled || groupIsDisabled ? "not-allowed" : "pointer" }
      },
      /* @__PURE__ */ React3.createElement(
        import_radio_group.HiddenRadio,
        {
          ...props,
          ...state,
          value,
          "aria-describedby": helpTextId,
          "aria-invalid": state.hasError,
          "aria-labelledby": labelId,
          onChange: (e) => {
            handleChange(e);
          },
          id: radioId,
          ref: mergedRef
        }
      ),
      /* @__PURE__ */ React3.createElement(import_base_radio_checkbox.BaseRadioCheckboxLabel, { disabled: state.disabled, htmlFor: radioId, id: labelId }, /* @__PURE__ */ React3.createElement(import_screen_reader_only.ScreenReaderOnly, null, labelText), /* @__PURE__ */ React3.createElement(
        import_base_radio_checkbox.BaseRadioCheckboxControl,
        {
          onClick: (e) => e.stopPropagation(),
          element: `${element}_CONTROL`,
          borderRadius: "borderRadiusCircle",
          disabled: state.disabled,
          type: "radio",
          applyHoverStyles: isHovering,
          _checkedAndDisabledSibling: {
            color: "colorTextWeaker"
          }
        },
        /* @__PURE__ */ React3.createElement(
          import_box.Box,
          {
            as: "span",
            element: `${element}_CONTROL_CIRCLE`,
            lineHeight: "lineHeight0",
            display: state.checked ? "block" : "none",
            color: "inherit",
            size: "sizeIcon10"
          },
          /* @__PURE__ */ React3.createElement("svg", { role: "img", "aria-hidden": true, width: "100%", height: "100%", viewBox: "0 0 16 16", fill: "none" }, /* @__PURE__ */ React3.createElement("circle", { fill: "currentColor", cx: 8, cy: 8, r: 3 }))
        )
      )),
      /* @__PURE__ */ React3.createElement(import_box.Box, { element: `${element}_CONTENT`, id: helpTextId, paddingLeft: "space40", width: "100%" }, children)
    );
  }
);
VisualPickerRadio.displayName = "VisualPickerRadio";

// src/VisualPickerCheckboxGroup.tsx
var import_inline_control_group2 = require("@twilio-paste/inline-control-group");
var React4 = __toESM(require("react"));
var VisualPickerCheckboxGroup = React4.forwardRef(
  ({
    orientation = "vertical",
    children,
    element = "VISUAL_PICKER_CHECKBOX_GROUP",
    disabled = false,
    errorText,
    name,
    i18nRequiredLabel = "(required)",
    ...props
  }, ref) => {
    const contextValue = React4.useMemo(() => {
      return {
        name,
        groupIsDisabled: disabled,
        groupHasError: errorText != null,
        orientation
      };
    }, [name, disabled, errorText, orientation]);
    return /* @__PURE__ */ React4.createElement(VisualPickerCheckboxContext.Provider, { value: contextValue }, /* @__PURE__ */ React4.createElement(
      import_inline_control_group2.InlineControlGroup,
      {
        ...props,
        ref,
        name,
        element,
        errorText,
        disabled,
        orientation,
        i18nRequiredLabel,
        fieldStyleProps: {
          marginTop: orientation === "horizontal" ? "space40" : "space30",
          marginRight: orientation === "horizontal" ? "space30" : void 0
        }
      },
      children
    ));
  }
);
VisualPickerCheckboxGroup.displayName = "VisualPickerCheckboxGroup";

// src/VisualPickerCheckbox.tsx
var import_base_radio_checkbox2 = require("@twilio-paste/base-radio-checkbox");
var import_box2 = require("@twilio-paste/box");
var import_checkbox = require("@twilio-paste/checkbox");
var import_screen_reader_only2 = require("@twilio-paste/screen-reader-only");
var import_uid_library2 = require("@twilio-paste/uid-library");
var import_utils2 = require("@twilio-paste/utils");
var React5 = __toESM(require("react"));
var VisualPickerCheckbox = React5.forwardRef(
  ({ checked, element = "VISUAL_PICKER_CHECKBOX", children, id, indeterminate, labelText, onChange, ...props }, ref) => {
    const [isHovering, setIsHovering] = React5.useState(false);
    const visualPickerCheckboxGroupContext = React5.useContext(VisualPickerCheckboxContext);
    const helpTextId = (0, import_uid_library2.useUID)();
    const labelId = (0, import_uid_library2.useUID)();
    const checkboxId = id ? id : (0, import_uid_library2.useUID)();
    const disabled = props.disabled != null ? props.disabled : visualPickerCheckboxGroupContext.groupIsDisabled;
    const name = visualPickerCheckboxGroupContext.name || props.name || visualPickerCheckboxGroupContext.name;
    const hasError = props.hasError != null ? props.hasError : visualPickerCheckboxGroupContext.groupHasError;
    const { groupHasError, groupIsDisabled } = visualPickerCheckboxGroupContext;
    const internalRef = React5.useRef();
    const mergedRef = (0, import_utils2.useMergeRefs)(internalRef, ref);
    let pickerBorderColor = "colorBorderWeak";
    if (disabled || groupIsDisabled)
      pickerBorderColor = "colorBorderWeak";
    else if (hasError || groupHasError)
      pickerBorderColor = "colorBorderError";
    else if (checked)
      pickerBorderColor = "colorBorderPrimary";
    return /* @__PURE__ */ React5.createElement(
      import_box2.Box,
      {
        element,
        position: "relative",
        display: "inline-flex",
        alignItems: visualPickerCheckboxGroupContext.orientation === "vertical" ? "center" : "flex-start",
        flexDirection: "row",
        height: "100%",
        width: "100%",
        backgroundColor: disabled || groupIsDisabled ? "colorBackgroundWeak" : void 0,
        borderStyle: "solid",
        borderColor: pickerBorderColor,
        borderWidth: "borderWidth10",
        borderRadius: "borderRadius30",
        padding: "space50",
        onMouseEnter: () => setIsHovering(true),
        onMouseLeave: () => setIsHovering(false),
        onClick: (e) => {
          internalRef.current?.click();
          internalRef.current?.focus();
          e.stopPropagation();
        },
        _hover: disabled || groupIsDisabled ? { cursor: "not-allowed" } : {
          cursor: "pointer",
          borderColor: hasError || groupHasError ? "colorBorderErrorStronger" : "colorBorderPrimaryStrong"
        }
      },
      /* @__PURE__ */ React5.createElement(
        import_checkbox.HiddenCheckbox,
        {
          ...(0, import_box2.safelySpreadBoxProps)(props),
          checked,
          disabled,
          name,
          onChange: (e) => {
            onChange(e);
          },
          "aria-describedby": helpTextId,
          "aria-checked": indeterminate ? "mixed" : checked,
          "aria-labelledby": labelId,
          "aria-invalid": hasError,
          id: checkboxId,
          ref: mergedRef
        }
      ),
      /* @__PURE__ */ React5.createElement(import_base_radio_checkbox2.BaseRadioCheckboxLabel, { disabled, htmlFor: checkboxId, id: labelId }, /* @__PURE__ */ React5.createElement(import_screen_reader_only2.ScreenReaderOnly, null, labelText), /* @__PURE__ */ React5.createElement(
        import_base_radio_checkbox2.BaseRadioCheckboxControl,
        {
          onClick: (e) => e.preventDefault(),
          borderRadius: "borderRadius20",
          element: `${element}_CONTROL`,
          disabled,
          position: "relative",
          type: "checkbox",
          applyHoverStyles: isHovering
        },
        /* @__PURE__ */ React5.createElement(import_checkbox.CheckboxIcon, { element: `${element}_ICON`, checked, indeterminate })
      )),
      /* @__PURE__ */ React5.createElement(import_box2.Box, { element: `${element}_CONTENT`, id: helpTextId, paddingLeft: "space40", width: "100%" }, children)
    );
  }
);
VisualPickerCheckbox.displayName = "VisualPickerCheckbox";
