// @twilio-paste/anchor/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Anchor: () => Anchor,
  isExternalUrl: () => isExternalUrl,
  secureExternalLink: () => secureExternalLink
});
module.exports = __toCommonJS(src_exports);

// src/Anchor.tsx
var import_box3 = require("@twilio-paste/box");
var import_LinkExternalIcon = require("@twilio-paste/icons/cjs/LinkExternalIcon");
var React3 = __toESM(require("react"));

// src/DefaultAnchor.tsx
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));
var DefaultAnchor = React.forwardRef((props, ref) => /* @__PURE__ */ React.createElement(
  import_box.Box,
  {
    ...props,
    as: "a",
    color: "colorTextLink",
    fontSize: "inherit",
    fontWeight: "inherit",
    lineHeight: "inherit",
    outline: "none",
    ref,
    textDecoration: "underline",
    _active: {
      color: "colorTextLinkStrongest",
      textDecoration: "none"
    },
    _focus: {
      boxShadow: "shadowFocus",
      color: "colorTextLink",
      textDecoration: "underline",
      borderRadius: "borderRadius20"
    },
    _hover: {
      color: "colorTextLinkStronger",
      textDecoration: "none"
    }
  },
  props.children
));
DefaultAnchor.displayName = "DefaultAnchor";

// src/InverseAnchor.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var InverseAnchor = React2.forwardRef((props, ref) => /* @__PURE__ */ React2.createElement(
  import_box2.Box,
  {
    ...props,
    as: "a",
    color: "colorTextInverse",
    fontSize: "inherit",
    fontWeight: "inherit",
    lineHeight: "inherit",
    outline: "none",
    ref,
    textDecoration: "underline",
    _active: {
      color: "colorTextInverse",
      textDecoration: "none"
    },
    _focus: {
      boxShadow: "shadowFocusInverse",
      color: "colorTextInverse",
      textDecoration: "underline",
      borderRadius: "borderRadius20"
    },
    _hover: {
      color: "colorTextInverse",
      textDecoration: "none"
    }
  },
  props.children
));
InverseAnchor.displayName = "InverseAnchor";

// src/Anchor.tsx
var AnchorVariants = {
  inverse: InverseAnchor,
  default: DefaultAnchor
};
var EXTERNAL_URL_REGEX = /^(https?:)\S*$/;
var EXTERNAL_TARGET_DEFAULT = "_blank";
var EXTERNAL_REL_DEFAULT = "noreferrer noopener";
var isExternalUrl = (url) => EXTERNAL_URL_REGEX.test(url);
var secureExternalLink = (href) => {
  if (!isExternalUrl(href))
    return void 0;
  return {
    rel: EXTERNAL_REL_DEFAULT,
    target: EXTERNAL_TARGET_DEFAULT
  };
};
var Anchor = React3.forwardRef(
  ({
    element = "ANCHOR",
    variant = "default",
    showExternal,
    display,
    height,
    minHeight,
    maxHeight,
    width,
    minWidth,
    maxWidth,
    size,
    margin,
    marginBottom,
    marginLeft,
    marginRight,
    marginTop,
    marginX = null,
    marginY = null,
    padding,
    paddingBottom,
    paddingLeft,
    paddingRight,
    paddingTop,
    paddingX = null,
    paddingY = null,
    verticalAlign,
    i18nExternalLinkLabel = "(link takes you to an external page)",
    ...props
  }, ref) => {
    const AnchorComponent = AnchorVariants[variant];
    return /* @__PURE__ */ React3.createElement(
      AnchorComponent,
      {
        href: props.href,
        ref,
        variant,
        ...secureExternalLink(props.href),
        ...(0, import_box3.safelySpreadBoxProps)(props),
        element,
        display,
        height,
        minHeight,
        maxHeight,
        width,
        minWidth,
        maxWidth,
        size,
        margin,
        marginBottom,
        marginLeft,
        marginRight,
        marginTop,
        marginX,
        marginY,
        padding,
        paddingBottom,
        paddingLeft,
        paddingRight,
        paddingTop,
        paddingX,
        paddingY,
        verticalAlign
      },
      showExternal ? /* @__PURE__ */ React3.createElement(import_box3.Box, { as: "span" }, props.children, /* @__PURE__ */ React3.createElement(import_box3.Box, { as: "span", display: "inline-block", flexShrink: 0, verticalAlign: "middle" }, /* @__PURE__ */ React3.createElement(import_LinkExternalIcon.LinkExternalIcon, { decorative: false, title: i18nExternalLinkLabel }))) : props.children
    );
  }
);
Anchor.displayName = "Anchor";
