// @twilio-paste/skeleton-loader/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  SkeletonLoader: () => SkeletonLoader
});
module.exports = __toCommonJS(src_exports);

// src/SkeletonLoader.tsx
var import_box = require("@twilio-paste/box");
var import_styling_library2 = require("@twilio-paste/styling-library");
var React = __toESM(require("react"));

// src/keyframes.tsx
var import_styling_library = require("@twilio-paste/styling-library");
var SkeletonLoaderKeyframes = import_styling_library.keyframes`
  0% {
    transform: translateX(-100%) skew(155deg);
  }
  50%,
  100% {
    transform: translateX(100%) skew(155deg);
  }
`;

// src/SkeletonLoader.tsx
var SkeletonLoaderInner = import_styling_library2.styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: linear-gradient(
    90deg,
    transparent,
    ${(0, import_styling_library2.themeGet)("backgroundColors.colorBackground")} 40%,
    ${(0, import_styling_library2.themeGet)("backgroundColors.colorBackground")} 60%,
    transparent
  );
  transform: translateX(-100%) skew(155deg);
  animation-timing-function: cubic-bezier(0.06, 0.42, 0.57, 0.89);
  animation-name: ${SkeletonLoaderKeyframes};
  animation-duration: 4.5s;
  animation-iteration-count: infinite;
`;
var SkeletonLoader = React.forwardRef(
  ({
    borderBottomLeftRadius,
    borderBottomRightRadius,
    borderRadius = "borderRadius20",
    borderTopLeftRadius,
    borderTopRightRadius,
    element = "SKELETON_LOADER",
    display,
    height = "sizeIcon20",
    maxHeight,
    maxWidth,
    minHeight,
    minWidth,
    size = null,
    width = null,
    ...props
  }, ref) => {
    return /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        "aria-busy": "true",
        backgroundColor: "colorBackgroundStrong",
        borderBottomLeftRadius,
        borderBottomRightRadius,
        borderRadius,
        borderTopLeftRadius,
        borderTopRightRadius,
        display,
        element,
        height,
        maxHeight,
        maxWidth,
        minHeight,
        minWidth,
        overflow: "hidden",
        pointerEvents: "none",
        position: "relative",
        userSelect: "none",
        size,
        width,
        ref
      },
      /* @__PURE__ */ React.createElement(SkeletonLoaderInner, null)
    );
  }
);
SkeletonLoader.displayName = "SkeletonLoader";
