// @twilio-paste/topbar/dist/index.debug.es.js
// src/Topbar.tsx
import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
import * as React from "react";
var Topbar = React.forwardRef(
  ({ element = "TOPBAR", id, children, ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(
      Box,
      {
        ...safelySpreadBoxProps(props),
        id,
        ref,
        element,
        width: "100%",
        minHeight: "sizeTopbar",
        backgroundColor: "colorBackgroundWeak",
        borderBottomWidth: "borderWidth10",
        borderBottomStyle: "solid",
        borderBottomColor: "colorBorderWeaker",
        paddingX: "space70",
        position: "sticky",
        top: "0",
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
        columnGap: "space70",
        zIndex: "zIndex40"
      },
      children
    );
  }
);
Topbar.displayName = "Topbar";

// src/TopbarActions.tsx
import { Box as Box2, safelySpreadBoxProps as safelySpreadBoxProps2 } from "@twilio-paste/box";
import * as React2 from "react";
var TopbarActions = React2.forwardRef(
  ({ children, element = "TOPBAR_ACTIONS", justify = "end", display = "flex", ...props }, ref) => {
    const justifyContentValue = {
      start: "flex-start",
      end: "flex-end",
      center: "center"
    };
    return /* @__PURE__ */ React2.createElement(
      Box2,
      {
        ...safelySpreadBoxProps2(props),
        display,
        justifyContent: justifyContentValue[justify],
        flexShrink: justify === "start" ? null : 0,
        flexWrap: "wrap",
        flexGrow: 1,
        columnGap: "space50",
        element,
        alignItems: "center",
        ref
      },
      children
    );
  }
);
TopbarActions.displayName = "TopbarActions";
export {
  Topbar,
  TopbarActions
};
