// @twilio-paste/topbar/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Topbar: () => Topbar,
  TopbarActions: () => TopbarActions
});
module.exports = __toCommonJS(src_exports);

// src/Topbar.tsx
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));
var Topbar = React.forwardRef(
  ({ element = "TOPBAR", id, children, ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        id,
        ref,
        element,
        width: "100%",
        minHeight: "sizeTopbar",
        backgroundColor: "colorBackgroundWeak",
        borderBottomWidth: "borderWidth10",
        borderBottomStyle: "solid",
        borderBottomColor: "colorBorderWeaker",
        paddingX: "space70",
        position: "sticky",
        top: "0",
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
        columnGap: "space70",
        zIndex: "zIndex40"
      },
      children
    );
  }
);
Topbar.displayName = "Topbar";

// src/TopbarActions.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var TopbarActions = React2.forwardRef(
  ({ children, element = "TOPBAR_ACTIONS", justify = "end", display = "flex", ...props }, ref) => {
    const justifyContentValue = {
      start: "flex-start",
      end: "flex-end",
      center: "center"
    };
    return /* @__PURE__ */ React2.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        display,
        justifyContent: justifyContentValue[justify],
        flexShrink: justify === "start" ? null : 0,
        flexWrap: "wrap",
        flexGrow: 1,
        columnGap: "space50",
        element,
        alignItems: "center",
        ref
      },
      children
    );
  }
);
TopbarActions.displayName = "TopbarActions";
