// @twilio-paste/progress-steps/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  ProgressStepComplete: () => ProgressStepComplete,
  ProgressStepContent: () => ProgressStepContent,
  ProgressStepCurrent: () => ProgressStepCurrent,
  ProgressStepError: () => ProgressStepError,
  ProgressStepIncomplete: () => ProgressStepIncomplete,
  ProgressStepSeparator: () => ProgressStepSeparator,
  ProgressSteps: () => ProgressSteps
});
module.exports = __toCommonJS(src_exports);

// src/ProgressSteps.tsx
var import_box = require("@twilio-paste/box");
var import_styling_library = require("@twilio-paste/styling-library");
var React2 = __toESM(require("react"));

// src/ProgressStepsContext.tsx
var React = __toESM(require("react"));
var ProgressStepsContext = React.createContext({});

// src/ProgressSteps.tsx
var ItemSeparatorStyles = {
  vertical: {
    flexDirection: "column",
    alignItems: "flex-start",
    "[role='listitem']>*>div:first-child::after": {
      content: "''",
      borderLeftWidth: "borderWidth20",
      borderLeftStyle: "solid",
      borderLeftColor: "colorBorderWeaker",
      borderRadius: "borderRadius0",
      minHeight: "sizeBase80",
      flexGrow: "1",
      marginY: "space30"
    },
    "[role='listitem']:last-child>*>div:first-child::after": {
      content: "none"
    }
  },
  horizontal: {
    flexWrap: "nowrap",
    "[role='listitem']": {
      display: "flex",
      alignItems: "flex-start",
      flexGrow: "1",
      "&::after": {
        content: "''",
        borderBottomWidth: "borderWidth20",
        borderBottomStyle: "solid",
        borderBottomColor: "colorBorderWeaker",
        borderRadius: "borderRadius20",
        minWidth: "sizeBase40",
        flexGrow: 1,
        marginTop: "10px"
      },
      "&>*:first-child": {
        marginX: "space30"
      }
    },
    "[role='listitem']:first-child": {
      "&>*:first-child": {
        marginLeft: "space0"
      }
    },
    "[role='listitem']:last-child": {
      flexGrow: "0",
      "&::after": {
        content: "none"
      },
      "&>*:last-child": {
        marginRight: "space0"
      }
    }
  }
};
var ProgressSteps = React2.forwardRef(
  ({ element = "PROGRESS_STEPS", orientation = "horizontal", ...props }, ref) => {
    const ContainerStyled = React2.useMemo(
      () => import_styling_library.styled.div(
        (0, import_styling_library.css)({
          display: "flex",
          ...ItemSeparatorStyles[orientation]
        })
      ),
      [orientation]
    );
    return /* @__PURE__ */ React2.createElement(ProgressStepsContext.Provider, { value: { orientation } }, /* @__PURE__ */ React2.createElement(import_box.Box, { ...(0, import_box.safelySpreadBoxProps)(props), as: ContainerStyled, ref, element, role: "list" }, props.children));
  }
);
ProgressSteps.displayName = "ProgressSteps";

// src/ProgressStepSeparator.tsx
var React3 = __toESM(require("react"));
var ProgressStepSeparator = React3.forwardRef(() => {
  return null;
});
ProgressStepSeparator.displayName = "ProgressStepSeparator";

// src/ProgressStepContent.tsx
var import_box2 = require("@twilio-paste/box");
var import_react = __toESM(require("react"));
var ProgressStepContent = import_react.default.forwardRef(
  ({ element = "PROGRESS_STEP_CONTENT", paddingY = "space40", ...props }, ref) => {
    return /* @__PURE__ */ import_react.default.createElement(import_box2.Box, { element, color: "colorText", ref, paddingY }, props.children);
  }
);
ProgressStepContent.displayName = "ProgressStepContent";

// src/ProgressStepComplete.tsx
var import_box3 = require("@twilio-paste/box");
var import_SuccessIcon = require("@twilio-paste/icons/cjs/SuccessIcon");
var React5 = __toESM(require("react"));
var ProgressStepComplete = React5.forwardRef(
  ({ element = "PROGRESS_STEP_COMPLETE", as = "div", children, i18nCompleteLabel = "Completed", ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement("div", { role: "listitem" }, /* @__PURE__ */ React5.createElement(
      import_box3.Box,
      {
        ...(0, import_box3.safelySpreadBoxProps)(props),
        element,
        as,
        type: as === "button" ? "button" : void 0,
        ref,
        display: "flex",
        columnGap: "space30",
        border: "none",
        background: "none",
        padding: "space0",
        margin: "space0",
        fontFamily: "fontFamilyText",
        fontSize: "fontSize20",
        lineHeight: "lineHeight20",
        fontWeight: as === "button" ? "fontWeightSemibold" : "fontWeightMedium",
        outline: "none",
        borderRadius: "borderRadius20",
        cursor: as !== "div" ? "pointer" : "default",
        color: "colorTextWeak",
        _hover: as !== "div" ? { textDecoration: "none", color: "colorTextPrimary" } : void 0,
        _focus: as !== "div" ? { boxShadow: "shadowFocus" } : void 0
      },
      /* @__PURE__ */ React5.createElement(import_box3.Box, { display: "flex", flexDirection: "column", alignItems: "center" }, /* @__PURE__ */ React5.createElement(import_SuccessIcon.SuccessIcon, { decorative: false, title: i18nCompleteLabel })),
      /* @__PURE__ */ React5.createElement(import_box3.Box, { textAlign: "left" }, children)
    ));
  }
);
ProgressStepComplete.displayName = "ProgressStepComplete";

// src/ProgressStepIncomplete.tsx
var import_box4 = require("@twilio-paste/box");
var React7 = __toESM(require("react"));

// src/icons/ProgressIncompleteIcon.tsx
var import_IconWrapper = require("@twilio-paste/icons/cjs/helpers/IconWrapper");
var import_uid_library = require("@twilio-paste/uid-library");
var React6 = __toESM(require("react"));
var ProgressIncompleteIcon = React6.forwardRef(
  ({ as, display, element = "ICON", size, color, title, decorative }, ref) => {
    const titleId = `ProgressIncompleteIcon-${(0, import_uid_library.useUID)()}`;
    if (!decorative && title == null) {
      throw new Error("[ProgressIncompleteIcon]: Missing a title for non-decorative icon.");
    }
    return /* @__PURE__ */ React6.createElement(import_IconWrapper.IconWrapper, { as, display, element, size, color, ref }, /* @__PURE__ */ React6.createElement(
      "svg",
      {
        role: "img",
        "aria-hidden": decorative,
        fill: "none",
        height: "100%",
        viewBox: "0 0 20 20",
        width: "100%",
        xmlns: "http://www.w3.org/2000/svg",
        "aria-labelledby": titleId
      },
      title ? /* @__PURE__ */ React6.createElement("title", { id: titleId }, title) : null,
      /* @__PURE__ */ React6.createElement(
        "path",
        {
          fill: "currentColor",
          clipRule: "evenodd",
          d: "M9.993 2h.014a1.23 1.23 0 01-.002 2.462h-.01A1.23 1.23 0 019.993 2zm2.321 1.518a1.23 1.23 0 011.681-.45l.013.007a1.23 1.23 0 01-1.235 2.13l-.008-.005a1.23 1.23 0 01-.45-1.682zm-4.628 0a1.23 1.23 0 01-.45 1.682l-.009.004a1.23 1.23 0 01-1.235-2.13l.012-.006a1.23 1.23 0 011.682.45zm7.557 2.027a1.23 1.23 0 011.682.447l.007.012A1.23 1.23 0 1114.8 7.235l-.004-.008a1.23 1.23 0 01.447-1.682zm-10.486 0a1.23 1.23 0 01.447 1.682l-.004.008a1.23 1.23 0 01-2.132-1.23l.007-.013a1.23 1.23 0 011.682-.447zm12.011 3.218c.68 0 1.231.55 1.232 1.23v.014a1.23 1.23 0 01-2.461-.002v-.01c-.001-.68.55-1.231 1.23-1.232zm-13.536 0c.68 0 1.23.552 1.23 1.232v.01A1.23 1.23 0 012 10.007v-.014c0-.68.552-1.23 1.232-1.23zm13.25 3.551c.588.34.79 1.093.45 1.681l-.007.013a1.23 1.23 0 01-2.13-1.235l.005-.008a1.23 1.23 0 011.682-.45zm-12.964 0a1.23 1.23 0 011.682.45l.004.009a1.23 1.23 0 01-2.13 1.235l-.006-.012a1.23 1.23 0 01.45-1.682zm2.027 2.93a1.23 1.23 0 011.682-.448l.008.004a1.23 1.23 0 01-1.23 2.132l-.013-.007a1.23 1.23 0 01-.447-1.682zm8.91 0a1.23 1.23 0 01-.447 1.681l-.012.007a1.23 1.23 0 11-1.231-2.132l.008-.004a1.23 1.23 0 011.682.447zm-5.692 1.524c0-.68.552-1.23 1.232-1.23h.01A1.23 1.23 0 1110.007 18h-.014a1.23 1.23 0 01-1.23-1.232z",
          fillRule: "evenodd"
        }
      )
    ));
  }
);
ProgressIncompleteIcon.displayName = "ProgressIncompleteIcon";

// src/ProgressStepIncomplete.tsx
var ProgressStepIncomplete = React7.forwardRef(
  ({
    element = "PROGRESS_STEP_INCOMPLETE",
    as = "div",
    disabled = false,
    children,
    i18nIncompleteLabel = "Incomplete",
    ...props
  }, ref) => {
    return /* @__PURE__ */ React7.createElement("div", { role: "listitem" }, /* @__PURE__ */ React7.createElement(
      import_box4.Box,
      {
        ...(0, import_box4.safelySpreadBoxProps)(props),
        element,
        as,
        type: as === "button" ? "button" : void 0,
        disabled,
        ref,
        display: "flex",
        columnGap: "space30",
        border: "none",
        background: "none",
        padding: "space0",
        margin: "space0",
        fontFamily: "fontFamilyText",
        fontSize: "fontSize20",
        lineHeight: "lineHeight20",
        fontWeight: as === "button" ? "fontWeightSemibold" : "fontWeightMedium",
        outline: "none",
        borderRadius: "borderRadius20",
        cursor: as !== "div" ? "pointer" : "default",
        color: disabled ? "colorTextWeaker" : "colorTextWeak",
        _hover: as !== "div" ? { textDecoration: "none", color: "colorTextPrimary" } : void 0,
        _focus: as !== "div" ? { boxShadow: "shadowFocus" } : void 0,
        _disabled: { textDecoration: "none", color: "colorTextWeaker", cursor: "not-allowed" }
      },
      /* @__PURE__ */ React7.createElement(import_box4.Box, { display: "flex", flexDirection: "column", alignItems: "center" }, /* @__PURE__ */ React7.createElement(ProgressIncompleteIcon, { decorative: false, title: i18nIncompleteLabel })),
      /* @__PURE__ */ React7.createElement(import_box4.Box, { textAlign: "left" }, children)
    ));
  }
);
ProgressStepIncomplete.displayName = "ProgressStepIncomplete";

// src/ProgressStepCurrent.tsx
var import_box5 = require("@twilio-paste/box");
var React9 = __toESM(require("react"));

// src/icons/ProgressCurrentIcon.tsx
var import_IconWrapper2 = require("@twilio-paste/icons/cjs/helpers/IconWrapper");
var import_uid_library2 = require("@twilio-paste/uid-library");
var React8 = __toESM(require("react"));
var ProgressCurrentIcon = React8.forwardRef(
  ({ as, display, element = "ICON", size, color, title, decorative }, ref) => {
    const titleId = `ProgressCurrentIcon-${(0, import_uid_library2.useUID)()}`;
    if (!decorative && title == null) {
      throw new Error("[ProgressCurrentIcon]: Missing a title for non-decorative icon.");
    }
    return /* @__PURE__ */ React8.createElement(import_IconWrapper2.IconWrapper, { as, display, element, size, color, ref }, /* @__PURE__ */ React8.createElement(
      "svg",
      {
        role: "img",
        "aria-hidden": decorative,
        fill: "none",
        height: "100%",
        viewBox: "0 0 20 20",
        width: "100%",
        xmlns: "http://www.w3.org/2000/svg",
        "aria-labelledby": titleId
      },
      title ? /* @__PURE__ */ React8.createElement("title", { id: titleId }, title) : null,
      /* @__PURE__ */ React8.createElement(
        "path",
        {
          fill: "currentColor",
          clipRule: "evenodd",
          d: "M10 16a6 6 0 100-12 6 6 0 000 12zm0 2a8 8 0 100-16 8 8 0 000 16z",
          fillRule: "evenodd"
        }
      )
    ));
  }
);
ProgressCurrentIcon.displayName = "ProgressCurrentIcon";

// src/ProgressStepCurrent.tsx
var ProgressStepCurrent = React9.forwardRef(
  ({ element = "PROGRESS_STEP_CURRENT", as = "div", children, i18nCurrentLabel = "Current", ...props }, ref) => {
    return /* @__PURE__ */ React9.createElement("div", { role: "listitem" }, /* @__PURE__ */ React9.createElement(
      import_box5.Box,
      {
        ...(0, import_box5.safelySpreadBoxProps)(props),
        element,
        as,
        type: as === "button" ? "button" : void 0,
        ref,
        "aria-current": "step",
        display: "flex",
        columnGap: "space30",
        border: "none",
        background: "none",
        padding: "space0",
        margin: "space0",
        fontFamily: "fontFamilyText",
        fontSize: "fontSize20",
        lineHeight: "lineHeight20",
        fontWeight: as === "button" ? "fontWeightSemibold" : "fontWeightMedium",
        outline: "none",
        borderRadius: "borderRadius20",
        cursor: as !== "div" ? "pointer" : "default",
        color: "colorTextPrimary",
        _hover: as !== "div" ? { textDecoration: "none", color: "colorTextPrimaryStrongest" } : void 0,
        _focus: as !== "div" ? { boxShadow: "shadowFocus" } : void 0
      },
      /* @__PURE__ */ React9.createElement(import_box5.Box, { display: "flex", flexDirection: "column", alignItems: "center" }, /* @__PURE__ */ React9.createElement(ProgressCurrentIcon, { decorative: false, title: i18nCurrentLabel })),
      /* @__PURE__ */ React9.createElement(import_box5.Box, { textAlign: "left" }, children)
    ));
  }
);
ProgressStepCurrent.displayName = "ProgressStepCurrent";

// src/ProgressStepError.tsx
var import_box6 = require("@twilio-paste/box");
var React11 = __toESM(require("react"));

// src/icons/ProgressErrorIcon.tsx
var import_IconWrapper3 = require("@twilio-paste/icons/cjs/helpers/IconWrapper");
var import_uid_library3 = require("@twilio-paste/uid-library");
var React10 = __toESM(require("react"));
var ProgressErrorIcon = React10.forwardRef(
  ({ as, display, element = "ICON", size, color, title, decorative }, ref) => {
    const titleId = `ProgressErrorIcon-${(0, import_uid_library3.useUID)()}`;
    if (!decorative && title == null) {
      throw new Error("[ProgressErrorIcon]: Missing a title for non-decorative icon.");
    }
    return /* @__PURE__ */ React10.createElement(import_IconWrapper3.IconWrapper, { as, display, element, size, color, ref }, /* @__PURE__ */ React10.createElement(
      "svg",
      {
        role: "img",
        "aria-hidden": decorative,
        fill: "none",
        height: "100%",
        viewBox: "0 0 20 20",
        width: "100%",
        xmlns: "http://www.w3.org/2000/svg",
        "aria-labelledby": titleId
      },
      title ? /* @__PURE__ */ React10.createElement("title", { id: titleId }, title) : null,
      /* @__PURE__ */ React10.createElement(
        "path",
        {
          fill: "currentColor",
          clipRule: "evenodd",
          d: "M10 18a8 8 0 100-16 8 8 0 000 16zM9.985 5h.03c.916.011 1.609.889 1.466 1.866l-.641 3.357c-.061.444-.418.777-.835.777h-.01c-.418 0-.774-.333-.835-.777l-.642-3.357C8.376 5.889 9.078 5 9.985 5zM10 15.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z",
          fillRule: "evenodd"
        }
      )
    ));
  }
);
ProgressErrorIcon.displayName = "ProgressErrorIcon";

// src/ProgressStepError.tsx
var ProgressStepError = React11.forwardRef(
  ({ element = "PROGRESS_STEP_ERROR", as = "div", children, i18nErrorLabel = "Error", ...props }, ref) => {
    return /* @__PURE__ */ React11.createElement("div", { role: "listitem" }, /* @__PURE__ */ React11.createElement(
      import_box6.Box,
      {
        ...(0, import_box6.safelySpreadBoxProps)(props),
        element,
        as,
        type: as === "button" ? "button" : void 0,
        ref,
        display: "flex",
        columnGap: "space30",
        border: "none",
        background: "none",
        padding: "space0",
        margin: "space0",
        fontFamily: "fontFamilyText",
        fontSize: "fontSize20",
        lineHeight: "lineHeight20",
        fontWeight: as === "button" ? "fontWeightSemibold" : "fontWeightMedium",
        outline: "none",
        borderRadius: "borderRadius20",
        cursor: as !== "div" ? "pointer" : "default",
        color: "colorTextError",
        _hover: as !== "div" ? { textDecoration: "none", color: "colorTextErrorStrongest" } : void 0,
        _focus: as !== "div" ? { boxShadow: "shadowFocus" } : void 0
      },
      /* @__PURE__ */ React11.createElement(import_box6.Box, { display: "flex", flexDirection: "column", alignItems: "center" }, /* @__PURE__ */ React11.createElement(ProgressErrorIcon, { decorative: false, title: i18nErrorLabel })),
      /* @__PURE__ */ React11.createElement(import_box6.Box, { textAlign: "left" }, children)
    ));
  }
);
ProgressStepError.displayName = "ProgressStepError";
