// @twilio-paste/account-switcher/dist/index.debug.es.js
// src/AccountSwitcherBadge.tsx
import { MenuBadge } from "@twilio-paste/menu";
import * as React from "react";
var AccountSwitcherBadge = React.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_BADGE", ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(MenuBadge, { element, ...props, variant: "default", ref }, children);
  }
);
AccountSwitcherBadge.displayName = "AccountSwitcherBadge";

// src/AccountSwitcher.tsx
import { Menu } from "@twilio-paste/menu";
import * as React2 from "react";
var AccountSwitcher = React2.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER", ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(Menu, { element, ...props, ref }, children);
  }
);
AccountSwitcher.displayName = "AccountSwitcher";

// src/AccountSwitcherItem.tsx
import { MenuItem } from "@twilio-paste/menu";
import * as React3 from "react";
var AccountSwitcherItem = React3.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_ITEM", ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(MenuItem, { element, ...props, ref }, children);
  }
);
AccountSwitcherItem.displayName = "AccountSwitcherItem";

// src/AccountSwitcherItemRadio.tsx
import { MenuItemRadio } from "@twilio-paste/menu";
import * as React4 from "react";
var AccountSwitcherItemRadio = React4.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_ITEM_RADIO", ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(MenuItemRadio, { element, ...props, ref }, children);
  }
);
AccountSwitcherItemRadio.displayName = "AccountSwitcherItemRadio";

// src/AccountSwitcherGroup.tsx
import { MenuGroup } from "@twilio-paste/menu";
import * as React5 from "react";
var AccountSwitcherGroup = React5.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_GROUP", label, ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement(MenuGroup, { label, element, ...props, ref }, children);
  }
);
AccountSwitcherGroup.displayName = "AccountSwitcherGroup";

// src/AccountSwitcherSeparator.tsx
import { MenuSeparator } from "@twilio-paste/menu";
import * as React6 from "react";
var AccountSwitcherSeparator = React6.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_SEPARATOR", ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(MenuSeparator, { element, ...props, ref }, children);
  }
);
AccountSwitcherSeparator.displayName = "AccountSwitcherSeparator";

// src/useAccountSwitcher.ts
import { useMenuState } from "@twilio-paste/menu";
export {
  AccountSwitcher,
  AccountSwitcherBadge,
  AccountSwitcherGroup,
  AccountSwitcherItem,
  AccountSwitcherItemRadio,
  AccountSwitcherSeparator,
  useMenuState as useAccountSwitcherState
};
