// @twilio-paste/account-switcher/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  AccountSwitcher: () => AccountSwitcher,
  AccountSwitcherBadge: () => AccountSwitcherBadge,
  AccountSwitcherGroup: () => AccountSwitcherGroup,
  AccountSwitcherItem: () => AccountSwitcherItem,
  AccountSwitcherItemRadio: () => AccountSwitcherItemRadio,
  AccountSwitcherSeparator: () => AccountSwitcherSeparator,
  useAccountSwitcherState: () => import_menu7.useMenuState
});
module.exports = __toCommonJS(src_exports);

// src/AccountSwitcherBadge.tsx
var import_menu = require("@twilio-paste/menu");
var React = __toESM(require("react"));
var AccountSwitcherBadge = React.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_BADGE", ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(import_menu.MenuBadge, { element, ...props, variant: "default", ref }, children);
  }
);
AccountSwitcherBadge.displayName = "AccountSwitcherBadge";

// src/AccountSwitcher.tsx
var import_menu2 = require("@twilio-paste/menu");
var React2 = __toESM(require("react"));
var AccountSwitcher = React2.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER", ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(import_menu2.Menu, { element, ...props, ref }, children);
  }
);
AccountSwitcher.displayName = "AccountSwitcher";

// src/AccountSwitcherItem.tsx
var import_menu3 = require("@twilio-paste/menu");
var React3 = __toESM(require("react"));
var AccountSwitcherItem = React3.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_ITEM", ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(import_menu3.MenuItem, { element, ...props, ref }, children);
  }
);
AccountSwitcherItem.displayName = "AccountSwitcherItem";

// src/AccountSwitcherItemRadio.tsx
var import_menu4 = require("@twilio-paste/menu");
var React4 = __toESM(require("react"));
var AccountSwitcherItemRadio = React4.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_ITEM_RADIO", ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(import_menu4.MenuItemRadio, { element, ...props, ref }, children);
  }
);
AccountSwitcherItemRadio.displayName = "AccountSwitcherItemRadio";

// src/AccountSwitcherGroup.tsx
var import_menu5 = require("@twilio-paste/menu");
var React5 = __toESM(require("react"));
var AccountSwitcherGroup = React5.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_GROUP", label, ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement(import_menu5.MenuGroup, { label, element, ...props, ref }, children);
  }
);
AccountSwitcherGroup.displayName = "AccountSwitcherGroup";

// src/AccountSwitcherSeparator.tsx
var import_menu6 = require("@twilio-paste/menu");
var React6 = __toESM(require("react"));
var AccountSwitcherSeparator = React6.forwardRef(
  ({ children, element = "ACCOUNT_SWITCHER_SEPARATOR", ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(import_menu6.MenuSeparator, { element, ...props, ref }, children);
  }
);
AccountSwitcherSeparator.displayName = "AccountSwitcherSeparator";

// src/useAccountSwitcher.ts
var import_menu7 = require("@twilio-paste/menu");
