// @twilio-paste/slider/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Slider: () => Slider
});
module.exports = __toCommonJS(src_exports);

// src/Slider.tsx
var import_box3 = require("@twilio-paste/box");
var import_react_spectrum_library = require("@twilio-paste/react-spectrum-library");
var import_screen_reader_only = require("@twilio-paste/screen-reader-only");
var import_utils = require("@twilio-paste/utils");
var React3 = __toESM(require("react"));

// src/SliderThumb.tsx
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));

// src/helpers.ts
var getColorToken = (states) => {
  if (states.disabled) {
    return "colorBackgroundStrong";
  } else if (states.error) {
    if (states.hovered) {
      return "colorBackgroundErrorStrong";
    }
    return "colorBackgroundError";
  } else if (states.dragging) {
    return "colorBackgroundPrimary";
  } else if (states.focused) {
    return "colorBackgroundPrimaryStronger";
  } else if (states.hovered) {
    return "colorBackgroundPrimary";
  }
  return "colorBackgroundPrimaryStronger";
};

// src/SliderThumb.tsx
var SliderThumb = ({ element = "SLIDER_THUMB", left = "0%", ...props }) => {
  return /* @__PURE__ */ React.createElement(
    import_box.Box,
    {
      ...props,
      backgroundColor: getColorToken(props),
      boxShadow: props.dragging || props.focused ? "shadowFocus" : "none",
      borderRadius: "borderRadiusCircle",
      transition: "background-color 150ms ease, box-shadow 150ms ease",
      width: "20px",
      height: "20px",
      marginTop: "space10",
      cursor: props.disabled ? "not-allowed" : "pointer",
      left,
      element
    }
  );
};
SliderThumb.displayName = "SliderThumb";

// src/SliderTrack.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var SliderTrack = ({
  element = "SLIDER_TRACK",
  fillPercent = "0%",
  children,
  ...props
}) => {
  return /* @__PURE__ */ React2.createElement(import_box2.Box, { ...props, position: "relative", height: "4px", width: "100%", element }, /* @__PURE__ */ React2.createElement(
    import_box2.Box,
    {
      element: `${element}_BACKGROUND`,
      borderRadius: "borderRadius20",
      position: "absolute",
      top: 0,
      left: 0,
      backgroundColor: "colorBackgroundStrong",
      height: "100%",
      width: "100%"
    }
  ), /* @__PURE__ */ React2.createElement(
    import_box2.Box,
    {
      element: `${element}_FILL`,
      borderRadius: "borderRadius20",
      position: "absolute",
      top: 0,
      left: 0,
      backgroundColor: getColorToken(props),
      height: "100%",
      width: fillPercent,
      transition: "background-color 150ms ease"
    }
  ), children);
};
SliderTrack.displayName = "SliderTrack";

// src/Slider.tsx
var DefaultNumberFormatter = new Intl.NumberFormat("en-US");
var Slider = React3.forwardRef((props, ref) => {
  const inputRef = React3.useRef(null);
  const trackRef = React3.useRef(null);
  const mergedInputRef = (0, import_utils.useMergeRefs)(inputRef, ref);
  const [hovered, setHovered] = React3.useState(false);
  const [focused, setFocused] = React3.useState(false);
  const {
    element = "SLIDER",
    minValue = 0,
    maxValue = 100,
    hideRangeLabels = false,
    disabled: isDisabled = false,
    id,
    i18nMaxRangeLabel = "Maximum value:",
    i18nMinRangeLabel = "Minimum value:"
  } = props;
  const remappedProps = {
    ...props,
    id,
    isDisabled,
    "aria-labelledby": id,
    numberFormatter: props.numberFormatter || DefaultNumberFormatter
  };
  const state = (0, import_react_spectrum_library.useSliderState)(remappedProps);
  const { trackProps } = (0, import_react_spectrum_library.useSlider)(remappedProps, state, trackRef);
  const { thumbProps, inputProps, isDragging } = (0, import_react_spectrum_library.useSliderThumb)(
    {
      index: 0,
      trackRef,
      inputRef
    },
    state
  );
  const uiStateProps = React3.useMemo(() => {
    return {
      disabled: isDisabled,
      error: props.hasError,
      hovered,
      focused,
      dragging: isDragging
    };
  }, [isDisabled, props.hasError, hovered, focused, isDragging]);
  const onClickHandler = React3.useCallback(() => {
    inputRef?.current?.focus();
  }, [inputRef]);
  return /* @__PURE__ */ React3.createElement(import_box3.Box, { element }, /* @__PURE__ */ React3.createElement(
    import_box3.Box,
    {
      ...trackProps,
      element: `${element}_TRACK_CONTAINER`,
      ref: trackRef,
      onMouseEnter: () => setHovered(true),
      onMouseLeave: () => setHovered(false),
      display: "flex",
      alignItems: "center",
      height: "20px",
      width: "100%",
      cursor: isDisabled ? "not-allowed" : "pointer",
      onClick: onClickHandler
    },
    /* @__PURE__ */ React3.createElement(
      SliderTrack,
      {
        ...uiStateProps,
        fillPercent: thumbProps?.style?.left,
        element: `${element}_TRACK`,
        onClick: onClickHandler
      },
      /* @__PURE__ */ React3.createElement(SliderThumb, { ...thumbProps, ...uiStateProps, element: `${element}_THUMB`, onClick: onClickHandler }, /* @__PURE__ */ React3.createElement(import_screen_reader_only.ScreenReaderOnly, null, /* @__PURE__ */ React3.createElement(
        "input",
        {
          ref: mergedInputRef,
          ...inputProps,
          "aria-labelledby": props["aria-labelledby"],
          "aria-describedby": props["aria-describedby"],
          id: inputProps.id?.replace("-0", ""),
          onFocus: () => setFocused(true),
          onBlur: () => setFocused(false)
        }
      )))
    )
  ), !hideRangeLabels && /* @__PURE__ */ React3.createElement(
    import_box3.Box,
    {
      element: `${element}_RANGE_LABELS`,
      display: "flex",
      justifyContent: "space-between",
      columnGap: "space20",
      fontSize: "fontSize30",
      lineHeight: "lineHeight30",
      fontWeight: "fontWeightNormal",
      color: "colorTextWeak",
      overflowWrap: "anywhere"
    },
    /* @__PURE__ */ React3.createElement(import_box3.Box, { element: `${element}_RANGE_LABELS_MIN` }, /* @__PURE__ */ React3.createElement(import_screen_reader_only.ScreenReaderOnly, null, i18nMinRangeLabel), remappedProps.numberFormatter.format(minValue)),
    /* @__PURE__ */ React3.createElement(import_box3.Box, { element: `${element}_RANGE_LABELS_MAX`, textAlign: "end" }, /* @__PURE__ */ React3.createElement(import_screen_reader_only.ScreenReaderOnly, null, i18nMaxRangeLabel), remappedProps.numberFormatter.format(maxValue))
  ));
});
Slider.displayName = "Slider";
