// @twilio-paste/status/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  StatusBadge: () => StatusBadge,
  StatusMenu: () => StatusMenu,
  StatusMenuBadge: () => StatusMenuBadge,
  StatusMenuItem: () => StatusMenuItem,
  StatusMenuItemCheckbox: () => StatusMenuItemCheckbox,
  StatusMenuItemChild: () => StatusMenuItemChild,
  StatusMenuItemRadio: () => StatusMenuItemRadio,
  StatusObject: () => StatusObject,
  useStatusMenuState: () => import_menu6.useMenuState
});
module.exports = __toCommonJS(src_exports);

// src/StatusBadge.tsx
var import_badge = require("@twilio-paste/badge");
var import_box = require("@twilio-paste/box");
var React2 = __toESM(require("react"));

// src/constants.tsx
var import_ConnectivityAvailableIcon = require("@twilio-paste/icons/cjs/ConnectivityAvailableIcon");
var import_ConnectivityBusyIcon = require("@twilio-paste/icons/cjs/ConnectivityBusyIcon");
var import_ConnectivityNeutralIcon = require("@twilio-paste/icons/cjs/ConnectivityNeutralIcon");
var import_ConnectivityOfflineIcon = require("@twilio-paste/icons/cjs/ConnectivityOfflineIcon");
var import_ConnectivityUnavailableIcon = require("@twilio-paste/icons/cjs/ConnectivityUnavailableIcon");
var import_ProcessDisabledIcon = require("@twilio-paste/icons/cjs/ProcessDisabledIcon");
var import_ProcessDraftIcon = require("@twilio-paste/icons/cjs/ProcessDraftIcon");
var import_ProcessErrorIcon = require("@twilio-paste/icons/cjs/ProcessErrorIcon");
var import_ProcessInProgressIcon = require("@twilio-paste/icons/cjs/ProcessInProgressIcon");
var import_ProcessNeutralIcon = require("@twilio-paste/icons/cjs/ProcessNeutralIcon");
var import_ProcessSuccessIcon = require("@twilio-paste/icons/cjs/ProcessSuccessIcon");
var import_ProcessWarningIcon = require("@twilio-paste/icons/cjs/ProcessWarningIcon");
var React = __toESM(require("react"));
var StatusObject = {
  ProcessError: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ProcessErrorIcon.ProcessErrorIcon, { color: "colorTextIconError", decorative: true, size: "sizeIcon10" }),
    color: "colorTextError"
  },
  ProcessWarning: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ProcessWarningIcon.ProcessWarningIcon, { color: "colorTextIconWarning", decorative: true, size: "sizeIcon10" }),
    color: "colorTextWarning"
  },
  ProcessSuccess: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ProcessSuccessIcon.ProcessSuccessIcon, { color: "colorTextIconSuccess", decorative: true, size: "sizeIcon10" }),
    color: "colorTextSuccess"
  },
  ProcessNeutral: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ProcessNeutralIcon.ProcessNeutralIcon, { color: "colorTextIconNeutral", decorative: true, size: "sizeIcon10" }),
    color: "colorTextNeutral"
  },
  ProcessInProgress: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ProcessInProgressIcon.ProcessInProgressIcon, { color: "colorTextIconNeutral", decorative: true, size: "sizeIcon10" }),
    color: "colorTextNeutral"
  },
  ProcessDisabled: {
    badgeVariant: "decorative10",
    icon: /* @__PURE__ */ React.createElement(import_ProcessDisabledIcon.ProcessDisabledIcon, { color: "colorTextIcon", decorative: true, size: "sizeIcon10" }),
    color: "colorTextWeak"
  },
  ProcessDraft: {
    badgeVariant: "decorative10",
    icon: /* @__PURE__ */ React.createElement(import_ProcessDraftIcon.ProcessDraftIcon, { color: "colorTextIcon", decorative: true, size: "sizeIcon10" }),
    color: "colorTextWeak"
  },
  ConnectivityAvailable: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ConnectivityAvailableIcon.ConnectivityAvailableIcon, { color: "colorTextIconAvailable", decorative: true, size: "sizeIcon10" }),
    color: "colorTextSuccess"
  },
  ConnectivityBusy: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ConnectivityBusyIcon.ConnectivityBusyIcon, { color: "colorTextIconBusy", decorative: true, size: "sizeIcon10" }),
    color: "colorTextWarning"
  },
  ConnectivityUnavailable: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ConnectivityUnavailableIcon.ConnectivityUnavailableIcon, { color: "colorTextIconUnavailable", decorative: true, size: "sizeIcon10" }),
    color: "colorTextError"
  },
  ConnectivityNeutral: {
    badgeVariant: "default",
    icon: /* @__PURE__ */ React.createElement(import_ConnectivityNeutralIcon.ConnectivityNeutralIcon, { color: "colorTextIconNeutral", decorative: true, size: "sizeIcon10" }),
    color: "colorTextNeutral"
  },
  ConnectivityOffline: {
    badgeVariant: "decorative10",
    icon: /* @__PURE__ */ React.createElement(import_ConnectivityOfflineIcon.ConnectivityOfflineIcon, { color: "colorTextIconOffline", decorative: true, size: "sizeIcon10" }),
    color: "colorTextWeak"
  }
};

// src/StatusBadge.tsx
var paddingX = (size) => {
  switch (size) {
    case "small":
      return "space20";
    case "borderless":
      return "space0";
    default:
      return "space30";
  }
};
var paddingY = (size) => {
  switch (size) {
    case "small":
      return "space10";
    case "borderless":
      return "space0";
    default:
      return "space20";
  }
};
var badgeStyles = { ...import_badge.badgeBaseStyles, ...import_badge.badgeVariantStyles.default };
var badgeBorderlessStyles = {
  boxShadow: "none",
  backgroundColor: "transparent"
};
var StatusBadge = React2.forwardRef(
  ({ children, element = "STATUS_BADGE", size, variant, ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        as: "span",
        element,
        variant,
        ...badgeStyles,
        color: StatusObject[variant].color,
        paddingX: paddingX(size),
        paddingY: paddingY(size),
        display: "flex",
        flexDirection: "row",
        columnGap: "space20",
        alignItems: "center",
        ...size === "borderless" && badgeBorderlessStyles,
        ref
      },
      StatusObject[variant].icon,
      children
    );
  }
);
StatusBadge.displayName = "StatusBadge";

// src/index.tsx
var import_menu6 = require("@twilio-paste/menu");

// src/StatusMenuBadge.tsx
var import_menu = require("@twilio-paste/menu");
var React3 = __toESM(require("react"));
var StatusMenuBadge = React3.forwardRef(
  ({ children, variant, element = "STATUS_MENU_BADGE", ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(StatusBadge, { as: "span", element, variant }, /* @__PURE__ */ React3.createElement(import_menu.MenuBadgeContent, { ...props, element, ref }, children));
  }
);
StatusMenuBadge.displayName = "StatusMenuBadge";

// src/StatusMenu.tsx
var import_menu2 = require("@twilio-paste/menu");
var React4 = __toESM(require("react"));
var StatusMenu = React4.forwardRef((props, ref) => {
  return /* @__PURE__ */ React4.createElement(import_menu2.Menu, { element: "STATUS_MENU", ...props, ref });
});
StatusMenu.displayName = "StatusMenu";

// src/StatusMenuItem.tsx
var import_menu3 = require("@twilio-paste/menu");
var React5 = __toESM(require("react"));
var StatusMenuItem = React5.forwardRef((props, ref) => {
  return /* @__PURE__ */ React5.createElement(import_menu3.MenuItem, { element: "STATUS_MENU_ITEM", ...props, ref });
});
StatusMenuItem.displayName = "StatusMenuItem";

// src/StatusMenuItemCheckbox.tsx
var import_menu4 = require("@twilio-paste/menu");
var React6 = __toESM(require("react"));
var StatusMenuItemCheckbox = React6.forwardRef((props, ref) => {
  return /* @__PURE__ */ React6.createElement(import_menu4.MenuItemCheckbox, { element: "STATUS_MENU_ITEM_CHECKBOX", ...props, ref });
});
StatusMenuItemCheckbox.displayName = "StatusMenuItemCheckbox";

// src/StatusMenuItemRadio.tsx
var import_menu5 = require("@twilio-paste/menu");
var React7 = __toESM(require("react"));
var StatusMenuItemRadio = React7.forwardRef((props, ref) => {
  return /* @__PURE__ */ React7.createElement(import_menu5.MenuItemRadio, { element: "STATUS_MENU_ITEM_RADIO", ...props, ref });
});
StatusMenuItemRadio.displayName = "StatusMenuItemRadio";

// src/StatusMenuItemChild.tsx
var import_box2 = require("@twilio-paste/box");
var React8 = __toESM(require("react"));
var StatusMenuItemChild = React8.forwardRef(
  ({ children, variant, element = "MENU_ITEM_STATUS", ...props }, ref) => {
    return /* @__PURE__ */ React8.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        element,
        display: "flex",
        flexDirection: "row",
        columnGap: "space20",
        alignItems: "center",
        variant,
        ref
      },
      StatusObject[variant].icon,
      children
    );
  }
);
StatusMenuItemChild.displayName = "StatusMenuItemChild";
