// @twilio-paste/user-dialog/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  UserDialog: () => UserDialog,
  UserDialogContainer: () => UserDialogContainer,
  UserDialogList: () => UserDialogList,
  UserDialogListItem: () => UserDialogListItem,
  UserDialogSeparator: () => UserDialogSeparator,
  UserDialogUserEmail: () => UserDialogUserEmail,
  UserDialogUserInfo: () => UserDialogUserInfo,
  UserDialogUserName: () => UserDialogUserName,
  useUserDialogListState: () => useUserDialogListState,
  useUserDialogState: () => useUserDialogState
});
module.exports = __toCommonJS(src_exports);

// src/UserDialog.tsx
var import_avatar = require("@twilio-paste/avatar");
var import_box = require("@twilio-paste/box");
var import_button = require("@twilio-paste/button");
var import_ChevronDownIcon = require("@twilio-paste/icons/cjs/ChevronDownIcon");
var import_non_modal_dialog_primitive2 = require("@twilio-paste/non-modal-dialog-primitive");
var import_theme = require("@twilio-paste/theme");
var React2 = __toESM(require("react"));

// src/UserDialogContainer.tsx
var import_non_modal_dialog_primitive = require("@twilio-paste/non-modal-dialog-primitive");
var React = __toESM(require("react"));
var UserDialogContext = React.createContext({});
var useUserDialogState = (props) => {
  return (0, import_non_modal_dialog_primitive.useNonModalDialogPrimitiveState)({ ...props, placement: "bottom-start", gutter: 0 });
};
var UserDialogContainer = ({
  children,
  state,
  name,
  src,
  icon,
  ...initialState
}) => {
  const nonModalDialogState = state || (0, import_non_modal_dialog_primitive.useNonModalDialogPrimitiveState)({
    modal: true,
    visible: false,
    placement: "bottom-start",
    gutter: 0,
    ...initialState
  });
  const avatarProps = { name, src, icon };
  return /* @__PURE__ */ React.createElement(UserDialogContext.Provider, { value: { userDialogState: nonModalDialogState, avatarProps } }, children);
};
UserDialogContainer.displayName = "UserDialogContainer";

// src/UserDialog.tsx
var UserDialogPopover = React2.forwardRef(({ style, ...props }, ref) => {
  return /* @__PURE__ */ React2.createElement(
    import_box.Box,
    {
      ...(0, import_box.safelySpreadBoxProps)(props),
      ref,
      style,
      boxShadow: "shadow",
      borderWidth: "borderWidth10",
      borderColor: "colorBorderWeaker",
      borderRadius: "borderRadius30",
      padding: "space0",
      marginTop: "space30",
      width: "size30",
      zIndex: "zIndex80",
      outline: "none"
    },
    props.children
  );
});
UserDialogPopover.displayName = "UserDialogPopover";
var UserDialogButton = React2.forwardRef(
  ({ children, element, ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      import_button.Button,
      {
        ...props,
        variant: "reset",
        size: "reset",
        borderRadius: "borderRadius20",
        element: `${element}_BUTTON`,
        ref
      },
      children
    );
  }
);
UserDialogButton.displayName = "UserDialogButton";
var UserDialog = React2.forwardRef(
  ({ children, "aria-label": label, element = "USER_DIALOG", ...props }, ref) => {
    const { userDialogState, avatarProps } = React2.useContext(UserDialogContext);
    return /* @__PURE__ */ React2.createElement(React2.Fragment, null, /* @__PURE__ */ React2.createElement(import_non_modal_dialog_primitive2.NonModalDialogDisclosurePrimitive, { ...userDialogState, as: UserDialogButton, element }, /* @__PURE__ */ React2.createElement(
      import_box.Box,
      {
        display: "flex",
        columnGap: "space20",
        alignItems: "center",
        element: `${element}_BUTTON_CONTENTS`,
        color: "colorTextIcon",
        _hover: { color: "colorTextPrimaryStrongest" }
      },
      /* @__PURE__ */ React2.createElement(
        import_avatar.Avatar,
        {
          variant: "user",
          size: "sizeIcon40",
          name: avatarProps.name,
          icon: avatarProps.icon,
          src: avatarProps.src,
          element: `${element}_BUTTON_AVATAR`
        }
      ),
      /* @__PURE__ */ React2.createElement(import_ChevronDownIcon.ChevronDownIcon, { decorative: true, size: "sizeIcon30", element: `${element}_BUTTON_ICON` })
    )), /* @__PURE__ */ React2.createElement(
      import_non_modal_dialog_primitive2.NonModalDialogPrimitive,
      {
        ...userDialogState,
        ...props,
        preventBodyScroll: false,
        "aria-label": label,
        as: UserDialogPopover,
        element,
        ref
      },
      /* @__PURE__ */ React2.createElement(import_theme.StyledBase, null, children)
    ));
  }
);
UserDialog.displayName = "UserDialog";

// src/UserDialogUserInfo.tsx
var import_avatar2 = require("@twilio-paste/avatar");
var import_box2 = require("@twilio-paste/box");
var import_media_object = require("@twilio-paste/media-object");
var import_separator = require("@twilio-paste/separator");
var React3 = __toESM(require("react"));
var UserDialogUserInfo = React3.forwardRef(
  ({ children, element = "USER_DIALOG", ...props }, ref) => {
    const { avatarProps } = React3.useContext(UserDialogContext);
    return /* @__PURE__ */ React3.createElement(React3.Fragment, null, /* @__PURE__ */ React3.createElement(
      import_box2.Box,
      {
        ...(0, import_box2.safelySpreadBoxProps)(props),
        backgroundColor: "colorBackgroundWeak",
        padding: "space50",
        borderTopLeftRadius: "borderRadius30",
        borderTopRightRadius: "borderRadius30",
        element: `${element}_USER_INFO`,
        ref
      },
      /* @__PURE__ */ React3.createElement(import_media_object.MediaObject, { as: "div", verticalAlign: "center" }, /* @__PURE__ */ React3.createElement(import_media_object.MediaFigure, { as: "div", spacing: "space50" }, /* @__PURE__ */ React3.createElement(
        import_avatar2.Avatar,
        {
          variant: "user",
          size: "sizeIcon80",
          name: avatarProps.name,
          src: avatarProps.src,
          icon: avatarProps.icon,
          element: `${element}_USER_AVATAR`
        }
      )), /* @__PURE__ */ React3.createElement(import_media_object.MediaBody, { as: "div" }, children))
    ), /* @__PURE__ */ React3.createElement(import_separator.Separator, { orientation: "horizontal", element: `${element}_USER_INFO_SEPARATOR` }));
  }
);
UserDialogUserInfo.displayName = "UserDialogUserInfo";

// src/UserDialogUserName.tsx
var import_text = require("@twilio-paste/text");
var React4 = __toESM(require("react"));
var UserDialogUserName = React4.forwardRef(
  ({ children, element = "USER_DIALOG_USER_NAME", ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(
      import_text.Text,
      {
        ...props,
        as: "div",
        fontSize: "fontSize40",
        color: "colorText",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight40",
        element,
        ref
      },
      children
    );
  }
);
UserDialogUserName.displayName = "UserDialogName";

// src/UserDialogUserEmail.tsx
var import_text2 = require("@twilio-paste/text");
var React5 = __toESM(require("react"));
var UserDialogUserEmail = React5.forwardRef(
  ({ children, element = "USER_DIALOG_USER_EMAIL", ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement(
      import_text2.Text,
      {
        ...props,
        as: "div",
        fontSize: "fontSize30",
        color: "colorTextWeak",
        fontWeight: "fontWeightLight",
        lineHeight: "lineHeight20",
        element,
        ref
      },
      children
    );
  }
);
UserDialogUserEmail.displayName = "UserDialogEmail";

// src/UserDialogList.tsx
var import_box3 = require("@twilio-paste/box");
var import_listbox_primitive = require("@twilio-paste/listbox-primitive");
var React6 = __toESM(require("react"));
var useUserDialogListState = (props) => {
  return (0, import_listbox_primitive.useListboxPrimitiveState)({ ...props, orientation: "vertical" });
};
var UserDialogListbox = React6.forwardRef(
  ({ element = "USER_DIALOG", ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(
      import_box3.Box,
      {
        ...(0, import_box3.safelySpreadBoxProps)(props),
        element: `${element}_LIST`,
        backgroundColor: "colorBackgroundBody",
        borderRadius: "borderRadius30",
        paddingY: "space30",
        ref
      },
      props.children
    );
  }
);
UserDialogListbox.displayName = "UserDialogListbox";
var UserDialogList = React6.forwardRef(({ children, ...props }, ref) => {
  return /* @__PURE__ */ React6.createElement(import_listbox_primitive.ListboxPrimitive, { ...props, as: UserDialogListbox, ref }, children);
});
UserDialogList.displayName = "UserDialogList";

// src/UserDialogListItem.tsx
var import_box4 = require("@twilio-paste/box");
var import_listbox_primitive2 = require("@twilio-paste/listbox-primitive");
var React7 = __toESM(require("react"));
var UserDialogListboxItem = React7.forwardRef(
  ({ children, href, element = "USER_DIALOG", ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(
      import_box4.Box,
      {
        ...(0, import_box4.safelySpreadBoxProps)(props),
        element: `${element}_LIST_ITEM`,
        as: href ? "a" : "div",
        href,
        ref,
        width: "100%",
        paddingLeft: "space60",
        paddingRight: "space70",
        paddingY: "space30",
        display: "flex",
        columnGap: "space30",
        color: "colorText",
        borderLeftStyle: "solid",
        borderLeftWidth: "borderWidth20",
        borderLeftColor: "colorBorderWeakest",
        outline: "none",
        _hover: {
          cursor: "pointer",
          backgroundColor: "colorBackgroundPrimaryWeakest",
          color: "colorTextPrimary",
          borderLeftStyle: "solid",
          borderLeftWidth: "borderWidth20",
          borderLeftColor: "colorBorderPrimary"
        },
        _focus: {
          cursor: "pointer",
          backgroundColor: "colorBackgroundPrimaryWeakest",
          color: "colorTextPrimary",
          borderLeftStyle: "solid",
          borderLeftWidth: "borderWidth20",
          borderLeftColor: "colorBorderPrimary"
        }
      },
      children
    );
  }
);
UserDialogListboxItem.displayName = "UserDialogListboxItem";
var UserDialogListItem = React7.forwardRef(
  ({ children, ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(import_listbox_primitive2.ListboxPrimitiveItem, { ...props, ref, as: UserDialogListboxItem }, children);
  }
);
UserDialogListItem.displayName = "UserDialogListItem";

// src/UserDialogSeparator.tsx
var import_box5 = require("@twilio-paste/box");
var import_separator2 = require("@twilio-paste/separator");
var React8 = __toESM(require("react"));
var UserDialogSeparator = React8.forwardRef(({ element = "USER_DIALOG_SEPARATOR", ...props }, ref) => {
  return /* @__PURE__ */ React8.createElement(import_box5.Box, { paddingY: "space30" }, /* @__PURE__ */ React8.createElement(import_separator2.Separator, { ...props, element, ref, orientation: "horizontal" }));
});
UserDialogSeparator.displayName = "UserDialogSeparator";
