// @twilio-paste/listbox-primitive/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  ListboxPrimitive: () => ListboxPrimitive,
  ListboxPrimitiveGroup: () => import_reakit_library4.CompositeGroup,
  ListboxPrimitiveItem: () => ListboxPrimitiveItem,
  useListboxPrimitiveState: () => useListboxPrimitiveState
});
module.exports = __toCommonJS(src_exports);

// src/ListboxPrimitive.tsx
var import_reakit_library = require("@twilio-paste/reakit-library");
var React = __toESM(require("react"));
var ListboxPrimitive = React.forwardRef(
  ({ variant = "single", orientation, ...props }, ref) => /* @__PURE__ */ React.createElement(
    import_reakit_library.Composite,
    {
      as: "div",
      ...props,
      ref,
      role: "listbox",
      "aria-multiselectable": variant === "multiple",
      "aria-orientation": orientation
    }
  )
);
ListboxPrimitive.displayName = "ListboxPrimitive";

// src/ListboxPrimitiveItem.tsx
var import_reakit_library2 = require("@twilio-paste/reakit-library");
var React2 = __toESM(require("react"));
var ListboxPrimitiveItem = React2.forwardRef(
  ({ selected = false, onSelect, ...props }, ref) => /* @__PURE__ */ React2.createElement(
    import_reakit_library2.CompositeItem,
    {
      as: "button",
      ...props,
      ref,
      role: "option",
      "aria-selected": selected,
      type: "button",
      onClick: onSelect
    }
  )
);
ListboxPrimitiveItem.displayName = "ListboxPrimitiveItem";

// src/useListboxPrimitiveState.tsx
var import_reakit_library3 = require("@twilio-paste/reakit-library");
var useListboxPrimitiveState = (config = {}) => {
  return {
    ...(0, import_reakit_library3.useCompositeState)({
      orientation: "vertical",
      ...config,
      loop: "horizontal"
    })
  };
};

// src/index.tsx
var import_reakit_library4 = require("@twilio-paste/reakit-library");
