// @twilio-paste/listbox-primitive/dist/index.debug.es.js
// src/ListboxPrimitive.tsx
import { Composite } from "@twilio-paste/reakit-library";
import * as React from "react";
var ListboxPrimitive = React.forwardRef(
  ({ variant = "single", orientation, ...props }, ref) => /* @__PURE__ */ React.createElement(
    Composite,
    {
      as: "div",
      ...props,
      ref,
      role: "listbox",
      "aria-multiselectable": variant === "multiple",
      "aria-orientation": orientation
    }
  )
);
ListboxPrimitive.displayName = "ListboxPrimitive";

// src/ListboxPrimitiveItem.tsx
import { CompositeItem } from "@twilio-paste/reakit-library";
import * as React2 from "react";
var ListboxPrimitiveItem = React2.forwardRef(
  ({ selected = false, onSelect, ...props }, ref) => /* @__PURE__ */ React2.createElement(
    CompositeItem,
    {
      as: "button",
      ...props,
      ref,
      role: "option",
      "aria-selected": selected,
      type: "button",
      onClick: onSelect
    }
  )
);
ListboxPrimitiveItem.displayName = "ListboxPrimitiveItem";

// src/useListboxPrimitiveState.tsx
import { useCompositeState } from "@twilio-paste/reakit-library";
var useListboxPrimitiveState = (config = {}) => {
  return {
    ...useCompositeState({
      orientation: "vertical",
      ...config,
      loop: "horizontal"
    })
  };
};

// src/index.tsx
import { CompositeGroup } from "@twilio-paste/reakit-library";
export {
  ListboxPrimitive,
  CompositeGroup as ListboxPrimitiveGroup,
  ListboxPrimitiveItem,
  useListboxPrimitiveState
};
