// @twilio-paste/display-heading/dist/index.debug.es.js
// src/DisplayHeading.tsx
import { Text, safelySpreadTextProps } from "@twilio-paste/text";
import * as React from "react";
var getDisplayHeadingProps = (displayHeadingVariant, marginBottom) => {
  switch (displayHeadingVariant) {
    case "displayHeading10":
      return {
        marginBottom: marginBottom || "space170",
        fontSize: "fontSizeDisplay30",
        fontWeight: "fontWeightExtrabold",
        lineHeight: "lineHeightDisplay30"
      };
    case "displayHeading30":
      return {
        marginBottom: marginBottom || "space90",
        fontSize: "fontSizeDisplay10",
        fontWeight: "fontWeightExtrabold",
        lineHeight: "lineHeightDisplay10"
      };
    case "displayHeading20":
    default:
      return {
        marginBottom: marginBottom || "space130",
        fontSize: "fontSizeDisplay20",
        fontWeight: "fontWeightExtrabold",
        lineHeight: "lineHeightDisplay20"
      };
  }
};
var DisplayHeading = React.forwardRef(
  ({ element = "DISPLAY_HEADING", as, children, display = "block", id, marginBottom, variant, ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(
      Text,
      {
        ...safelySpreadTextProps(props),
        ...getDisplayHeadingProps(variant, marginBottom),
        as,
        display,
        element,
        id,
        color: "colorText",
        fontFamily: "fontFamilyDisplay",
        variant,
        ref
      },
      children
    );
  }
);
DisplayHeading.displayName = "DisplayHeading";
export {
  DisplayHeading
};
