// @twilio-paste/help-text/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  HelpText: () => HelpText
});
module.exports = __toCommonJS(src_exports);

// src/HelpText.tsx
var import_box = require("@twilio-paste/box");
var import_ErrorIcon = require("@twilio-paste/icons/cjs/ErrorIcon");
var import_SuccessIcon = require("@twilio-paste/icons/cjs/SuccessIcon");
var import_WarningIcon = require("@twilio-paste/icons/cjs/WarningIcon");
var React = __toESM(require("react"));
var HelpTextVariants = {
  DEFAULT: "default",
  ERROR: "error",
  ERROR_INVERSE: "error_inverse",
  INVERSE: "inverse",
  SUCCESS: "success",
  WARNING: "warning"
};
var VariantOptions = {
  [HelpTextVariants.DEFAULT]: {
    textColor: "colorTextWeak",
    icon: null
  },
  [HelpTextVariants.INVERSE]: {
    textColor: "colorTextInverseWeaker",
    icon: null
  },
  [HelpTextVariants.ERROR]: {
    textColor: "colorTextError",
    icon: /* @__PURE__ */ React.createElement(import_box.Box, { flexShrink: 0 }, /* @__PURE__ */ React.createElement(import_ErrorIcon.ErrorIcon, { color: "colorTextError", decorative: true, size: "sizeIcon20" }))
  },
  [HelpTextVariants.ERROR_INVERSE]: {
    textColor: "colorTextErrorWeak",
    icon: /* @__PURE__ */ React.createElement(import_box.Box, { flexShrink: 0 }, /* @__PURE__ */ React.createElement(import_ErrorIcon.ErrorIcon, { color: "colorTextErrorWeak", decorative: true, size: "sizeIcon20" }))
  },
  [HelpTextVariants.SUCCESS]: {
    textColor: "colorTextSuccess",
    icon: /* @__PURE__ */ React.createElement(import_box.Box, { flexShrink: 0 }, /* @__PURE__ */ React.createElement(import_SuccessIcon.SuccessIcon, { color: "colorTextIconSuccess", decorative: true, size: "sizeIcon20" }))
  },
  [HelpTextVariants.WARNING]: {
    textColor: "colorTextWarning",
    icon: /* @__PURE__ */ React.createElement(import_box.Box, { flexShrink: 0 }, /* @__PURE__ */ React.createElement(import_WarningIcon.WarningIcon, { color: "colorTextWarning", decorative: true, size: "sizeIcon20" }))
  }
};
var HelpText = React.forwardRef(
  ({ marginTop, children, variant = "default", element = "HELP_TEXT", ...props }, ref) => {
    const { textColor, icon } = VariantOptions[variant];
    return /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        display: "flex",
        columnGap: "space20",
        marginTop: marginTop || "space30",
        ref,
        element,
        variant,
        as: "div",
        color: textColor,
        fontSize: "fontSize30",
        lineHeight: "lineHeight30"
      },
      icon,
      /* @__PURE__ */ React.createElement("span", null, children)
    );
  }
);
HelpText.displayName = "HelpText";
