// @twilio-paste/button/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// ../../../../node_modules/deepmerge/dist/cjs.js
var require_cjs = __commonJS({
  "../../../../node_modules/deepmerge/dist/cjs.js"(exports, module2) {
    "use strict";
    var isMergeableObject = function isMergeableObject2(value) {
      return isNonNullObject(value) && !isSpecial(value);
    };
    function isNonNullObject(value) {
      return !!value && typeof value === "object";
    }
    function isSpecial(value) {
      var stringValue = Object.prototype.toString.call(value);
      return stringValue === "[object RegExp]" || stringValue === "[object Date]" || isReactElement(value);
    }
    var canUseSymbol = typeof Symbol === "function" && Symbol.for;
    var REACT_ELEMENT_TYPE = canUseSymbol ? Symbol.for("react.element") : 60103;
    function isReactElement(value) {
      return value.$$typeof === REACT_ELEMENT_TYPE;
    }
    function emptyTarget(val) {
      return Array.isArray(val) ? [] : {};
    }
    function cloneUnlessOtherwiseSpecified(value, options) {
      return options.clone !== false && options.isMergeableObject(value) ? deepmerge(emptyTarget(value), value, options) : value;
    }
    function defaultArrayMerge(target, source, options) {
      return target.concat(source).map(function(element) {
        return cloneUnlessOtherwiseSpecified(element, options);
      });
    }
    function getMergeFunction(key, options) {
      if (!options.customMerge) {
        return deepmerge;
      }
      var customMerge = options.customMerge(key);
      return typeof customMerge === "function" ? customMerge : deepmerge;
    }
    function getEnumerableOwnPropertySymbols(target) {
      return Object.getOwnPropertySymbols ? Object.getOwnPropertySymbols(target).filter(function(symbol) {
        return target.propertyIsEnumerable(symbol);
      }) : [];
    }
    function getKeys(target) {
      return Object.keys(target).concat(getEnumerableOwnPropertySymbols(target));
    }
    function propertyIsOnObject(object, property) {
      try {
        return property in object;
      } catch (_) {
        return false;
      }
    }
    function propertyIsUnsafe(target, key) {
      return propertyIsOnObject(target, key) && !(Object.hasOwnProperty.call(target, key) && Object.propertyIsEnumerable.call(target, key));
    }
    function mergeObject(target, source, options) {
      var destination = {};
      if (options.isMergeableObject(target)) {
        getKeys(target).forEach(function(key) {
          destination[key] = cloneUnlessOtherwiseSpecified(target[key], options);
        });
      }
      getKeys(source).forEach(function(key) {
        if (propertyIsUnsafe(target, key)) {
          return;
        }
        if (propertyIsOnObject(target, key) && options.isMergeableObject(source[key])) {
          destination[key] = getMergeFunction(key, options)(target[key], source[key], options);
        } else {
          destination[key] = cloneUnlessOtherwiseSpecified(source[key], options);
        }
      });
      return destination;
    }
    function deepmerge(target, source, options) {
      options = options || {};
      options.arrayMerge = options.arrayMerge || defaultArrayMerge;
      options.isMergeableObject = options.isMergeableObject || isMergeableObject;
      options.cloneUnlessOtherwiseSpecified = cloneUnlessOtherwiseSpecified;
      var sourceIsArray = Array.isArray(source);
      var targetIsArray = Array.isArray(target);
      var sourceAndTargetTypesMatch = sourceIsArray === targetIsArray;
      if (!sourceAndTargetTypesMatch) {
        return cloneUnlessOtherwiseSpecified(source, options);
      } else if (sourceIsArray) {
        return options.arrayMerge(target, source, options);
      } else {
        return mergeObject(target, source, options);
      }
    }
    deepmerge.all = function deepmergeAll(array, options) {
      if (!Array.isArray(array)) {
        throw new Error("first argument should be an array");
      }
      return array.reduce(function(prev, next) {
        return deepmerge(prev, next, options);
      }, {});
    };
    var deepmerge_1 = deepmerge;
    module2.exports = deepmerge_1;
  }
});

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Button: () => Button,
  ButtonToggleStyles: () => ToggleStyles,
  DestructiveSecondaryButtonToggleStyles: () => DestructiveSecondaryToggleStyles
});
module.exports = __toCommonJS(src_exports);

// src/Button.tsx
var import_anchor = require("@twilio-paste/anchor");
var import_animation_library = require("@twilio-paste/animation-library");
var import_box13 = require("@twilio-paste/box");
var import_ArrowForwardIcon = require("@twilio-paste/icons/cjs/ArrowForwardIcon");
var import_LinkExternalIcon = require("@twilio-paste/icons/cjs/LinkExternalIcon");
var import_spinner = require("@twilio-paste/spinner");
var React13 = __toESM(require("react"));

// src/DestructiveButton.tsx
var import_box = require("@twilio-paste/box");
var import_deepmerge2 = __toESM(require_cjs());
var React = __toESM(require("react"));

// src/styles.ts
var import_deepmerge = __toESM(require_cjs());
var ResetStyles = {
  appearance: "none",
  background: "none",
  display: "inline-block",
  border: "none",
  outline: "none",
  transition: "background-color 100ms ease-in, box-shadow 100ms ease-in, color 100ms ease-in",
  fontFamily: "inherit",
  fontWeight: "fontWeightSemibold",
  textDecoration: "none",
  position: "relative",
  margin: "space0",
  borderRadius: "borderRadius20",
  _hover: { textDecoration: "none" },
  _focus: { textDecoration: "none", boxShadow: "shadowFocus" },
  _active: { textDecoration: "none" }
};
var BaseStyles = {
  default: (0, import_deepmerge.default)(ResetStyles, {
    cursor: "pointer",
    _active: { boxShadow: "none" }
  }),
  disabled: (0, import_deepmerge.default)(ResetStyles, {
    cursor: "not-allowed"
  }),
  loading: (0, import_deepmerge.default)(ResetStyles, {
    cursor: "wait"
  })
};
var SizeStyles = {
  default: {
    paddingTop: "space30",
    paddingBottom: "space30",
    paddingLeft: "space40",
    paddingRight: "space40",
    borderRadius: "borderRadius20",
    fontSize: "fontSize30",
    lineHeight: "lineHeight20"
  },
  small: {
    paddingTop: "space20",
    paddingBottom: "space20",
    paddingLeft: "space30",
    paddingRight: "space30",
    borderRadius: "borderRadius10",
    fontSize: "fontSize30",
    lineHeight: "lineHeight20"
  },
  icon: {
    padding: "space30",
    borderRadius: "borderRadius20"
  },
  icon_small: {
    padding: "space20",
    borderRadius: "borderRadius20"
  },
  reset: {
    paddingTop: "space0",
    paddingRight: "space0",
    paddingBottom: "space0",
    paddingLeft: "space0",
    borderWidth: "borderWidth0",
    fontSize: "inherit"
  },
  rounded_small: {
    borderRadius: "borderRadiusPill",
    paddingTop: "space20",
    paddingBottom: "space20",
    paddingLeft: "space30",
    paddingRight: "space30",
    fontSize: "fontSize30",
    lineHeight: "lineHeight20"
  },
  circle: {
    padding: "space30",
    borderRadius: "borderRadiusCircle"
  },
  circle_small: {
    padding: "space20",
    borderRadius: "borderRadiusCircle"
  }
};
var ToggleStyles = {
  transition: "background-color 150ms ease-in, box-shadow 150ms ease-in, color 150ms ease-in",
  color: "colorText",
  backgroundColor: "colorBackgroundBody",
  _disabled: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderWeaker",
    color: "colorTextWeaker"
  },
  _hover: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderPrimary",
    color: "colorTextPrimary"
  },
  _active: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderPrimaryStrongest",
    color: "colorTextPrimaryStrongest"
  },
  _pressed: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimary",
    color: "colorTextPrimary"
  },
  _pressed_hover: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimaryStronger",
    color: "colorTextPrimaryStronger"
  },
  _pressed_active: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimaryStrongest",
    color: "colorTextPrimaryStrongest"
  },
  _pressed_focus: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowFocusShadowBorder",
    color: "colorTextPrimary"
  },
  _pressed_disabled: {
    backgroundColor: "colorBackgroundStrong",
    boxShadow: "shadowBorderWeaker",
    color: "colorTextWeakest"
  }
};
var DestructiveSecondaryToggleStyles = {
  transition: "background-color 150ms ease-in, box-shadow 150ms ease-in, color 150ms ease-in",
  color: "colorTextError",
  backgroundColor: "colorBackgroundBody",
  _disabled: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderWeaker",
    color: "colorTextWeaker"
  },
  _hover: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderError",
    color: "colorTextError"
  },
  _active: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderErrorStrongest",
    color: "colorTextErrorStrongest"
  },
  _pressed: {
    backgroundColor: "colorBackgroundErrorWeakest",
    boxShadow: "shadowBorderError",
    color: "colorTextError"
  },
  _pressed_hover: {
    backgroundColor: "colorBackgroundErrorWeakest",
    boxShadow: "shadowBorderErrorStronger",
    color: "colorTextErrorStronger"
  },
  _pressed_active: {
    backgroundColor: "colorBackgroundErrorWeakest",
    boxShadow: "shadowBorderError",
    color: "colorTextError"
  },
  _pressed_focus: {
    backgroundColor: "colorBackgroundErrorWeakest",
    boxShadow: "shadowFocusShadowBorder",
    color: "colorTextError"
  },
  _pressed_disabled: {
    backgroundColor: "colorBackgroundStrong",
    boxShadow: "shadowBorderWeaker",
    color: "colorTextWeakest"
  }
};
var ToggleIconButtonStyles = {
  transition: "background-color 150ms ease-in, box-shadow 150ms ease-in, color 150ms ease-in",
  color: "colorTextIcon",
  backgroundColor: "colorBackgroundBody",
  _hover: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderPrimary",
    color: "colorTextPrimaryStronger"
  },
  _active: {
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderPrimaryStrongest",
    color: "colorTextPrimaryStrongest"
  },
  _disabled: {
    backgroundColor: "colorBackgroundBody",
    color: "colorTextWeaker"
  },
  _pressed: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimary",
    color: "colorTextPrimary"
  },
  _pressed_hover: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimaryStronger",
    color: "colorTextPrimaryStronger"
  },
  _pressed_focus: {
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowFocusShadowBorder",
    color: "colorTextPrimary"
  },
  _pressed_disabled: {
    backgroundColor: "colorBackgroundStrong",
    boxShadow: "shadowBorderWeaker",
    color: "colorTextWeakest"
  }
};

// src/DestructiveButton.tsx
var defaultStyles = (0, import_deepmerge2.default)(BaseStyles.default, {
  color: "colorTextInverse",
  backgroundColor: "colorBackgroundDestructive",
  boxShadow: "shadowBorderDestructive",
  _hover: {
    color: "colorTextDestructive",
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderDestructive"
  },
  _focus: {
    boxShadow: "shadowFocus"
  },
  _active: {
    color: "colorTextDestructive",
    backgroundColor: "colorBackgroundDestructiveWeakest",
    boxShadow: "shadowBorderDestructive"
  }
});
var loadingStyles = (0, import_deepmerge2.default)(BaseStyles.loading, {
  color: "colorTextDestructive",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeak"
});
var disabledStyles = (0, import_deepmerge2.default)(BaseStyles.disabled, {
  color: "colorTextWeakest",
  backgroundColor: "colorBackgroundStrong",
  boxShadow: "shadowBorderWeaker"
});
var ButtonStyleMapping = {
  default: defaultStyles,
  loading: loadingStyles,
  disabled: disabledStyles
};
var DestructiveButton = React.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React.createElement(
      import_box.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
DestructiveButton.defaultProps = {
  as: "button"
};
DestructiveButton.displayName = "DestructiveButton";

// src/DestructiveIconButton.tsx
var import_box2 = require("@twilio-paste/box");
var import_deepmerge3 = __toESM(require_cjs());
var React2 = __toESM(require("react"));
var defaultStyles2 = (0, import_deepmerge3.default)(BaseStyles.default, {
  color: "colorTextDestructive",
  _hover: {
    color: "colorTextDestructiveStrongest"
  },
  _active: {
    color: "colorTextDestructiveStrongest"
  }
});
var loadingStyles2 = (0, import_deepmerge3.default)(BaseStyles.loading, {
  color: "colorTextDestructive"
});
var disabledStyles2 = (0, import_deepmerge3.default)(BaseStyles.disabled, {
  color: "colorTextWeaker"
});
var ButtonStyleMapping2 = {
  default: defaultStyles2,
  loading: loadingStyles2,
  disabled: disabledStyles2
};
var DestructiveIconButton = React2.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(
      import_box2.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box2.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping2[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
DestructiveIconButton.defaultProps = {
  as: "button"
};
DestructiveIconButton.displayName = "DestructiveIconButton";

// src/DestructiveLinkButton.tsx
var import_box3 = require("@twilio-paste/box");
var import_deepmerge4 = __toESM(require_cjs());
var React3 = __toESM(require("react"));
var defaultStyles3 = (0, import_deepmerge4.default)(BaseStyles.default, {
  color: "colorTextLinkDestructive",
  textAlign: "left",
  transition: "none",
  _hover: { color: "colorTextLinkDestructiveStrongest", textDecoration: "underline" },
  _active: { color: "colorTextLinkDestructiveStrongest", textDecoration: "underline" }
});
var loadingStyles3 = (0, import_deepmerge4.default)(BaseStyles.loading, {
  color: "colorTextLinkDestructive",
  textAlign: "left",
  _hover: { color: "colorTextLinkDestructiveStronger" },
  _active: { color: "colorTextLinkDestructiveStronger" },
  _focus: { color: "colorTextLinkDestructiveStronger" }
});
var disabledStyles3 = (0, import_deepmerge4.default)(BaseStyles.disabled, {
  color: "colorTextWeaker",
  textAlign: "left",
  _hover: { color: "colorTextLinkDestructiveWeak" },
  _active: { color: "colorTextLinkDestructiveWeak" },
  _focus: { color: "colorTextLinkDestructiveWeak" }
});
var ButtonStyleMapping3 = {
  default: defaultStyles3,
  loading: loadingStyles3,
  disabled: disabledStyles3
};
var DestructiveLinkButton = React3.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React3.createElement(
      import_box3.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box3.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping3[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
DestructiveLinkButton.defaultProps = {
  as: "a"
};
DestructiveLinkButton.displayName = "DestructiveLinkButton";

// src/DestructiveSecondaryButton.tsx
var import_box4 = require("@twilio-paste/box");
var import_deepmerge5 = __toESM(require_cjs());
var React4 = __toESM(require("react"));
var defaultStyles4 = (0, import_deepmerge5.default)(BaseStyles.default, {
  color: "colorTextDestructive",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeak",
  _hover: {
    color: "colorTextDestructive",
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderDestructive"
  },
  _focus: {
    boxShadow: "shadowFocusShadowBorder"
  },
  _active: {
    color: "colorTextDestructive",
    backgroundColor: "colorBackgroundDestructiveWeakest",
    boxShadow: "shadowBorderDestructive"
  }
});
var baseLoadingStyles = {
  color: "colorTextDestructive",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeak"
};
var loadingStyles4 = (0, import_deepmerge5.default)(BaseStyles.loading, {
  ...baseLoadingStyles,
  _hover: baseLoadingStyles,
  _active: baseLoadingStyles,
  _focus: baseLoadingStyles
});
var baseDisabledStyles = {
  color: "colorTextWeaker",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeaker"
};
var disabledStyles4 = (0, import_deepmerge5.default)(BaseStyles.disabled, {
  ...baseDisabledStyles,
  _hover: baseDisabledStyles,
  _active: baseDisabledStyles,
  _focus: baseDisabledStyles
});
var ButtonStyleMapping4 = {
  default: defaultStyles4,
  loading: loadingStyles4,
  disabled: disabledStyles4
};
var DestructiveSecondaryButton = React4.forwardRef(
  ({ size, buttonState, fullWidth, pressed, ...props }, ref) => {
    const toggleStyles = pressed === void 0 ? {} : DestructiveSecondaryToggleStyles;
    return /* @__PURE__ */ React4.createElement(
      import_box4.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        "aria-pressed": pressed,
        ...(0, import_box4.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping4[buttonState],
        ...toggleStyles,
        ...SizeStyles[size]
      }
    );
  }
);
DestructiveSecondaryButton.defaultProps = {
  as: "button"
};
DestructiveSecondaryButton.displayName = "DestructiveSecondaryButton";

// src/InverseButton.tsx
var import_box5 = require("@twilio-paste/box");
var import_deepmerge6 = __toESM(require_cjs());
var React5 = __toESM(require("react"));
var defaultStyles5 = (0, import_deepmerge6.default)(BaseStyles.default, {
  color: "colorTextInverse",
  backgroundColor: "colorBackgroundInverse",
  boxShadow: "shadowBorderInverseWeaker",
  _hover: {
    color: "colorTextInverse",
    backgroundColor: "colorBackgroundInverseStrong",
    boxShadow: "shadowBorderInverseStronger"
  },
  _focus: {
    boxShadow: "shadowFocusInverse"
  },
  _active: {
    color: "colorTextInverse",
    backgroundColor: "colorBackgroundInverseStrong",
    boxShadow: "shadowBorderInverseStrongest"
  }
});
var loadingStyles5 = (0, import_deepmerge6.default)(BaseStyles.loading, {
  color: "colorTextInverse",
  backgroundColor: "colorBackgroundInverseStrong",
  boxShadow: "shadowBorderInverseWeaker"
});
var disabledStyles5 = (0, import_deepmerge6.default)(BaseStyles.disabled, {
  color: "colorTextInverseWeakest",
  backgroundColor: "colorBackgroundInverseStrong",
  boxShadow: "shadowBorderInverseWeakest"
});
var ButtonStyleMapping5 = {
  default: defaultStyles5,
  loading: loadingStyles5,
  disabled: disabledStyles5
};
var InverseButton = React5.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React5.createElement(
      import_box5.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box5.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping5[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
InverseButton.defaultProps = {
  as: "button"
};
InverseButton.displayName = "InverseButton";

// src/InverseLinkButton.tsx
var import_box6 = require("@twilio-paste/box");
var import_deepmerge7 = __toESM(require_cjs());
var React6 = __toESM(require("react"));
var defaultStyles6 = (0, import_deepmerge7.default)(BaseStyles.default, {
  color: "colorTextInverse",
  textAlign: "left",
  transition: "none",
  _hover: { color: "colorTextInverseWeaker", textDecoration: "underline" },
  _focus: { boxShadow: "shadowFocusInverse" },
  _active: { color: "colorTextInverseWeaker", textDecoration: "underline" }
});
var loadingStyles6 = (0, import_deepmerge7.default)(BaseStyles.loading, {
  color: "colorTextInverse",
  textAlign: "left"
});
var disabledStyles6 = (0, import_deepmerge7.default)(BaseStyles.disabled, {
  color: "colorTextInverseWeakest",
  textAlign: "left"
});
var ButtonStyleMapping6 = {
  default: defaultStyles6,
  loading: loadingStyles6,
  disabled: disabledStyles6
};
var InverseLinkButton = React6.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(
      import_box6.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box6.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping6[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
InverseLinkButton.defaultProps = {
  as: "a"
};
InverseLinkButton.displayName = "InverseLinkButton";

// src/LinkButton.tsx
var import_box7 = require("@twilio-paste/box");
var import_deepmerge8 = __toESM(require_cjs());
var React7 = __toESM(require("react"));
var defaultStyles7 = (0, import_deepmerge8.default)(BaseStyles.default, {
  color: "colorTextPrimary",
  textAlign: "left",
  transition: "none",
  _hover: { color: "colorTextPrimaryStrongest", textDecoration: "underline" },
  _active: { color: "colorTextPrimaryStrongest", textDecoration: "underline" }
});
var loadingStyles7 = (0, import_deepmerge8.default)(BaseStyles.loading, {
  color: "colorTextPrimary",
  textAlign: "left"
});
var disabledStyles7 = (0, import_deepmerge8.default)(BaseStyles.disabled, {
  color: "colorTextWeaker",
  textAlign: "left"
});
var ButtonStyleMapping7 = {
  default: defaultStyles7,
  loading: loadingStyles7,
  disabled: disabledStyles7
};
var LinkButton = React7.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React7.createElement(
      import_box7.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box7.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping7[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
LinkButton.defaultProps = {
  as: "a"
};
LinkButton.displayName = "LinkButton";

// src/PrimaryButton.tsx
var import_box8 = require("@twilio-paste/box");
var import_deepmerge9 = __toESM(require_cjs());
var React8 = __toESM(require("react"));
var defaultStyles8 = (0, import_deepmerge9.default)(BaseStyles.default, {
  color: "colorTextInverse",
  backgroundColor: "colorBackgroundPrimary",
  boxShadow: "shadowBorderPrimary",
  _hover: {
    color: "colorTextPrimary",
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderPrimary"
  },
  _focus: {
    boxShadow: "shadowFocus"
  },
  _active: {
    color: "colorTextPrimaryStrong",
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimaryStrong"
  }
});
var loadingStyles8 = (0, import_deepmerge9.default)(BaseStyles.loading, {
  color: "colorTextPrimary",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeak"
});
var disabledStyles8 = (0, import_deepmerge9.default)(BaseStyles.disabled, {
  color: "colorTextWeakest",
  backgroundColor: "colorBackgroundStrong",
  boxShadow: "shadowBorderWeaker"
});
var ButtonStyleMapping8 = {
  default: defaultStyles8,
  loading: loadingStyles8,
  disabled: disabledStyles8
};
var PrimaryButton = React8.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React8.createElement(
      import_box8.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box8.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping8[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
PrimaryButton.defaultProps = {
  as: "button"
};
PrimaryButton.displayName = "PrimaryButton";

// src/PrimaryIconButton.tsx
var import_box9 = require("@twilio-paste/box");
var import_deepmerge10 = __toESM(require_cjs());
var React9 = __toESM(require("react"));
var defaultStyles9 = (0, import_deepmerge10.default)(BaseStyles.default, {
  color: "colorTextPrimary",
  _hover: {
    color: "colorTextPrimaryStrongest"
  },
  _active: {
    color: "colorTextPrimaryStrongest"
  }
});
var loadingStyles9 = (0, import_deepmerge10.default)(BaseStyles.loading, {
  color: "colorTextPrimary"
});
var disabledStyles9 = (0, import_deepmerge10.default)(BaseStyles.disabled, {
  color: "colorTextWeaker"
});
var ButtonStyleMapping9 = {
  default: defaultStyles9,
  loading: loadingStyles9,
  disabled: disabledStyles9
};
var PrimaryIconButton = React9.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React9.createElement(
      import_box9.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box9.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping9[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
PrimaryIconButton.defaultProps = {
  as: "button"
};
PrimaryIconButton.displayName = "PrimaryIconButton";

// src/ResetButton.tsx
var import_box10 = require("@twilio-paste/box");
var import_deepmerge11 = __toESM(require_cjs());
var React10 = __toESM(require("react"));
var defaultStyles10 = (0, import_deepmerge11.default)(BaseStyles.default, {
  fontWeight: "inherit",
  color: "inherit"
});
var loadingStyles10 = (0, import_deepmerge11.default)(BaseStyles.loading, { fontSize: "inherit", fontWeight: "inherit" });
var disabledStyles10 = (0, import_deepmerge11.default)(BaseStyles.disabled, { fontSize: "inherit", fontWeight: "inherit" });
var ButtonStyleMapping10 = {
  default: defaultStyles10,
  loading: loadingStyles10,
  disabled: disabledStyles10
};
var ResetButton = React10.forwardRef(
  ({ size, buttonState, fullWidth, ...props }, ref) => {
    return /* @__PURE__ */ React10.createElement(
      import_box10.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        ...ButtonStyleMapping10[buttonState],
        ...SizeStyles[size],
        ...props
      }
    );
  }
);
ResetButton.defaultProps = {
  as: "button"
};
ResetButton.displayName = "ResetButton";

// src/SecondaryButton.tsx
var import_box11 = require("@twilio-paste/box");
var import_deepmerge12 = __toESM(require_cjs());
var React11 = __toESM(require("react"));
var defaultStyles11 = (0, import_deepmerge12.default)(BaseStyles.default, {
  color: "colorText",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeak",
  _hover: {
    color: "colorTextPrimary",
    backgroundColor: "colorBackgroundBody",
    boxShadow: "shadowBorderPrimary"
  },
  _focus: {
    boxShadow: "shadowFocusShadowBorder"
  },
  _active: {
    color: "colorTextPrimaryStrong",
    backgroundColor: "colorBackgroundPrimaryWeakest",
    boxShadow: "shadowBorderPrimaryStrong"
  }
});
var baseLoadingStyles2 = {
  color: "colorTextPrimary",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeak"
};
var loadingStyles11 = (0, import_deepmerge12.default)(BaseStyles.loading, {
  ...baseLoadingStyles2,
  _hover: baseLoadingStyles2,
  _active: baseLoadingStyles2,
  _focus: baseLoadingStyles2
});
var baseDisabledStyles2 = {
  color: "colorTextWeaker",
  backgroundColor: "colorBackgroundBody",
  boxShadow: "shadowBorderWeaker"
};
var disabledStyles11 = (0, import_deepmerge12.default)(BaseStyles.disabled, {
  ...baseDisabledStyles2,
  _hover: baseDisabledStyles2,
  _active: baseDisabledStyles2,
  _focus: baseDisabledStyles2,
  _pressed: {
    backgroundColor: "colorBackgroundStrong",
    color: "colorTextWeak",
    boxShadow: "none"
  },
  _pressed_hover: {
    backgroundColor: "colorBackgroundStrong",
    color: "colorTextWeak",
    boxShadow: "none"
  }
});
var ButtonStyleMapping11 = {
  default: defaultStyles11,
  loading: loadingStyles11,
  disabled: disabledStyles11
};
var SecondaryButton = React11.forwardRef(
  ({ size, buttonState, fullWidth, pressed, ...props }, ref) => {
    const toggleStyles = pressed === void 0 ? {} : ToggleStyles;
    return /* @__PURE__ */ React11.createElement(
      import_box11.Box,
      {
        ref,
        width: fullWidth ? "100%" : "auto",
        "aria-pressed": pressed,
        ...(0, import_box11.safelySpreadBoxProps)(props),
        ...ButtonStyleMapping11[buttonState],
        ...toggleStyles,
        ...SizeStyles[size]
      }
    );
  }
);
SecondaryButton.defaultProps = {
  as: "button"
};
SecondaryButton.displayName = "SecondaryButton";

// src/SecondaryIconButton.tsx
var import_box12 = require("@twilio-paste/box");
var import_deepmerge13 = __toESM(require_cjs());
var React12 = __toESM(require("react"));
var defaultStyles12 = (0, import_deepmerge13.default)(BaseStyles.default, {
  color: "colorTextIcon",
  _hover: {
    color: "colorTextPrimaryStrongest"
  },
  _active: {
    color: "colorTextPrimaryStrongest"
  }
});
var loadingStyles12 = (0, import_deepmerge13.default)(BaseStyles.loading, {
  color: "colorTextPrimary"
});
var disabledStyles12 = (0, import_deepmerge13.default)(BaseStyles.disabled, {
  color: "colorTextWeaker",
  _pressed: {
    backgroundColor: "colorBackgroundStrong",
    color: "colorTextWeak"
  },
  _pressed_hover: {
    backgroundColor: "colorBackgroundStrong",
    color: "colorTextWeak"
  }
});
var ButtonStyleMapping12 = {
  default: defaultStyles12,
  loading: loadingStyles12,
  disabled: disabledStyles12
};
var SecondaryIconButton = React12.forwardRef(
  ({ size, buttonState, fullWidth, pressed, ...props }, ref) => {
    const toggleStyles = pressed === void 0 ? {} : ToggleIconButtonStyles;
    return /* @__PURE__ */ React12.createElement(
      import_box12.Box,
      {
        ref,
        "aria-pressed": pressed,
        width: fullWidth ? "100%" : "auto",
        ...(0, import_box12.safelySpreadBoxProps)(props),
        ...toggleStyles,
        ...ButtonStyleMapping12[buttonState],
        ...SizeStyles[size]
      }
    );
  }
);
SecondaryIconButton.defaultProps = {
  as: "button"
};
SecondaryIconButton.displayName = "SecondaryIconButton";

// src/Button.tsx
var AnimatedBox = (0, import_animation_library.animated)(import_box13.Box);
var getButtonSize = (variant, children, size) => {
  let smartSize = "default";
  if (size != null) {
    smartSize = size;
  } else if (variant === "link" || variant === "destructive_link" || variant === "reset") {
    smartSize = "reset";
  } else if (React13.Children.count(children) === 1) {
    React13.Children.forEach(children, (child) => {
      if (React13.isValidElement(child) && typeof child.type.displayName === "string" && child.type.displayName.includes("Icon")) {
        smartSize = "icon";
      }
    });
  }
  return smartSize;
};
var getButtonState = (disabled, loading) => {
  if (disabled) {
    return "disabled";
  }
  if (loading) {
    return "loading";
  }
  return "default";
};
var handlePropValidation = ({
  as,
  href,
  tabIndex,
  variant,
  size,
  fullWidth,
  children,
  disabled,
  loading,
  pressed
}) => {
  const hasHref = href != null && href !== "";
  const hasTabIndex = tabIndex != null;
  if (as !== "a" && hasHref) {
    throw new Error(`[Paste: Button] You cannot pass href into a button without the 'a' tag.  Use 'as="a"'.`);
  }
  if (as === "a") {
    if (!hasHref) {
      throw new Error("[Paste: Button] Missing href prop for link button.");
    }
    if (variant === "link" || variant === "inverse_link") {
      throw new Error("[Paste: Button] Using Button component as an Anchor. Use the Paste Anchor component instead.");
    }
    if (variant !== "primary" && variant !== "secondary" && variant !== "reset" && variant !== "inverse") {
      throw new Error(`[Paste: Button] <Button as="a"> only works with the following variants: primary and secondary.`);
    }
    if (disabled || loading) {
      throw new Error(`[Paste: Button] <Button as="a"> cannot be disabled or loading.`);
    }
  }
  if (variant === "reset" && size !== "reset") {
    throw new Error('[Paste: Button] The "RESET" variant can only be used with the "RESET" size.');
  }
  if ((size === "icon" || size === "icon_small" || size === "circle" || size === "circle_small") && fullWidth) {
    throw new Error("[Paste: Button] Icon buttons should not be fullWidth.");
  }
  if (children == null) {
    throw new Error("[Paste: Button] Must have non-null children.");
  }
  if (hasTabIndex && !(tabIndex === 0 || tabIndex === -1)) {
    throw new Error("[Paste: Button] tabIndex must be 0 or -1.");
  }
  if (pressed && !(variant === "secondary" || variant === "secondary_icon" || variant === "destructive_secondary")) {
    throw new Error(
      `[Paste: Button] pressed can only be used with "secondary" and "secondary_icon" and "destructive_secondary" variants.`
    );
  }
};
var variantsWithoutBoundingBox = /* @__PURE__ */ new Set(["link", "destructive_link", "inverse_link", "reset"]);
var ButtonContents = ({
  buttonState,
  children,
  showLoading,
  variant
}) => {
  const buttonVariantHasBoundingBox = variant && variantsWithoutBoundingBox.has(variant);
  return /* @__PURE__ */ React13.createElement(React13.Fragment, null, /* @__PURE__ */ React13.createElement(
    import_box13.Box,
    {
      as: "span",
      display: "flex",
      textDecoration: "inherit",
      opacity: buttonState === "loading" ? "0" : "1",
      justifyContent: buttonVariantHasBoundingBox ? null : "center",
      columnGap: "space20",
      alignItems: "center",
      width: "100%"
    },
    children
  ), showLoading ? /* @__PURE__ */ React13.createElement(
    import_box13.Box,
    {
      as: "span",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      lineHeight: "lineHeight30"
    },
    /* @__PURE__ */ React13.createElement(import_spinner.Spinner, { decorative: true, delay: 0 })
  ) : null);
};
ButtonContents.displayName = "ButtonContents";
var getButtonComponent = (variant) => {
  switch (variant) {
    case "primary_icon":
      return PrimaryIconButton;
    case "secondary":
      return SecondaryButton;
    case "secondary_icon":
      return SecondaryIconButton;
    case "destructive":
      return DestructiveButton;
    case "destructive_icon":
      return DestructiveIconButton;
    case "destructive_secondary":
      return DestructiveSecondaryButton;
    case "link":
      return LinkButton;
    case "destructive_link":
      return DestructiveLinkButton;
    case "reset":
      return ResetButton;
    case "inverse":
      return InverseButton;
    case "inverse_link":
      return InverseLinkButton;
    case "primary":
    default:
      return PrimaryButton;
  }
};
var Button = React13.forwardRef(
  ({ element = "BUTTON", i18nExternalLinkLabel = "(link takes you to an external page)", ...props }, ref) => {
    const { size, variant, children, disabled, loading, ...rest } = props;
    const [hovered, setHovered] = React13.useState(false);
    const arrowIconStyles = (0, import_animation_library.useSpring)({
      translateX: hovered ? "4px" : "0px",
      config: {
        mass: 0.1,
        tension: 275,
        friction: 16
      }
    });
    const smartDefaultSize = React13.useMemo(() => {
      return getButtonSize(variant, children, size);
    }, [size, variant, children]);
    handlePropValidation({ ...props, size: smartDefaultSize });
    const buttonState = getButtonState(disabled, loading);
    const showLoading = buttonState === "loading";
    const showDisabled = buttonState !== "default";
    const ButtonComponent = getButtonComponent(variant);
    const externalLinkProps = props.href != null ? (0, import_anchor.secureExternalLink)(props.href) : null;
    let injectIconChildren = children;
    if (props.as === "a" && props.href != null && typeof children === "string" && variant !== "reset") {
      injectIconChildren = /* @__PURE__ */ React13.createElement(React13.Fragment, null, children, externalLinkProps != null ? /* @__PURE__ */ React13.createElement(import_LinkExternalIcon.LinkExternalIcon, { decorative: false, title: i18nExternalLinkLabel }) : /* @__PURE__ */ React13.createElement(AnimatedBox, { style: arrowIconStyles }, /* @__PURE__ */ React13.createElement(import_ArrowForwardIcon.ArrowForwardIcon, { decorative: true })));
    }
    return /* @__PURE__ */ React13.createElement(
      ButtonComponent,
      {
        ...externalLinkProps,
        ...rest,
        onMouseEnter: (event) => {
          if (typeof rest.onMouseEnter === "function") {
            rest.onMouseEnter(event);
          }
          setHovered(true);
        },
        onMouseLeave: (event) => {
          if (typeof rest.onMouseLeave === "function") {
            rest.onMouseLeave(event);
          }
          setHovered(false);
        },
        buttonState,
        disabled: showDisabled,
        element,
        variant,
        size: smartDefaultSize,
        "aria-busy": buttonState === "loading" ? "true" : "false",
        ref
      },
      /* @__PURE__ */ React13.createElement(ButtonContents, { buttonState, showLoading, variant }, injectIconChildren)
    );
  }
);
Button.defaultProps = {
  as: "button",
  fullWidth: false,
  disabled: false,
  loading: false,
  type: "button",
  variant: "primary"
};
Button.displayName = "Button";
