// @twilio-paste/keyboard-key/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  KeyboardKey: () => KeyboardKey,
  KeyboardKeyGroup: () => KeyboardKeyGroup,
  useKeyCombination: () => useKeyCombination,
  useKeyCombinations: () => useKeyCombinations
});
module.exports = __toCommonJS(src_exports);

// src/KeyboardKey.tsx
var import_box = require("@twilio-paste/box");
var React2 = __toESM(require("react"));

// src/KeyboardKeyContext.tsx
var React = __toESM(require("react"));
var KeyboardKeyCombinationContext = React.createContext(
  {}
);

// src/KeyboardKey.tsx
var BaseStyles = {
  default: {
    borderColor: "colorBorderWeak",
    backgroundColor: "colorBackgroundWeak",
    boxShadow: "shadowBorderBottomWeak"
  },
  inverse: {
    borderColor: "colorBorderInverseWeaker",
    backgroundColor: "colorBackgroundInverse",
    color: "colorTextInverse",
    boxShadow: "shadowBorderBottomInverseWeaker"
  }
};
var DisabledStyles = {
  default: {
    color: "colorTextWeak",
    borderColor: "colorBorderWeakest",
    boxShadow: void 0
  },
  inverse: {
    color: "colorTextInverseWeaker",
    borderColor: "colorBorderInverseWeakest",
    boxShadow: void 0
  }
};
var PressedStyles = {
  default: {
    backgroundColor: "colorBackgroundStrong",
    boxShadow: void 0
  },
  inverse: {
    backgroundColor: "colorBackgroundInverseStronger",
    boxShadow: void 0
  }
};
var KeyboardKey = React2.forwardRef(
  ({ element = "KEYBOARD_KEY", keyEvent, ...props }, ref) => {
    const {
      disabled,
      activeKeys,
      enablePressStyles,
      variant = "default"
    } = React2.useContext(KeyboardKeyCombinationContext);
    const isKeyActive = !disabled && activeKeys && keyEvent && activeKeys.map((k) => k.toLowerCase()).includes(keyEvent.toLowerCase());
    return /* @__PURE__ */ React2.createElement(
      import_box.Box,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        element,
        ref,
        borderWidth: "borderWidth10",
        borderRadius: "borderRadius20",
        borderStyle: "solid",
        width: "fit-content",
        minWidth: "sizeBase60",
        display: "inline-flex",
        justifyContent: "center",
        paddingX: "space20",
        as: "kbd",
        fontFamily: "fontFamilyText",
        ...BaseStyles[variant],
        ...disabled ? DisabledStyles[variant] : {},
        ...isKeyActive && enablePressStyles ? PressedStyles[variant] : {}
      },
      props.children
    );
  }
);
KeyboardKey.displayName = "KeyboardKey";

// src/KeyboardKeyGroup.tsx
var import_box2 = require("@twilio-paste/box");
var React3 = __toESM(require("react"));
var KeyboardKeyGroup = React3.forwardRef(
  ({
    element = "KEYBOARD_KEY_GROUP",
    activeKeys,
    disabled = false,
    enablePressStyles = false,
    variant = "default",
    ...props
  }, ref) => {
    return /* @__PURE__ */ React3.createElement(KeyboardKeyCombinationContext.Provider, { value: { activeKeys, disabled, enablePressStyles, variant } }, /* @__PURE__ */ React3.createElement(import_box2.Box, { ...(0, import_box2.safelySpreadBoxProps)(props), element, ref, display: "inline-flex", columnGap: "space20" }, props.children));
  }
);
KeyboardKeyGroup.displayName = "KeyboardKeyGroup";

// src/hooks.ts
var React4 = __toESM(require("react"));
var useKeyEvents = () => {
  const [activeKeys, setActiveKeys] = React4.useState([]);
  const handleKeyDown = (e) => {
    if (!e.repeat) {
      setActiveKeys((prev) => {
        return Array.from(/* @__PURE__ */ new Set([...prev, e.key.toLowerCase()]));
      });
    }
  };
  const handleKeyUp = (e) => {
    if (e.key === "Meta") {
      setActiveKeys([]);
    } else {
      setActiveKeys((prev) => [...prev].filter((k) => k.toLowerCase() !== e.key.toLowerCase()));
    }
  };
  React4.useEffect(() => {
    window.addEventListener("keydown", handleKeyDown);
    window.addEventListener("keyup", handleKeyUp);
    return () => {
      window.removeEventListener("keydown", handleKeyDown);
      window.removeEventListener("keyup", handleKeyUp);
    };
  }, []);
  return { activeKeys };
};
var stringArrayMatches = (arr1, arr2) => JSON.stringify(arr1.sort((a, b) => a.localeCompare(b)).map((s) => s.toLowerCase())) === JSON.stringify(arr2.sort((a, b) => a.localeCompare(b)));
var useKeyCombination = ({
  keys,
  onCombinationPress,
  disabled,
  enablePressStyles
}) => {
  const { activeKeys } = useKeyEvents();
  React4.useEffect(() => {
    const combinationMatch = stringArrayMatches(keys, activeKeys);
    if (combinationMatch && !disabled) {
      onCombinationPress();
    }
  }, [activeKeys]);
  return { activeKeys, disabled, enablePressStyles };
};
var useKeyCombinations = ({ combinations }) => {
  const { activeKeys } = useKeyEvents();
  React4.useEffect(() => {
    const combinationMatch = combinations.find((combos) => stringArrayMatches(combos.keys, activeKeys));
    if (combinationMatch && !combinationMatch.disabled) {
      combinationMatch.onCombinationPress();
    }
  }, [activeKeys, combinations]);
  return { activeKeys };
};
