// @twilio-paste/checkbox/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Checkbox: () => Checkbox,
  CheckboxDisclaimer: () => CheckboxDisclaimer,
  CheckboxGroup: () => CheckboxGroup,
  CheckboxIcon: () => CheckboxIcon,
  HiddenCheckbox: () => HiddenCheckbox
});
module.exports = __toCommonJS(src_exports);

// src/Checkbox.tsx
var import_base_radio_checkbox = require("@twilio-paste/base-radio-checkbox");
var import_box = require("@twilio-paste/box");
var import_CheckboxCheckIcon = require("@twilio-paste/icons/cjs/CheckboxCheckIcon");
var import_MinusIcon = require("@twilio-paste/icons/cjs/MinusIcon");
var import_label = require("@twilio-paste/label");
var import_media_object = require("@twilio-paste/media-object");
var import_uid_library = require("@twilio-paste/uid-library");
var React2 = __toESM(require("react"));

// src/CheckboxContext.tsx
var React = __toESM(require("react"));
var CheckboxContext = React.createContext({
  disabled: false,
  name: void 0,
  hasError: void 0,
  onChange: () => {
  }
});

// src/Checkbox.tsx
var selectAllStyleProps = {
  paddingTop: "space20",
  paddingRight: "space30",
  paddingBottom: "space20",
  paddingLeft: "space30",
  borderRadius: "borderRadius10",
  backgroundColor: "colorBackground"
};
var selectAllActiveStyleProps = {
  ...selectAllStyleProps,
  backgroundColor: "colorBackground"
};
var selectAllChildStyleProps = {
  paddingLeft: "space30",
  paddingRight: "space30"
};
var HiddenCheckbox = React2.forwardRef((props, ref) => /* @__PURE__ */ React2.createElement(
  import_box.Box,
  {
    ...props,
    ref,
    as: "input",
    type: "checkbox",
    size: "size0",
    border: "none",
    overflow: "hidden",
    padding: "space0",
    margin: "space0",
    whiteSpace: "nowrap",
    textTransform: "none",
    position: "absolute",
    clip: "rect(0 0 0 0)"
  }
));
HiddenCheckbox.displayName = "HiddenCheckbox";
var CheckboxIcon = ({ checked, element, indeterminate }) => {
  if (indeterminate) {
    return /* @__PURE__ */ React2.createElement(import_MinusIcon.MinusIcon, { element, decorative: true, color: "inherit", size: "sizeIcon10" });
  }
  return /* @__PURE__ */ React2.createElement(
    import_CheckboxCheckIcon.CheckboxCheckIcon,
    {
      element,
      display: !checked ? "none" : "block",
      decorative: true,
      color: "inherit",
      size: "sizeIcon10"
    }
  );
};
CheckboxIcon.displayName = "CheckboxIcon";
var Checkbox = React2.forwardRef(
  ({
    checked,
    defaultChecked,
    element = "CHECKBOX",
    children,
    helpText,
    id,
    indeterminate,
    isSelectAll,
    isSelectAllChild,
    required,
    onChange,
    ...props
  }, ref) => {
    if (checked != null && defaultChecked != null) {
      throw new Error(
        `[Paste Checkbox] Do not provide both "defaultChecked" and "checked" to Checkbox at the same time. Please consider if you want this component to be controlled or uncontrolled.`
      );
    }
    const [checkedState, setCheckedState] = React2.useState(defaultChecked);
    const checkboxGroupContext = React2.useContext(CheckboxContext);
    const helpTextId = (0, import_uid_library.useUID)();
    const checkboxId = id ? id : (0, import_uid_library.useUID)();
    const isControlled = React2.useMemo(() => checked !== void 0, []);
    const mergedChecked = isControlled ? checked : checkedState;
    const handleChange = React2.useCallback(
      (event) => {
        if (!isControlled) {
          setCheckedState(event.target.checked);
        } else if (onChange) {
          onChange(event);
        } else {
          checkboxGroupContext.onChange(event);
        }
      },
      [onChange, checkboxGroupContext.onChange]
    );
    const disabled = props.disabled != null ? props.disabled : checkboxGroupContext.disabled;
    const name = props.name != null ? props.name : checkboxGroupContext.name;
    const hasError = props.hasError != null ? props.hasError : checkboxGroupContext.hasError;
    let selectAllStyles = {};
    if (isSelectAll) {
      selectAllStyles = !disabled && (mergedChecked || indeterminate) ? selectAllActiveStyleProps : selectAllStyleProps;
    }
    if (isSelectAllChild) {
      selectAllStyles = selectAllChildStyleProps;
    }
    return /* @__PURE__ */ React2.createElement(import_box.Box, { element, display: "inline-flex", position: "relative", flexDirection: "column", ...selectAllStyles }, /* @__PURE__ */ React2.createElement(
      HiddenCheckbox,
      {
        ...(0, import_box.safelySpreadBoxProps)(props),
        checked,
        defaultChecked,
        disabled,
        name,
        onChange: handleChange,
        "aria-describedby": helpTextId,
        "aria-checked": indeterminate ? "mixed" : checked,
        "aria-invalid": hasError,
        id: checkboxId,
        required,
        ref
      }
    ), /* @__PURE__ */ React2.createElement(import_base_radio_checkbox.BaseRadioCheckboxLabel, { disabled, htmlFor: checkboxId }, /* @__PURE__ */ React2.createElement(
      import_base_radio_checkbox.BaseRadioCheckboxControl,
      {
        alignItems: "center",
        borderRadius: "borderRadius20",
        element: `${element}_CONTROL`,
        disabled,
        position: "relative",
        type: "checkbox"
      },
      /* @__PURE__ */ React2.createElement(CheckboxIcon, { element: `${element}_ICON`, checked: mergedChecked, indeterminate })
    ), /* @__PURE__ */ React2.createElement(
      import_base_radio_checkbox.BaseRadioCheckboxLabelText,
      {
        element: `${element}_LABEL_TEXT`,
        fontWeight: isSelectAll ? null : "fontWeightMedium"
      },
      /* @__PURE__ */ React2.createElement(import_media_object.MediaObject, { verticalAlign: "top" }, required && /* @__PURE__ */ React2.createElement(import_media_object.MediaFigure, { spacing: "space20" }, /* @__PURE__ */ React2.createElement(import_label.RequiredDot, null)), /* @__PURE__ */ React2.createElement(import_media_object.MediaBody, null, children))
    )), helpText && /* @__PURE__ */ React2.createElement(import_base_radio_checkbox.BaseRadioCheckboxHelpText, { element: `${element}_HELP_TEXT_WRAPPER`, helpTextId }, helpText));
  }
);
Checkbox.displayName = "Checkbox";

// src/CheckboxGroup.tsx
var import_inline_control_group = require("@twilio-paste/inline-control-group");
var React3 = __toESM(require("react"));
var CheckboxGroup = React3.forwardRef(
  ({
    children,
    element = "CHECKBOX_GROUP",
    disabled = false,
    errorText,
    isSelectAll = false,
    name,
    onChange,
    orientation = "vertical",
    i18nRequiredLabel = "(required)",
    ...props
  }, ref) => {
    const onChangeHandler = React3.useCallback(
      (event) => {
        if (onChange != null) {
          onChange(event.target.checked);
        }
      },
      [onChange]
    );
    const contextValue = React3.useMemo(() => {
      return {
        disabled,
        name,
        onChange: onChangeHandler,
        hasError: errorText ? true : void 0
      };
    }, [disabled, name, onChangeHandler]);
    return /* @__PURE__ */ React3.createElement(CheckboxContext.Provider, { value: contextValue }, /* @__PURE__ */ React3.createElement(
      import_inline_control_group.InlineControlGroup,
      {
        ...props,
        element,
        disabled,
        errorText,
        name,
        orientation,
        ref,
        i18nRequiredLabel
      },
      React3.Children.map(children, (child, index) => {
        return React3.isValidElement(child) ? React3.cloneElement(child, {
          isSelectAll: isSelectAll && index === 0,
          isSelectAllChild: isSelectAll && orientation === "vertical" && index !== 0
        }) : child;
      })
    ));
  }
);
CheckboxGroup.displayName = "CheckboxGroup";

// src/CheckboxDisclaimer.tsx
var import_box2 = require("@twilio-paste/box");
var import_help_text = require("@twilio-paste/help-text");
var React4 = __toESM(require("react"));
var CheckboxDisclaimer = React4.forwardRef(
  ({ children, element = "CHECKBOX_DISCLAIMER", errorText, ...props }, ref) => {
    return /* @__PURE__ */ React4.createElement(React4.Fragment, null, /* @__PURE__ */ React4.createElement(import_box2.Box, { element, backgroundColor: "colorBackground", borderRadius: "borderRadius20", padding: "space50" }, /* @__PURE__ */ React4.createElement(Checkbox, { ...props, hasError: errorText != null, ref }, children)), errorText && /* @__PURE__ */ React4.createElement(import_box2.Box, { marginTop: "space30", element: `${element}_ERROR_TEXT_WRAPPER`, marginLeft: "space50" }, /* @__PURE__ */ React4.createElement(import_help_text.HelpText, { variant: "error" }, errorText)));
  }
);
CheckboxDisclaimer.displayName = "CheckboxDisclaimer";
