// @twilio-paste/timeline/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Timeline: () => Timeline,
  TimelineItem: () => TimelineItem,
  TimelineItemGroup: () => TimelineItemGroup
});
module.exports = __toCommonJS(src_exports);

// src/Timeline.tsx
var import_box = require("@twilio-paste/box");
var import_styling_library = require("@twilio-paste/styling-library");
var import_react = __toESM(require("react"));
var Timeline = import_react.default.forwardRef(
  ({ children, element = "TIMELINE", ...props }, ref) => {
    const ContainerStyled = import_styling_library.styled.ol(
      (0, import_styling_library.css)({
        listStyleType: "none",
        margin: "0",
        padding: "0",
        "li>div:first-child::after": {
          content: "''",
          borderLeftWidth: "borderWidth10",
          borderLeftStyle: "solid",
          borderLeftColor: "colorBorderWeaker",
          minHeight: "sizeBase80",
          flexGrow: "1"
        }
      })
    );
    return /* @__PURE__ */ import_react.default.createElement(
      import_box.Box,
      {
        ref,
        as: ContainerStyled,
        element,
        ...props
      },
      children
    );
  }
);
Timeline.displayName = "Timeline";

// src/TimelineItem.tsx
var import_box3 = require("@twilio-paste/box");
var import_styling_library2 = require("@twilio-paste/styling-library");
var import_react2 = __toESM(require("react"));

// src/TimelineContext.tsx
var React2 = __toESM(require("react"));
var TimelineGroupContext = React2.createContext(false);

// src/TimelineItemCollapsible.tsx
var import_summary_detail = require("@twilio-paste/summary-detail");
var import_text = require("@twilio-paste/text");
var React3 = __toESM(require("react"));
var TimelineItemCollapsible = React3.forwardRef(
  ({ children, timestamp, element }, ref) => {
    return /* @__PURE__ */ React3.createElement(import_summary_detail.SummaryDetail, { element: `${element}_SUMMARY_DETAIL`, ref }, /* @__PURE__ */ React3.createElement(import_summary_detail.SummaryDetailHeading, { element: `${element}_SUMMARY_DETAIL_HEADING` }, /* @__PURE__ */ React3.createElement(import_summary_detail.SummaryDetailToggleButton, { element: `${element}_SUMMARY_DETAIL_TOGGLE_BUTTON` }), /* @__PURE__ */ React3.createElement(import_summary_detail.SummaryDetailHeadingContent, { element: `${element}_SUMMARY_DETAIL_HEADING_CONTENT` }, /* @__PURE__ */ React3.createElement(
      import_text.Text,
      {
        element: `${element}_TIMESTAMP`,
        as: "span",
        fontSize: "fontSize20",
        color: "colorTextWeak",
        fontWeight: "fontWeightMedium",
        lineHeight: "lineHeight20",
        marginLeft: "spaceNegative30"
      },
      timestamp
    ))), /* @__PURE__ */ React3.createElement(import_summary_detail.SummaryDetailContent, { element: `${element}_SUMMARY_DETAIL_CONTENT` }, children));
  }
);
TimelineItemCollapsible.displayName = "TimelineItemCollapsible";

// src/TimelineItemIcon.tsx
var import_box2 = require("@twilio-paste/box");
var import_IconWrapper = require("@twilio-paste/icons/cjs/helpers/IconWrapper");
var React4 = __toESM(require("react"));
var TimelineItemIcon = React4.forwardRef(
  ({ icon: Icon, as, display, element = "TIMELINE_ITEM_ICON", color }, ref) => {
    return /* @__PURE__ */ React4.createElement(import_box2.Box, { element: `${element}_WRAPPER` }, Icon ? /* @__PURE__ */ React4.createElement(import_box2.Box, { marginY: "space20" }, /* @__PURE__ */ React4.createElement(import_IconWrapper.IconWrapper, { as, display, element, color, ref, size: "sizeIcon10" }, /* @__PURE__ */ React4.createElement(Icon, { decorative: true, size: "sizeIcon10" }))) : /* @__PURE__ */ React4.createElement(
      import_box2.Box,
      {
        element: `${element}_DOT`,
        backgroundColor: "colorBackgroundStronger",
        borderRadius: "borderRadiusCircle",
        size: "sizeBase20",
        marginY: "space30"
      }
    ));
  }
);
TimelineItemIcon.displayName = "TimelineItemIcon";

// src/TimelineItem.tsx
var TimelineItem = import_react2.default.forwardRef(
  ({ children, icon, timestamp, title, collapsible = false, collapsibleHeading, element = "TIMELINE_ITEM", ...props }, ref) => {
    const isGrouped = import_react2.default.useContext(TimelineGroupContext);
    const ContainerStyled = import_styling_library2.styled.li(
      (0, import_styling_library2.css)({
        "&:last-child>div:first-child::after": {
          content: "none"
        },
        "div::after": {
          display: isGrouped ? "none" : "block"
        }
      })
    );
    return /* @__PURE__ */ import_react2.default.createElement(
      import_box3.Box,
      {
        element,
        as: ContainerStyled,
        ref,
        position: "relative",
        display: "flex",
        columnGap: "space50",
        rowGap: "space50",
        flexShrink: 0,
        ...props
      },
      !isGrouped ? /* @__PURE__ */ import_react2.default.createElement(
        import_box3.Box,
        {
          element: `${element}_SEPARATOR`,
          display: "flex",
          width: "sizeBase50",
          flexDirection: "column",
          alignItems: "center",
          flexShrink: 0
        },
        /* @__PURE__ */ import_react2.default.createElement(TimelineItemIcon, { icon, element: `${element}_ICON` })
      ) : null,
      /* @__PURE__ */ import_react2.default.createElement(
        import_box3.Box,
        {
          element: `${element}_CONTENT_WRAPPER`,
          paddingX: "space0",
          fontFamily: "fontFamilyText",
          fontSize: "fontSize20",
          lineHeight: "lineHeight20",
          fontWeight: "fontWeightMedium",
          display: "flex",
          flexDirection: "column",
          columnGap: "space10",
          paddingBottom: "space60"
        },
        /* @__PURE__ */ import_react2.default.createElement(
          import_box3.Box,
          {
            element: `${element}_TITLE`,
            as: "span",
            color: "colorText",
            paddingY: "space10",
            fontWeight: "fontWeightSemibold",
            lineHeight: "lineHeight20",
            fontSize: "fontSize30",
            letterSpacing: "-0.28px"
          },
          title
        ),
        collapsible ? /* @__PURE__ */ import_react2.default.createElement(TimelineItemCollapsible, { element, timestamp: timestamp ? timestamp : collapsibleHeading }, children) : /* @__PURE__ */ import_react2.default.createElement(import_react2.default.Fragment, null, timestamp ? /* @__PURE__ */ import_react2.default.createElement(
          import_box3.Box,
          {
            element: `${element}_TIMESTAMP`,
            as: "span",
            color: "colorTextWeak",
            fontWeight: "fontWeightMedium",
            lineHeight: "lineHeight20",
            fontSize: "fontSize20"
          },
          timestamp
        ) : null, /* @__PURE__ */ import_react2.default.createElement(import_box3.Box, { element: `${element}_CONTENT` }, children))
      )
    );
  }
);
TimelineItem.displayName = "TimelineItem";

// src/TimelineItemGroup.tsx
var import_box4 = require("@twilio-paste/box");
var import_detail_text = require("@twilio-paste/detail-text");
var React6 = __toESM(require("react"));
var TimelineItemGroup = React6.forwardRef(
  ({ children, timestamp, icon, element = "TIMELINE_ITEM_GROUP", ...props }, ref) => {
    return /* @__PURE__ */ React6.createElement(
      import_box4.Box,
      {
        element,
        ref,
        as: "li",
        position: "relative",
        display: "flex",
        columnGap: "space50",
        rowGap: "space50",
        flexShrink: 0,
        ...props
      },
      /* @__PURE__ */ React6.createElement(
        import_box4.Box,
        {
          element: `${element}_SEPARATOR`,
          display: "flex",
          width: "sizeBase50",
          flexDirection: "column",
          alignItems: "center"
        },
        /* @__PURE__ */ React6.createElement(TimelineItemIcon, { icon })
      ),
      /* @__PURE__ */ React6.createElement(import_box4.Box, { element: `${element}_CONTENT_WRAPPER` }, /* @__PURE__ */ React6.createElement(import_box4.Box, { element: `${element}_TIMESTAMP`, marginTop: "space10", marginBottom: "space30" }, /* @__PURE__ */ React6.createElement(import_detail_text.DetailText, { element: `${element}_TIMESTAMP_DETAIL_TEXT`, marginTop: "space0" }, timestamp)), /* @__PURE__ */ React6.createElement(import_box4.Box, { element: `${element}_CONTENT`, as: "ol", margin: "space0", padding: "space0" }, /* @__PURE__ */ React6.createElement(TimelineGroupContext.Provider, { value: true }, children)))
    );
  }
);
TimelineItemGroup.displayName = "TimelineItemGroup";
