// @twilio-paste/file-picker/dist/index.debug.es.js
// src/FilePicker.tsx
import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
import { SiblingBox } from "@twilio-paste/sibling-box";
import { Text } from "@twilio-paste/text";
import { useUID } from "@twilio-paste/uid-library";
import * as React from "react";
var getTextColor = (disabled, fileDescription, i18nNoSelectionText) => {
  const noFileUploaded = fileDescription === i18nNoSelectionText;
  if (disabled)
    return "colorTextWeaker";
  if (noFileUploaded)
    return "colorTextWeak";
  return "colorText";
};
var FilePicker = React.forwardRef(
  ({
    element = "FILEPICKER",
    accept,
    id = useUID(),
    children,
    disabled = false,
    i18nNoSelectionText = "No file uploaded",
    required = false,
    onChange,
    ...props
  }, ref) => {
    const [fileDescription, setFileDescription] = React.useState(i18nNoSelectionText);
    const textId = useUID();
    const handleChange = (evt) => {
      if (onChange)
        onChange(evt);
      if (evt.currentTarget.files && evt.currentTarget.files.length > 0) {
        const file = evt.currentTarget.files[0].name;
        setFileDescription(file);
      } else {
        setFileDescription(i18nNoSelectionText);
      }
    };
    return /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(
      Box,
      {
        ...safelySpreadBoxProps(props),
        id,
        ref,
        as: "input",
        type: "file",
        accept,
        "aria-disabled": disabled,
        disabled,
        "aria-required": required,
        "aria-describedby": textId,
        size: "size0",
        border: "none",
        overflow: "hidden",
        padding: "space0",
        margin: "space0",
        whiteSpace: "nowrap",
        textTransform: "none",
        position: "absolute",
        clip: "rect(0 0 0 0)",
        onChange: handleChange
      }
    ), /* @__PURE__ */ React.createElement("label", { htmlFor: id }, /* @__PURE__ */ React.createElement(
      SiblingBox,
      {
        element,
        width: "100%",
        display: "flex",
        justifyContent: "start",
        as: "span",
        type: "file",
        borderRadius: "borderRadius30",
        padding: "space20",
        boxShadow: "shadowBorder",
        backgroundColor: "colorBackgroundBody",
        _focusSibling: {
          borderRadius: "borderRadius30",
          padding: "space20",
          boxShadow: "shadowFocus",
          backgroundColor: "colorBackgroundBody"
        },
        _disabledSibling: {
          backgroundColor: "colorBackground",
          boxShadow: "shadowBorderWeaker"
        }
      },
      React.cloneElement(children, { disabled, element: `${element}_BUTTON` }),
      /* @__PURE__ */ React.createElement(
        Text,
        {
          id: textId,
          as: "span",
          paddingX: "space30",
          color: getTextColor(disabled, fileDescription, i18nNoSelectionText),
          fontWeight: fileDescription === i18nNoSelectionText ? "fontWeightNormal" : "fontWeightMedium",
          fontStyle: fileDescription === i18nNoSelectionText ? "italic" : "none",
          element: `${element}_TEXT`
        },
        fileDescription
      )
    )));
  }
);
FilePicker.displayName = "FilePicker";

// src/FilePickerButton.tsx
import { Box as Box2 } from "@twilio-paste/box";
import { Button } from "@twilio-paste/button";
import * as React2 from "react";
var FilePickerButton = React2.forwardRef(
  ({ children, element, disabled, ...props }, ref) => {
    return /* @__PURE__ */ React2.createElement(Box2, { whiteSpace: "nowrap" }, /* @__PURE__ */ React2.createElement(Button, { ...props, element, ref, size: "default", disabled, as: "span", type: "button" }, children));
  }
);
FilePickerButton.displayName = "FilePickerButton";
export {
  FilePicker,
  FilePickerButton
};
