// @twilio-paste/toast/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Toast: () => Toast,
  ToastContainer: () => ToastContainer,
  Toaster: () => Toaster,
  useToaster: () => useToaster
});
module.exports = __toCommonJS(src_exports);

// src/Toast.tsx
var import_button = require("@twilio-paste/button");
var import_CloseIcon = require("@twilio-paste/icons/cjs/CloseIcon");
var import_ErrorIcon = require("@twilio-paste/icons/cjs/ErrorIcon");
var import_NeutralIcon = require("@twilio-paste/icons/cjs/NeutralIcon");
var import_SuccessIcon = require("@twilio-paste/icons/cjs/SuccessIcon");
var import_WarningIcon = require("@twilio-paste/icons/cjs/WarningIcon");
var import_media_object = require("@twilio-paste/media-object");
var import_screen_reader_only = require("@twilio-paste/screen-reader-only");
var React5 = __toESM(require("react"));

// src/ErrorToast.tsx
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));
var ErrorToast = React.forwardRef(({ element = "TOAST", variant, ...props }, ref) => {
  return /* @__PURE__ */ React.createElement(
    import_box.Box,
    {
      ...(0, import_box.safelySpreadBoxProps)(props),
      ref,
      color: "colorTextError",
      width: ["100%", "size40", "size40"],
      backgroundColor: "colorBackgroundErrorWeakest",
      borderColor: "colorBorderErrorWeak",
      borderRadius: "borderRadius30",
      borderStyle: "solid",
      borderWidth: "borderWidth10",
      boxShadow: "shadowHigh",
      element,
      padding: "space60",
      variant
    },
    props.children
  );
});
ErrorToast.displayName = "ErrorToast";

// src/NeutralToast.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var NeutralToast = React2.forwardRef(({ element = "TOAST", variant, ...props }, ref) => {
  return /* @__PURE__ */ React2.createElement(
    import_box2.Box,
    {
      ...(0, import_box2.safelySpreadBoxProps)(props),
      ref,
      color: "colorTextNeutral",
      width: ["100%", "size40", "size40"],
      backgroundColor: "colorBackgroundNeutralWeakest",
      borderColor: "colorBorderNeutralWeak",
      borderRadius: "borderRadius30",
      borderStyle: "solid",
      borderWidth: "borderWidth10",
      boxShadow: "shadowHigh",
      element,
      padding: "space60",
      variant
    },
    props.children
  );
});
NeutralToast.displayName = "NeutralToast";

// src/SuccessToast.tsx
var import_box3 = require("@twilio-paste/box");
var React3 = __toESM(require("react"));
var SuccessToast = React3.forwardRef(({ element = "TOAST", variant, ...props }, ref) => {
  return /* @__PURE__ */ React3.createElement(
    import_box3.Box,
    {
      ...(0, import_box3.safelySpreadBoxProps)(props),
      ref,
      color: "colorTextSuccess",
      width: ["100%", "size40", "size40"],
      backgroundColor: "colorBackgroundSuccessWeakest",
      borderColor: "colorBorderSuccessWeak",
      borderRadius: "borderRadius30",
      borderStyle: "solid",
      borderWidth: "borderWidth10",
      boxShadow: "shadowHigh",
      element,
      padding: "space60",
      variant
    },
    props.children
  );
});
SuccessToast.displayName = "SuccessToast";

// src/WarningToast.tsx
var import_box4 = require("@twilio-paste/box");
var React4 = __toESM(require("react"));
var WarningToast = React4.forwardRef(({ element = "TOAST", variant, ...props }, ref) => {
  return /* @__PURE__ */ React4.createElement(
    import_box4.Box,
    {
      ...(0, import_box4.safelySpreadBoxProps)(props),
      ref,
      color: "colorTextWarningStrong",
      width: ["100%", "size40", "size40"],
      backgroundColor: "colorBackgroundWarningWeakest",
      borderColor: "colorBorderWarningWeak",
      borderRadius: "borderRadius30",
      borderStyle: "solid",
      borderWidth: "borderWidth10",
      boxShadow: "shadowHigh",
      element,
      padding: "space60",
      variant
    },
    props.children
  );
});
WarningToast.displayName = "WarningToast";

// src/constants.ts
var ToastVariantObject = {
  ERROR: "error",
  NEUTRAL: "neutral",
  SUCCESS: "success",
  WARNING: "warning"
};

// src/Toast.tsx
var ToastComponentVariants = {
  error: ErrorToast,
  neutral: NeutralToast,
  success: SuccessToast,
  warning: WarningToast
};
var renderToastIcon = (variant, title, element) => {
  switch (variant) {
    case ToastVariantObject.ERROR:
      return /* @__PURE__ */ React5.createElement(
        import_ErrorIcon.ErrorIcon,
        {
          color: "colorTextIconError",
          decorative: false,
          title,
          element: `${element}_ICON`,
          size: "sizeIcon20"
        }
      );
    case ToastVariantObject.SUCCESS:
      return /* @__PURE__ */ React5.createElement(
        import_SuccessIcon.SuccessIcon,
        {
          color: "colorTextIconSuccess",
          decorative: false,
          title,
          element: `${element}_ICON`,
          size: "sizeIcon20"
        }
      );
    case ToastVariantObject.WARNING:
      return /* @__PURE__ */ React5.createElement(
        import_WarningIcon.WarningIcon,
        {
          color: "colorTextIconWarning",
          decorative: false,
          title,
          element: `${element}_ICON`,
          size: "sizeIcon20"
        }
      );
    case ToastVariantObject.NEUTRAL:
    default:
      return /* @__PURE__ */ React5.createElement(
        import_NeutralIcon.NeutralIcon,
        {
          color: "colorTextIconNeutral",
          decorative: false,
          title,
          element: `${element}_ICON`,
          size: "sizeIcon20"
        }
      );
  }
};
var Toast = React5.forwardRef(
  ({
    children,
    onDismiss,
    variant = "neutral",
    element = "TOAST",
    setFocus,
    i18nDismissLabel = "Dismiss toast",
    i18nErrorLabel = "(error)",
    i18nNeutralLabel = "(information)",
    i18nSuccessLabel = "(success)",
    i18nWarningLabel = "(warning)",
    ...props
  }, ref) => {
    const ToastComponent = ToastComponentVariants[variant];
    const buttonRef = React5.useRef(null);
    React5.useEffect(() => {
      if (setFocus && buttonRef.current) {
        buttonRef.current.focus({ preventScroll: true });
      }
    }, [setFocus]);
    const i18nVariants = {
      error: i18nErrorLabel,
      neutral: i18nNeutralLabel,
      success: i18nSuccessLabel,
      warning: i18nWarningLabel
    };
    return /* @__PURE__ */ React5.createElement(ToastComponent, { role: "status", variant, element, ref, ...props }, /* @__PURE__ */ React5.createElement(import_media_object.MediaObject, { as: "div" }, /* @__PURE__ */ React5.createElement(import_media_object.MediaFigure, { as: "div", spacing: "space60" }, renderToastIcon(variant, i18nVariants[variant], element)), /* @__PURE__ */ React5.createElement(import_media_object.MediaBody, { as: "div" }, children), onDismiss && typeof onDismiss === "function" && /* @__PURE__ */ React5.createElement(import_media_object.MediaFigure, { align: "end", spacing: "space40" }, /* @__PURE__ */ React5.createElement(
      import_button.Button,
      {
        onClick: onDismiss,
        variant: "secondary_icon",
        ref: buttonRef,
        size: "reset",
        element: `${element}_CLOSE_BUTTON`
      },
      /* @__PURE__ */ React5.createElement(import_CloseIcon.CloseIcon, { decorative: true, size: "sizeIcon20", element: `${element}_CLOSE_ICON` }),
      /* @__PURE__ */ React5.createElement(import_screen_reader_only.ScreenReaderOnly, null, i18nDismissLabel)
    ))));
  }
);
Toast.displayName = "Toast";

// src/ToastContainer.tsx
var import_stack = require("@twilio-paste/stack");
var React7 = __toESM(require("react"));

// src/ToastPortal.tsx
var import_box5 = require("@twilio-paste/box");
var import_reakit_library = require("@twilio-paste/reakit-library");
var import_theme = require("@twilio-paste/theme");
var React6 = __toESM(require("react"));
var ToastPortal = React6.forwardRef(({ children, left, ...props }, ref) => {
  return /* @__PURE__ */ React6.createElement(import_reakit_library.Portal, null, /* @__PURE__ */ React6.createElement(import_theme.StyledBase, null, /* @__PURE__ */ React6.createElement(
    import_box5.Box,
    {
      ...(0, import_box5.safelySpreadBoxProps)(props),
      position: "fixed",
      right: "space40",
      top: "space90",
      left,
      zIndex: "zIndex90",
      ref
    },
    children
  )));
});
ToastPortal.displayName = "ToastPortal";

// src/ToastContainer.tsx
var ToastContainer = React7.forwardRef(({ children, ...props }, ref) => {
  return /* @__PURE__ */ React7.createElement(ToastPortal, { ...props }, /* @__PURE__ */ React7.createElement(import_stack.Stack, { orientation: "vertical", spacing: "space40", ref }, children));
});
ToastContainer.displayName = "ToastContainer";

// src/Toaster.tsx
var import_animation_library = require("@twilio-paste/animation-library");
var import_box6 = require("@twilio-paste/box");
var import_theme2 = require("@twilio-paste/theme");
var React8 = __toESM(require("react"));
var AnimatedToast = (0, import_animation_library.animated)(import_box6.Box);
var Toaster = ({ toasts, pop, ...props }) => {
  const prefersReducedMotion = (0, import_animation_library.useReducedMotion)();
  const [refMap] = React8.useState(() => /* @__PURE__ */ new WeakMap());
  const [returnTarget, setReturnTarget] = React8.useState({ trigger: null });
  const theme = (0, import_theme2.useTheme)();
  const transitions = (0, import_animation_library.useTransition)(toasts, {
    from: {
      marginBottom: "0rem",
      opacity: 0,
      transform: "translateX(100%) scale(1)",
      height: prefersReducedMotion ? "auto" : 0
    },
    enter: prefersReducedMotion ? { marginBottom: theme.space.space40, opacity: 1, transform: "translateX(0px) scale(1)" } : (item) => async (next) => {
      await next({
        marginBottom: theme.space.space40,
        opacity: 1,
        transform: "translateX(0px) scale(1)",
        height: refMap.get(item).offsetHeight
      });
    },
    leave: {
      marginBottom: "0rem",
      opacity: 0,
      transform: "translateX(0px) scale(0.8)",
      height: prefersReducedMotion ? "auto" : 0
    },
    config: {
      mass: 1,
      tension: 150,
      friction: 20
    }
  });
  const clearReturnTarget = React8.useCallback(() => {
    setReturnTarget({ trigger: null });
  }, [returnTarget]);
  React8.useEffect(() => {
    if (returnTarget.trigger != null && toasts.length === 0) {
      returnTarget.trigger.focus();
      clearReturnTarget();
    }
  }, [toasts]);
  return /* @__PURE__ */ React8.createElement(ToastPortal, { ...props }, transitions((styles, item, state, index) => {
    return /* @__PURE__ */ React8.createElement(AnimatedToast, { as: "div", key: state.key, style: styles }, /* @__PURE__ */ React8.createElement(
      Toast,
      {
        key: item.id,
        ...item,
        onDismiss: () => {
          if (item.onDismiss != null) {
            item.onDismiss();
          }
          pop(item.id);
        },
        ref: (ref) => {
          if (ref != null) {
            refMap.set(item, ref);
          }
        },
        setFocus: index === 0,
        onFocus: (e) => {
          if (e.relatedTarget && !returnTarget.trigger) {
            setReturnTarget({ trigger: e.relatedTarget });
          }
        }
      },
      item.message
    ));
  }));
};
Toaster.displayName = "Toaster";

// src/useToaster.ts
var import_uid_library = require("@twilio-paste/uid-library");
var React9 = __toESM(require("react"));
var useToaster = () => {
  const isMounted = React9.useRef(null);
  const [toasts, setToasts] = React9.useState([]);
  React9.useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
      toasts.forEach((toast) => {
        if (toast.timeOutId) {
          window.clearTimeout(toast.timeOutId);
        }
      });
    };
  }, []);
  const pop = (id) => {
    if (!isMounted.current) {
      return;
    }
    setToasts(
      (currentToasts) => currentToasts.filter((toast) => {
        if (toast.id === id) {
          if (toast.timeOutId) {
            window.clearTimeout(toast.timeOutId);
          }
          if (toast.onDismiss) {
            toast.onDismiss();
          }
        }
        return toast.id !== id;
      })
    );
  };
  const push = (newToast) => {
    if (!isMounted.current) {
      return;
    }
    const generatedID = (0, import_uid_library.uid)(newToast);
    let timeOutId;
    if (newToast.dismissAfter != null && Number.isInteger(newToast.dismissAfter)) {
      timeOutId = window.setTimeout(pop, newToast.dismissAfter, newToast.id || generatedID);
    }
    setToasts((state) => {
      const existingToasts = state.map((toast) => {
        const tmpToast = toast;
        tmpToast.setFocus = false;
        return tmpToast;
      });
      return [{ id: generatedID, timeOutId, setFocus: true, ...newToast }, ...existingToasts];
    });
  };
  return { toasts, push, pop };
};
