// @twilio-paste/radio-button-group/dist/index.debug.es.js
// src/RadioButton.tsx
import { Box as Box2, safelySpreadBoxProps } from "@twilio-paste/box";
import { Button } from "@twilio-paste/button";
import { useUID } from "@twilio-paste/uid-library";
import * as React3 from "react";

// src/HiddenRadio.tsx
import { Box } from "@twilio-paste/box";
import * as React from "react";
var HiddenRadio = React.forwardRef((props, ref) => /* @__PURE__ */ React.createElement(
  Box,
  {
    as: "input",
    type: "radio",
    size: "size0",
    border: "none",
    overflow: "hidden",
    padding: "space0",
    margin: "space0",
    whiteSpace: "nowrap",
    textTransform: "none",
    position: "absolute",
    clip: "rect(0 0 0 0)",
    ref,
    ...props
  }
));
HiddenRadio.displayName = "HiddenRadio";

// src/RadioButtonContext.tsx
import * as React2 from "react";
var RadioButtonContext = React2.createContext({
  name: "",
  value: "",
  disabled: false,
  hasError: false,
  onChange: () => {
  },
  attached: false
});

// src/StyledRadioButtonLabel.tsx
import { ButtonToggleStyles, DestructiveSecondaryButtonToggleStyles } from "@twilio-paste/button";
import { css, styled } from "@twilio-paste/styling-library";
var StyledRadioButtonLabel = styled.label(
  css({
    "& > input[type=radio]:checked + span": {
      ...ButtonToggleStyles._pressed,
      zIndex: 1
    },
    "& > input[type=radio]:checked + span:hover": {
      ...ButtonToggleStyles._pressed_hover,
      zIndex: 2
    },
    "& > input[type=radio]:checked:focus + span": {
      ...ButtonToggleStyles._pressed_focus,
      zIndex: 3
    },
    "& > input[type=radio][aria-invalid=true]:checked + span": {
      ...DestructiveSecondaryButtonToggleStyles._pressed,
      zIndex: 1
    },
    "& > input[type=radio][aria-invalid=true]:checked + span:hover": {
      ...DestructiveSecondaryButtonToggleStyles._pressed_hover,
      zIndex: 2
    },
    "& > input[type=radio][aria-invalid=true]:checked:focus + span": {
      ...DestructiveSecondaryButtonToggleStyles._pressed_focus,
      zIndex: 3
    },
    "& > input[type=radio]:checked:disabled + span": {
      ...ButtonToggleStyles._pressed_disabled,
      zIndex: 0
    },
    "& > input[type=radio][aria-invalid=true]:checked:disabled + span:hover": {
      ...DestructiveSecondaryButtonToggleStyles._pressed_disabled,
      zIndex: 0
    },
    "& > input[type=radio]:disabled + span": {
      ...ButtonToggleStyles._disabled,
      zIndex: 0
    },
    "& > input[type=radio][aria-invalid=true]:disabled + span:hover": {
      ...DestructiveSecondaryButtonToggleStyles._disabled,
      zIndex: 0
    }
  })
);

// src/RadioButton.tsx
var RadioButton = React3.forwardRef(
  ({
    id,
    name,
    element = "RADIO_BUTTON",
    value,
    checked,
    defaultChecked,
    disabled,
    hasError,
    onChange,
    children,
    ...props
  }, ref) => {
    if (checked != null && defaultChecked != null) {
      throw new Error(
        `[Paste RadioButton] Do not provide both 'defaultChecked' and 'checked' to RadioButton at the same time. Please consider if you want this component to be controlled or uncontrolled.`
      );
    }
    const radioGroupContext = React3.useContext(RadioButtonContext);
    const radioId = id ? id : useUID();
    const isControlled = React3.useMemo(() => checked !== void 0 || radioGroupContext.value !== "", []);
    const handleChange = React3.useCallback(
      (event) => {
        if (onChange) {
          onChange(event);
        } else {
          radioGroupContext.onChange(event);
        }
      },
      [onChange, radioGroupContext.onChange]
    );
    const state = {
      name: name != null ? name : radioGroupContext.name,
      disabled: disabled != null ? disabled : radioGroupContext.disabled,
      hasError: hasError != null ? hasError : radioGroupContext.hasError
    };
    if (isControlled) {
      if (radioGroupContext.value !== "") {
        state.checked = radioGroupContext.value === value;
      } else {
        state.checked = checked;
      }
    } else {
      state.defaultChecked = defaultChecked;
    }
    return /* @__PURE__ */ React3.createElement(
      Box2,
      {
        element,
        as: StyledRadioButtonLabel
      },
      /* @__PURE__ */ React3.createElement(
        HiddenRadio,
        {
          ...safelySpreadBoxProps(props),
          ...state,
          value,
          "aria-invalid": state.hasError,
          onChange: handleChange,
          id: radioId,
          ref
        }
      ),
      /* @__PURE__ */ React3.createElement(Button, { variant: state.hasError ? "destructive_secondary" : "secondary", as: "span", disabled: state.disabled }, children)
    );
  }
);
RadioButton.displayName = "RadioButton";

// src/RadioButtonGroup.tsx
import { Box as Box3, safelySpreadBoxProps as safelySpreadBoxProps2 } from "@twilio-paste/box";
import { HelpText } from "@twilio-paste/help-text";
import { Label } from "@twilio-paste/label";
import * as React4 from "react";

// src/StyledRadioButtonGroupWrapper.tsx
import { css as css2, styled as styled2 } from "@twilio-paste/styling-library";
var AttachedRadioButtonGroupWrapper = styled2.span(
  css2({
    "& > label:focus-within": {
      zIndex: 2,
      position: "relative"
    },
    "& > label:first-of-type": {
      "& > span": {
        borderTopRightRadius: 0,
        borderBottomRightRadius: 0
      }
    },
    "& > label:not(:first-of-type):not(:last-of-type)": {
      "& > span": {
        borderRadius: 0,
        marginLeft: "1px"
      }
    },
    "& > label:last-of-type": {
      "& > span": {
        marginLeft: "1px",
        borderTopLeftRadius: 0,
        borderBottomLeftRadius: 0
      }
    }
  })
);
var UnattachedRadioButtonGroupWrapper = styled2.span(
  css2({
    "& > label:first-of-type": {
      marginLeft: "space0"
    },
    "& > label:not(:first-of-type)": {
      marginLeft: "space40"
    }
  })
);

// src/RadioButtonGroup.tsx
var RadioButtonGroup = React4.forwardRef(
  ({
    name,
    element = "RADIO_BUTTON_GROUP",
    value,
    onChange,
    disabled = false,
    errorText,
    children,
    i18nRequiredLabel = "(required)",
    helpText,
    legend,
    required,
    attached = false,
    ...props
  }, ref) => {
    const onChangeHandler = React4.useCallback(
      (event) => {
        if (onChange != null) {
          onChange(event.target.value);
        }
      },
      [onChange]
    );
    const contextValue = React4.useMemo(() => {
      return {
        name,
        value: value || "",
        disabled,
        hasError: errorText != null,
        onChange: onChangeHandler
      };
    }, [name, value, disabled, errorText, onChangeHandler]);
    return /* @__PURE__ */ React4.createElement(RadioButtonContext.Provider, { value: contextValue }, /* @__PURE__ */ React4.createElement(
      Box3,
      {
        as: "fieldset",
        element,
        margin: "space0",
        padding: "space0",
        border: "none",
        display: "flex",
        flexDirection: "column",
        ref,
        ...safelySpreadBoxProps2(props)
      },
      /* @__PURE__ */ React4.createElement(
        Label,
        {
          as: "legend",
          element: "LEGEND",
          htmlFor: void 0,
          required,
          marginBottom: "space0",
          disabled,
          i18nRequiredLabel
        },
        legend
      ),
      helpText && /* @__PURE__ */ React4.createElement(HelpText, { variant: "default", marginTop: "space0" }, helpText),
      /* @__PURE__ */ React4.createElement(
        Box3,
        {
          as: attached ? AttachedRadioButtonGroupWrapper : UnattachedRadioButtonGroupWrapper,
          marginTop: "space20"
        },
        children
      ),
      errorText && /* @__PURE__ */ React4.createElement(HelpText, { variant: "error" }, errorText)
    ));
  }
);
RadioButtonGroup.displayName = "RadioButtonGroup";
export {
  RadioButton,
  RadioButtonGroup
};
