// @twilio-paste/form/dist/index.debug.es.js
// src/Form.tsx
import { Box, safelySpreadBoxProps } from "@twilio-paste/box";
import * as React from "react";
var Form = React.forwardRef(
  ({ element = "FORM", maxWidth, children, ...props }, ref) => /* @__PURE__ */ React.createElement(
    Box,
    {
      as: "form",
      ref,
      element,
      maxWidth,
      display: "flex",
      flexDirection: "column",
      rowGap: "space80",
      ...safelySpreadBoxProps(props)
    },
    children
  )
);
Form.displayName = "Form";

// src/FormActions.tsx
import { Box as Box2, safelySpreadBoxProps as safelySpreadBoxProps2 } from "@twilio-paste/box";
import * as React2 from "react";
var FormActions = React2.forwardRef(
  ({ children, element = "FORM_ACTIONS", ...props }, ref) => /* @__PURE__ */ React2.createElement(
    Box2,
    {
      ref,
      element,
      ...safelySpreadBoxProps2(props),
      display: "flex",
      flexDirection: "row",
      columnGap: "space40",
      marginTop: "space60"
    },
    children
  )
);
FormActions.displayName = "FormActions";

// src/FormControl.tsx
import { Box as Box3, safelySpreadBoxProps as safelySpreadBoxProps3 } from "@twilio-paste/box";
import * as React3 from "react";
var FormControl = React3.forwardRef(
  ({ children, element = "FORM_CONTROL", ...props }, ref) => /* @__PURE__ */ React3.createElement(Box3, { ref, flexGrow: 1, element, ...safelySpreadBoxProps3(props) }, children)
);
FormControl.displayName = "FormControl";

// src/FormControlTwoColumn.tsx
import { Box as Box4, safelySpreadBoxProps as safelySpreadBoxProps4 } from "@twilio-paste/box";
import * as React4 from "react";
var FormControlTwoColumn = React4.forwardRef(
  ({ children, element = "FORM_CONTROL_TWO_COLUMN", ...props }, ref) => /* @__PURE__ */ React4.createElement(
    Box4,
    {
      display: "grid",
      gridTemplateColumns: "1fr 1fr",
      columnGap: "space50",
      ref,
      element,
      ...safelySpreadBoxProps4(props)
    },
    children
  )
);
FormControlTwoColumn.displayName = "FormControlTwoColumn";

// src/FormSection.tsx
import { Box as Box5, safelySpreadBoxProps as safelySpreadBoxProps5 } from "@twilio-paste/box";
import { useUID } from "@twilio-paste/uid-library";
import * as React6 from "react";

// src/FormSectionContext.tsx
import * as React5 from "react";
var FormSectionContext = React5.createContext({});

// src/FormSection.tsx
var FormSection = React6.forwardRef(
  ({ children, element = "FORM_SECTION", id: idProp, ...props }, ref) => {
    const id = idProp ? idProp : useUID();
    return /* @__PURE__ */ React6.createElement(FormSectionContext.Provider, { value: { id } }, /* @__PURE__ */ React6.createElement(
      Box5,
      {
        as: "fieldset",
        border: "none",
        margin: "space0",
        padding: "space0",
        marginY: "space20",
        ref,
        element,
        display: "flex",
        flexDirection: "column",
        rowGap: "space80",
        _first: { marginTop: "space0" },
        _last: { marginBottom: "space0" },
        ...safelySpreadBoxProps5(props),
        "aria-describedby": `${id}-section-description`,
        id
      },
      children
    ));
  }
);
FormSection.displayName = "FormSection";

// src/FormSectionDescription.tsx
import { Box as Box6, safelySpreadBoxProps as safelySpreadBoxProps6 } from "@twilio-paste/box";
import * as React7 from "react";
var FormSectionDescription = React7.forwardRef(
  ({ children, element = "FORM_SECTION_DESCRIPTION", ...props }, ref) => {
    const { id: sectionId } = React7.useContext(FormSectionContext);
    return /* @__PURE__ */ React7.createElement(
      Box6,
      {
        ref,
        ...safelySpreadBoxProps6(props),
        element,
        color: "colorText",
        fontSize: "fontSize30",
        lineHeight: "lineHeight20",
        id: `${sectionId}-section-description`
      },
      children
    );
  }
);
FormSectionDescription.displayName = "FormSectionDescription";

// src/FormSectionHeading.tsx
import { Box as Box7, safelySpreadBoxProps as safelySpreadBoxProps7 } from "@twilio-paste/box";
import { Heading } from "@twilio-paste/heading";
import * as React8 from "react";
var FormSectionHeading = React8.forwardRef(
  ({ children, element = "FORM_SECTION_HEADING", variant = "heading40", ...props }, ref) => /* @__PURE__ */ React8.createElement(Box7, { as: "legend", ref, ...safelySpreadBoxProps7(props), element, padding: "space0" }, /* @__PURE__ */ React8.createElement(Heading, { as: "span", variant }, children))
);
FormSectionHeading.displayName = "FormSectionHeading";
export {
  Form,
  FormActions,
  FormControl,
  FormControlTwoColumn,
  FormSection,
  FormSectionDescription,
  FormSectionHeading
};
