// @twilio-paste/form/dist/index.debug.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  Form: () => Form,
  FormActions: () => FormActions,
  FormControl: () => FormControl,
  FormControlTwoColumn: () => FormControlTwoColumn,
  FormSection: () => FormSection,
  FormSectionDescription: () => FormSectionDescription,
  FormSectionHeading: () => FormSectionHeading
});
module.exports = __toCommonJS(src_exports);

// src/Form.tsx
var import_box = require("@twilio-paste/box");
var React = __toESM(require("react"));
var Form = React.forwardRef(
  ({ element = "FORM", maxWidth, children, ...props }, ref) => /* @__PURE__ */ React.createElement(
    import_box.Box,
    {
      as: "form",
      ref,
      element,
      maxWidth,
      display: "flex",
      flexDirection: "column",
      rowGap: "space80",
      ...(0, import_box.safelySpreadBoxProps)(props)
    },
    children
  )
);
Form.displayName = "Form";

// src/FormActions.tsx
var import_box2 = require("@twilio-paste/box");
var React2 = __toESM(require("react"));
var FormActions = React2.forwardRef(
  ({ children, element = "FORM_ACTIONS", ...props }, ref) => /* @__PURE__ */ React2.createElement(
    import_box2.Box,
    {
      ref,
      element,
      ...(0, import_box2.safelySpreadBoxProps)(props),
      display: "flex",
      flexDirection: "row",
      columnGap: "space40",
      marginTop: "space60"
    },
    children
  )
);
FormActions.displayName = "FormActions";

// src/FormControl.tsx
var import_box3 = require("@twilio-paste/box");
var React3 = __toESM(require("react"));
var FormControl = React3.forwardRef(
  ({ children, element = "FORM_CONTROL", ...props }, ref) => /* @__PURE__ */ React3.createElement(import_box3.Box, { ref, flexGrow: 1, element, ...(0, import_box3.safelySpreadBoxProps)(props) }, children)
);
FormControl.displayName = "FormControl";

// src/FormControlTwoColumn.tsx
var import_box4 = require("@twilio-paste/box");
var React4 = __toESM(require("react"));
var FormControlTwoColumn = React4.forwardRef(
  ({ children, element = "FORM_CONTROL_TWO_COLUMN", ...props }, ref) => /* @__PURE__ */ React4.createElement(
    import_box4.Box,
    {
      display: "grid",
      gridTemplateColumns: "1fr 1fr",
      columnGap: "space50",
      ref,
      element,
      ...(0, import_box4.safelySpreadBoxProps)(props)
    },
    children
  )
);
FormControlTwoColumn.displayName = "FormControlTwoColumn";

// src/FormSection.tsx
var import_box5 = require("@twilio-paste/box");
var import_uid_library = require("@twilio-paste/uid-library");
var React6 = __toESM(require("react"));

// src/FormSectionContext.tsx
var React5 = __toESM(require("react"));
var FormSectionContext = React5.createContext({});

// src/FormSection.tsx
var FormSection = React6.forwardRef(
  ({ children, element = "FORM_SECTION", id: idProp, ...props }, ref) => {
    const id = idProp ? idProp : (0, import_uid_library.useUID)();
    return /* @__PURE__ */ React6.createElement(FormSectionContext.Provider, { value: { id } }, /* @__PURE__ */ React6.createElement(
      import_box5.Box,
      {
        as: "fieldset",
        border: "none",
        margin: "space0",
        padding: "space0",
        marginY: "space20",
        ref,
        element,
        display: "flex",
        flexDirection: "column",
        rowGap: "space80",
        _first: { marginTop: "space0" },
        _last: { marginBottom: "space0" },
        ...(0, import_box5.safelySpreadBoxProps)(props),
        "aria-describedby": `${id}-section-description`,
        id
      },
      children
    ));
  }
);
FormSection.displayName = "FormSection";

// src/FormSectionDescription.tsx
var import_box6 = require("@twilio-paste/box");
var React7 = __toESM(require("react"));
var FormSectionDescription = React7.forwardRef(
  ({ children, element = "FORM_SECTION_DESCRIPTION", ...props }, ref) => {
    const { id: sectionId } = React7.useContext(FormSectionContext);
    return /* @__PURE__ */ React7.createElement(
      import_box6.Box,
      {
        ref,
        ...(0, import_box6.safelySpreadBoxProps)(props),
        element,
        color: "colorText",
        fontSize: "fontSize30",
        lineHeight: "lineHeight20",
        id: `${sectionId}-section-description`
      },
      children
    );
  }
);
FormSectionDescription.displayName = "FormSectionDescription";

// src/FormSectionHeading.tsx
var import_box7 = require("@twilio-paste/box");
var import_heading = require("@twilio-paste/heading");
var React8 = __toESM(require("react"));
var FormSectionHeading = React8.forwardRef(
  ({ children, element = "FORM_SECTION_HEADING", variant = "heading40", ...props }, ref) => /* @__PURE__ */ React8.createElement(import_box7.Box, { as: "legend", ref, ...(0, import_box7.safelySpreadBoxProps)(props), element, padding: "space0" }, /* @__PURE__ */ React8.createElement(import_heading.Heading, { as: "span", variant }, children))
);
FormSectionHeading.displayName = "FormSectionHeading";
